<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 20 Analysis of Variance | Biology 304: Biological Data Analysis</title>
  <meta name="description" content="Hands-on materials for Bio 304 course at Duke University." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 20 Analysis of Variance | Biology 304: Biological Data Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Hands-on materials for Bio 304 course at Duke University." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 20 Analysis of Variance | Biology 304: Biological Data Analysis" />
  
  <meta name="twitter:description" content="Hands-on materials for Bio 304 course at Duke University." />
  

<meta name="author" content="Paul M. Magwene" />


<meta name="date" content="2025-06-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="comparing-sample-means.html"/>
<link rel="next" href="violations-of-assumptions.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="libs/rglWebGL-binding-1.3.18/rglWebGL.js"></script>
<link href="libs/rglwidgetClass-1.3.18/rgl.css" rel="stylesheet" />
<script src="libs/rglwidgetClass-1.3.18/rglClass.min.js"></script>
<script src="libs/CanvasMatrix4-1.3.18/CanvasMatrix.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#how-to-use-these-lecture-notes"><i class="fa fa-check"></i><b>1.1</b> How to use these lecture notes</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html"><i class="fa fa-check"></i><b>2</b> Data story: Women and children first on the Titanic?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#background"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#dataset"><i class="fa fa-check"></i><b>2.2</b> Dataset</a></li>
<li class="chapter" data-level="2.3" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#libraries"><i class="fa fa-check"></i><b>2.3</b> Libraries</a></li>
<li class="chapter" data-level="2.4" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#read-data"><i class="fa fa-check"></i><b>2.4</b> Read data</a></li>
<li class="chapter" data-level="2.5" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#whats-in-the-data"><i class="fa fa-check"></i><b>2.5</b> What’s in the data?</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#simple-data-wrangling"><i class="fa fa-check"></i><b>2.5.1</b> Simple data wrangling</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#categorizing-passengers"><i class="fa fa-check"></i><b>2.6</b> Categorizing passengers</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#how-many-people-survived"><i class="fa fa-check"></i><b>2.6.1</b> How many people survived?</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#women-first"><i class="fa fa-check"></i><b>2.6.2</b> Women first?</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#a-bar-plot-using-proportions-rather-than-counts"><i class="fa fa-check"></i><b>2.6.3</b> A bar plot using proportions rather than counts</a></li>
<li class="chapter" data-level="2.6.4" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#mosaic-plots"><i class="fa fa-check"></i><b>2.6.4</b> Mosaic plots</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#passenger-ages"><i class="fa fa-check"></i><b>2.7</b> Passenger ages</a></li>
<li class="chapter" data-level="2.8" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#how-does-age-relate-to-survival"><i class="fa fa-check"></i><b>2.8</b> How does age relate to survival?</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#strip-charts"><i class="fa fa-check"></i><b>2.8.1</b> Strip charts</a></li>
<li class="chapter" data-level="2.8.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#box-plots"><i class="fa fa-check"></i><b>2.8.2</b> Box plots</a></li>
<li class="chapter" data-level="2.8.3" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#fitting-a-model-relating-survival-to-age-and-sex"><i class="fa fa-check"></i><b>2.8.3</b> Fitting a model relating survival to age and sex</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#how-does-class-affect-survival"><i class="fa fa-check"></i><b>2.9</b> How does class affect survival?</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#combining-sex-and-class"><i class="fa fa-check"></i><b>2.9.1</b> Combining Sex and Class</a></li>
<li class="chapter" data-level="2.9.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#extending-the-logistic-regression-model-to-consider-class-age-sex-and-survival"><i class="fa fa-check"></i><b>2.9.2</b> Extending the logistic regression model to consider class, age, sex, and survival</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#conclusion"><i class="fa fa-check"></i><b>2.10</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>3</b> Getting Started with R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#what-is-r"><i class="fa fa-check"></i><b>3.1</b> What is R?</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#what-is-rstudio"><i class="fa fa-check"></i><b>3.2</b> What is RStudio?</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#entering-commands-in-the-console"><i class="fa fa-check"></i><b>3.3</b> Entering commands in the console</a></li>
<li class="chapter" data-level="3.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#comments"><i class="fa fa-check"></i><b>3.4</b> Comments</a></li>
<li class="chapter" data-level="3.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#using-r-as-a-calculator"><i class="fa fa-check"></i><b>3.5</b> Using R as a Calculator</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#common-mathematical-functions"><i class="fa fa-check"></i><b>3.5.1</b> Common mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#the-r-help-system"><i class="fa fa-check"></i><b>3.6</b> The R Help System</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#getting-help-from-the-console"><i class="fa fa-check"></i><b>3.6.1</b> Getting help from the console</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#variable-assignment-in-r"><i class="fa fa-check"></i><b>3.7</b> Variable assignment in R</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#valid-variable-names"><i class="fa fa-check"></i><b>3.7.1</b> Valid variable names</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#data-types"><i class="fa fa-check"></i><b>3.8</b> Data types</a></li>
<li class="chapter" data-level="3.9" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#logical-values"><i class="fa fa-check"></i><b>3.9</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#logical-operators"><i class="fa fa-check"></i><b>3.9.1</b> Logical operators</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#character-strings"><i class="fa fa-check"></i><b>3.10</b> Character strings</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#joining-strings"><i class="fa fa-check"></i><b>3.10.1</b> Joining strings</a></li>
<li class="chapter" data-level="3.10.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#splitting-strings"><i class="fa fa-check"></i><b>3.10.2</b> Splitting strings</a></li>
<li class="chapter" data-level="3.10.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#substrings"><i class="fa fa-check"></i><b>3.10.3</b> Substrings</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#packages"><i class="fa fa-check"></i><b>3.11</b> Packages</a>
<ul>
<li class="chapter" data-level="3.11.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-packages-from-the-console"><i class="fa fa-check"></i><b>3.11.1</b> Installing packages from the console</a></li>
<li class="chapter" data-level="3.11.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#install-the-tidyverse-package"><i class="fa fa-check"></i><b>3.11.2</b> Install the tidyverse package</a></li>
<li class="chapter" data-level="3.11.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-packages-from-the-rstudio-dialog"><i class="fa fa-check"></i><b>3.11.3</b> Installing packages from the RStudio dialog</a></li>
<li class="chapter" data-level="3.11.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#loading-packages-with-the-library-function"><i class="fa fa-check"></i><b>3.11.4</b> Loading packages with the <code>library()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="3.12" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#reading-data-from-a-file"><i class="fa fa-check"></i><b>3.12</b> Reading data from a file</a></li>
<li class="chapter" data-level="3.13" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#exploring-the-possums-data-set"><i class="fa fa-check"></i><b>3.13</b> Exploring the “possums” data set</a></li>
<li class="chapter" data-level="3.14" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#simple-tables"><i class="fa fa-check"></i><b>3.14</b> Simple tables</a></li>
<li class="chapter" data-level="3.15" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#simple-figures"><i class="fa fa-check"></i><b>3.15</b> Simple figures</a></li>
<li class="chapter" data-level="3.16" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#bar-plots"><i class="fa fa-check"></i><b>3.16</b> Bar plots</a></li>
<li class="chapter" data-level="3.17" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#histograms"><i class="fa fa-check"></i><b>3.17</b> Histograms</a></li>
<li class="chapter" data-level="3.18" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#scatter-plots"><i class="fa fa-check"></i><b>3.18</b> Scatter plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html"><i class="fa fa-check"></i><b>4</b> R Markdown and R Notebooks</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#r-notebooks"><i class="fa fa-check"></i><b>4.1</b> R Notebooks</a></li>
<li class="chapter" data-level="4.2" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#creating-an-r-notebook"><i class="fa fa-check"></i><b>4.2</b> Creating an R Notebook</a></li>
<li class="chapter" data-level="4.3" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#the-default-r-notebook-template"><i class="fa fa-check"></i><b>4.3</b> The default R Notebook template</a></li>
<li class="chapter" data-level="4.4" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#code-and-non-code-blocks"><i class="fa fa-check"></i><b>4.4</b> Code and Non-code blocks</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#non-code-blocks"><i class="fa fa-check"></i><b>4.4.1</b> Non-code blocks</a></li>
<li class="chapter" data-level="4.4.2" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#code-blocks"><i class="fa fa-check"></i><b>4.4.2</b> Code blocks</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#running-a-code-chunk"><i class="fa fa-check"></i><b>4.5</b> Running a code chunk</a></li>
<li class="chapter" data-level="4.6" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#running-all-code-chunks-above"><i class="fa fa-check"></i><b>4.6</b> Running all code chunks above</a></li>
<li class="chapter" data-level="4.7" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#knitting-an-r-markdown-to-html"><i class="fa fa-check"></i><b>4.7</b> “Knitting an” R Markdown to HTML</a></li>
<li class="chapter" data-level="4.8" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#sharing-your-reproducible-r-notebook"><i class="fa fa-check"></i><b>4.8</b> Sharing your reproducible R Notebook</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html"><i class="fa fa-check"></i><b>5</b> More R Basics: Data structures</a>
<ul>
<li class="chapter" data-level="5.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vectors"><i class="fa fa-check"></i><b>5.1</b> Vectors</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vector-arithmetic"><i class="fa fa-check"></i><b>5.1.1</b> Vector Arithmetic</a></li>
<li class="chapter" data-level="5.1.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vector-recycling"><i class="fa fa-check"></i><b>5.1.2</b> Vector recycling</a></li>
<li class="chapter" data-level="5.1.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#simple-statistical-functions-for-numeric-vectors"><i class="fa fa-check"></i><b>5.1.3</b> Simple statistical functions for numeric vectors</a></li>
<li class="chapter" data-level="5.1.4" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#indexing-vectors"><i class="fa fa-check"></i><b>5.1.4</b> Indexing Vectors</a></li>
<li class="chapter" data-level="5.1.5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#comparison-operators-applied-to-vectors"><i class="fa fa-check"></i><b>5.1.5</b> Comparison operators applied to vectors</a></li>
<li class="chapter" data-level="5.1.6" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#combining-indexing-and-comparison-of-vectors"><i class="fa fa-check"></i><b>5.1.6</b> Combining Indexing and Comparison of Vectors</a></li>
<li class="chapter" data-level="5.1.7" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vector-manipulation"><i class="fa fa-check"></i><b>5.1.7</b> Vector manipulation</a></li>
<li class="chapter" data-level="5.1.8" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vectors-from-regular-sequences"><i class="fa fa-check"></i><b>5.1.8</b> Vectors from regular sequences</a></li>
<li class="chapter" data-level="5.1.9" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#additional-functions-for-working-with-vectors"><i class="fa fa-check"></i><b>5.1.9</b> Additional functions for working with vectors</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#lists"><i class="fa fa-check"></i><b>5.2</b> Lists</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#length-and-type-of-lists"><i class="fa fa-check"></i><b>5.2.1</b> Length and type of lists</a></li>
<li class="chapter" data-level="5.2.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#indexing-lists"><i class="fa fa-check"></i><b>5.2.2</b> Indexing lists</a></li>
<li class="chapter" data-level="5.2.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#naming-list-elements"><i class="fa fa-check"></i><b>5.2.3</b> Naming list elements</a></li>
<li class="chapter" data-level="5.2.4" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#the-operator"><i class="fa fa-check"></i><b>5.2.4</b> The <code>$</code> operator</a></li>
<li class="chapter" data-level="5.2.5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#changing-and-adding-lists-items"><i class="fa fa-check"></i><b>5.2.5</b> Changing and adding lists items</a></li>
<li class="chapter" data-level="5.2.6" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#combining-lists"><i class="fa fa-check"></i><b>5.2.6</b> Combining lists</a></li>
<li class="chapter" data-level="5.2.7" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#converting-lists-to-vectors"><i class="fa fa-check"></i><b>5.2.7</b> Converting lists to vectors</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#data-frames"><i class="fa fa-check"></i><b>5.3</b> Data frames</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#creating-a-data-frame"><i class="fa fa-check"></i><b>5.3.1</b> Creating a data frame</a></li>
<li class="chapter" data-level="5.3.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#type-and-class-for-data-frames"><i class="fa fa-check"></i><b>5.3.2</b> Type and class for data frames</a></li>
<li class="chapter" data-level="5.3.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#length-and-dimension-for-data-frames"><i class="fa fa-check"></i><b>5.3.3</b> Length and dimension for data frames</a></li>
<li class="chapter" data-level="5.3.4" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#indexing-and-accessing-data-frames"><i class="fa fa-check"></i><b>5.3.4</b> Indexing and accessing data frames</a></li>
<li class="chapter" data-level="5.3.5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#logical-indexing-of-data-frames"><i class="fa fa-check"></i><b>5.3.5</b> Logical indexing of data frames</a></li>
<li class="chapter" data-level="5.3.6" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#adding-columns-to-a-data-frame"><i class="fa fa-check"></i><b>5.3.6</b> Adding columns to a data frame</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html"><i class="fa fa-check"></i><b>6</b> Introduction to ggplot2</a>
<ul>
<li class="chapter" data-level="6.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#loading-ggplot2"><i class="fa fa-check"></i><b>6.1</b> Loading ggplot2</a></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#example-data-set-andersons-iris-data"><i class="fa fa-check"></i><b>6.2</b> Example data set: Anderson’s Iris Data</a></li>
<li class="chapter" data-level="6.3" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#template-for-single-layer-plots-in-ggplot2"><i class="fa fa-check"></i><b>6.3</b> Template for single layer plots in ggplot2</a></li>
<li class="chapter" data-level="6.4" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#an-aside-about-function-arguments"><i class="fa fa-check"></i><b>6.4</b> An aside about function arguments</a></li>
<li class="chapter" data-level="6.5" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#strip-plots"><i class="fa fa-check"></i><b>6.5</b> Strip plots</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#jittering-data"><i class="fa fa-check"></i><b>6.5.1</b> Jittering data</a></li>
<li class="chapter" data-level="6.5.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#adding-categorical-information"><i class="fa fa-check"></i><b>6.5.2</b> Adding categorical information</a></li>
<li class="chapter" data-level="6.5.3" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#rotating-plot-coordinates"><i class="fa fa-check"></i><b>6.5.3</b> Rotating plot coordinates</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#histograms-1"><i class="fa fa-check"></i><b>6.6</b> Histograms</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#variations-on-histograms-when-considering-categorical-data"><i class="fa fa-check"></i><b>6.6.1</b> Variations on histograms when considering categorical data</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#faceting-to-depict-categorical-information"><i class="fa fa-check"></i><b>6.7</b> Faceting to depict categorical information</a></li>
<li class="chapter" data-level="6.8" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#density-plots"><i class="fa fa-check"></i><b>6.8</b> Density plots</a></li>
<li class="chapter" data-level="6.9" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#violin-or-beanplot"><i class="fa fa-check"></i><b>6.9</b> Violin or Beanplot</a></li>
<li class="chapter" data-level="6.10" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#boxplots"><i class="fa fa-check"></i><b>6.10</b> Boxplots</a></li>
<li class="chapter" data-level="6.11" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#building-complex-visualizations-with-layers"><i class="fa fa-check"></i><b>6.11</b> Building complex visualizations with layers</a></li>
<li class="chapter" data-level="6.12" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#useful-combination-plots"><i class="fa fa-check"></i><b>6.12</b> Useful combination plots</a>
<ul>
<li class="chapter" data-level="6.12.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#boxplot-plus-strip-plot"><i class="fa fa-check"></i><b>6.12.1</b> Boxplot plus strip plot</a></li>
<li class="chapter" data-level="6.12.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#setting-shared-aesthetics"><i class="fa fa-check"></i><b>6.12.2</b> Setting shared aesthetics</a></li>
</ul></li>
<li class="chapter" data-level="6.13" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#ggplot-layers-can-be-assigned-to-variables"><i class="fa fa-check"></i><b>6.13</b> ggplot layers can be assigned to variables</a>
<ul>
<li class="chapter" data-level="6.13.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#violin-plot-plus-strip-plot"><i class="fa fa-check"></i><b>6.13.1</b> Violin plot plus strip plot</a></li>
</ul></li>
<li class="chapter" data-level="6.14" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#adding-titles-and-tweaking-axis-labels"><i class="fa fa-check"></i><b>6.14</b> Adding titles and tweaking axis labels</a></li>
<li class="chapter" data-level="6.15" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#ggplot2-themes"><i class="fa fa-check"></i><b>6.15</b> ggplot2 themes</a>
<ul>
<li class="chapter" data-level="6.15.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#further-customization-with-ggplot2theme"><i class="fa fa-check"></i><b>6.15.1</b> Further customization with <code>ggplot2::theme</code></a></li>
</ul></li>
<li class="chapter" data-level="6.16" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#other-aspects-of-ggplots-can-be-assigned-to-variables"><i class="fa fa-check"></i><b>6.16</b> Other aspects of ggplots can be assigned to variables</a></li>
<li class="chapter" data-level="6.17" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#bivariate-plots"><i class="fa fa-check"></i><b>6.17</b> Bivariate plots</a>
<ul>
<li class="chapter" data-level="6.17.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#scatter-plots-1"><i class="fa fa-check"></i><b>6.17.1</b> Scatter plots</a></li>
<li class="chapter" data-level="6.17.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#adding-a-trend-line-to-a-scatter-plot"><i class="fa fa-check"></i><b>6.17.2</b> Adding a trend line to a scatter plot</a></li>
</ul></li>
<li class="chapter" data-level="6.18" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#bivariate-density-plots"><i class="fa fa-check"></i><b>6.18</b> Bivariate density plots</a></li>
<li class="chapter" data-level="6.19" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#combining-scatter-plots-and-density-plots-with-categorical-information"><i class="fa fa-check"></i><b>6.19</b> Combining Scatter Plots and Density Plots with Categorical Information</a></li>
<li class="chapter" data-level="6.20" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#density-plots-with-fill"><i class="fa fa-check"></i><b>6.20</b> Density plots with fill</a></li>
<li class="chapter" data-level="6.21" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#d-bin-and-hex-plots"><i class="fa fa-check"></i><b>6.21</b> 2D bin and hex plots</a></li>
<li class="chapter" data-level="6.22" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#the-cowplot-package"><i class="fa fa-check"></i><b>6.22</b> The <code>cowplot</code> package</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html"><i class="fa fa-check"></i><b>7</b> Introduction to the <code>dplyr</code> package</a>
<ul>
<li class="chapter" data-level="7.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#libraries-1"><i class="fa fa-check"></i><b>7.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#reading-data-with-the-readr-package"><i class="fa fa-check"></i><b>7.2</b> Reading data with the <code>readr</code> package</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#example-data-nc-births"><i class="fa fa-check"></i><b>7.2.1</b> Example data: NC Births</a></li>
<li class="chapter" data-level="7.2.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#reading-excel-files"><i class="fa fa-check"></i><b>7.2.2</b> Reading Excel files</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#a-note-on-tibbles"><i class="fa fa-check"></i><b>7.3</b> A note on “tibbles”</a></li>
<li class="chapter" data-level="7.4" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#data-filtering-and-transformation-with-dplyr"><i class="fa fa-check"></i><b>7.4</b> Data filtering and transformation with <code>dplyr</code></a></li>
<li class="chapter" data-level="7.5" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#dplyrs-verbs"><i class="fa fa-check"></i><b>7.5</b> dplyr’s “verbs”</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#select"><i class="fa fa-check"></i><b>7.5.1</b> <code>select</code></a></li>
<li class="chapter" data-level="7.5.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#filter"><i class="fa fa-check"></i><b>7.5.2</b> <code>filter</code></a></li>
<li class="chapter" data-level="7.5.3" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#mutate"><i class="fa fa-check"></i><b>7.5.3</b> <code>mutate</code></a></li>
<li class="chapter" data-level="7.5.4" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#arrange"><i class="fa fa-check"></i><b>7.5.4</b> <code>arrange</code></a></li>
<li class="chapter" data-level="7.5.5" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#summarize"><i class="fa fa-check"></i><b>7.5.5</b> <code>summarize</code></a></li>
<li class="chapter" data-level="7.5.6" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#group_by"><i class="fa fa-check"></i><b>7.5.6</b> <code>group_by</code></a></li>
<li class="chapter" data-level="7.5.7" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#combining-grouping-and-summarizing"><i class="fa fa-check"></i><b>7.5.7</b> Combining grouping and summarizing</a></li>
<li class="chapter" data-level="7.5.8" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#scoped-variants-of-mutate-and-summarize"><i class="fa fa-check"></i><b>7.5.8</b> Scoped variants of <code>mutate</code> and <code>summarize</code></a></li>
<li class="chapter" data-level="7.5.9" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#combining-summarize-with-grouping-aesthetics-in-ggplot2"><i class="fa fa-check"></i><b>7.5.9</b> Combining summarize with grouping aesthetics in <code>ggplot2</code></a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#pipes"><i class="fa fa-check"></i><b>7.6</b> Pipes</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#install-and-load-magrittr"><i class="fa fa-check"></i><b>7.6.1</b> Install and load <code>magrittr</code></a></li>
<li class="chapter" data-level="7.6.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-basic-pipe-operator"><i class="fa fa-check"></i><b>7.6.2</b> The basic pipe operator</a></li>
<li class="chapter" data-level="7.6.3" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#an-example-without-pipes"><i class="fa fa-check"></i><b>7.6.3</b> An example without pipes</a></li>
<li class="chapter" data-level="7.6.4" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-same-example-using-pipes"><i class="fa fa-check"></i><b>7.6.4</b> The same example using pipes</a></li>
<li class="chapter" data-level="7.6.5" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#assigning-the-output-of-a-statement-involving-pipes-to-a-variable"><i class="fa fa-check"></i><b>7.6.5</b> Assigning the output of a statement involving pipes to a variable</a></li>
<li class="chapter" data-level="7.6.6" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#compound-assignment-pipe-operator"><i class="fa fa-check"></i><b>7.6.6</b> Compound assignment pipe operator</a></li>
<li class="chapter" data-level="7.6.7" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-dot-operator-with-pipes"><i class="fa fa-check"></i><b>7.6.7</b> The dot operator with pipes</a></li>
<li class="chapter" data-level="7.6.8" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-exposition-pipe-operator"><i class="fa fa-check"></i><b>7.6.8</b> The exposition pipe operator</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html"><i class="fa fa-check"></i><b>8</b> Data wrangling, Part I</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#libraries-2"><i class="fa fa-check"></i><b>8.1</b> Libraries</a></li>
<li class="chapter" data-level="8.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#data"><i class="fa fa-check"></i><b>8.2</b> Data</a></li>
<li class="chapter" data-level="8.3" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#renaming-data-frame-columms"><i class="fa fa-check"></i><b>8.3</b> Renaming data frame columms</a></li>
<li class="chapter" data-level="8.4" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#dropping-unneeded-columns"><i class="fa fa-check"></i><b>8.4</b> Dropping unneeded columns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#finding-all-empty-columns"><i class="fa fa-check"></i><b>8.4.1</b> Finding all empty columns</a></li>
<li class="chapter" data-level="8.4.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#dropping-columns-by-matching-names"><i class="fa fa-check"></i><b>8.4.2</b> Dropping columns by matching names</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#merging-data-frames"><i class="fa fa-check"></i><b>8.5</b> Merging data frames</a></li>
<li class="chapter" data-level="8.6" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#reshaping-data-with-tidyr"><i class="fa fa-check"></i><b>8.6</b> Reshaping data with tidyr</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#wide-to-long-conversions-using-tidyrgather"><i class="fa fa-check"></i><b>8.6.1</b> Wide to long conversions using <code>tidyr::gather</code></a></li>
<li class="chapter" data-level="8.6.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#extracting-information-from-combined-variables-using-tidyrextract"><i class="fa fa-check"></i><b>8.6.2</b> Extracting information from combined variables using <code>tidyr::extract</code></a></li>
<li class="chapter" data-level="8.6.3" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#subsetting-rows"><i class="fa fa-check"></i><b>8.6.3</b> Subsetting rows</a></li>
<li class="chapter" data-level="8.6.4" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#splitting-columns"><i class="fa fa-check"></i><b>8.6.4</b> Splitting columns</a></li>
<li class="chapter" data-level="8.6.5" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#combining-data-frames"><i class="fa fa-check"></i><b>8.6.5</b> Combining data frames</a></li>
<li class="chapter" data-level="8.6.6" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#sorting-data-frame-rows"><i class="fa fa-check"></i><b>8.6.6</b> Sorting data frame rows</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#using-your-tidy-data"><i class="fa fa-check"></i><b>8.7</b> Using your tidy data</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#visualizing-gene-expression-time-series"><i class="fa fa-check"></i><b>8.7.1</b> Visualizing gene expression time series</a></li>
<li class="chapter" data-level="8.7.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#finding-the-most-variable-genes"><i class="fa fa-check"></i><b>8.7.2</b> Finding the most variable genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html"><i class="fa fa-check"></i><b>9</b> Data wrangling, Part II</a>
<ul>
<li class="chapter" data-level="9.1" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#libraries-3"><i class="fa fa-check"></i><b>9.1</b> Libraries</a></li>
<li class="chapter" data-level="9.2" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#data-1"><i class="fa fa-check"></i><b>9.2</b> Data</a></li>
<li class="chapter" data-level="9.3" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#heat-maps"><i class="fa fa-check"></i><b>9.3</b> Heat maps</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#better-color-schemes-with-rcolorbrewer"><i class="fa fa-check"></i><b>9.3.1</b> Better color schemes with RColorBrewer</a></li>
<li class="chapter" data-level="9.3.2" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#looking-for-patterns-using-sorted-data-and-heat-maps"><i class="fa fa-check"></i><b>9.3.2</b> Looking for patterns using sorted data and heat maps</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#long-to-wide-conversion-using-tidyrspread"><i class="fa fa-check"></i><b>9.4</b> Long-to-wide conversion using <code>tidyr::spread</code></a></li>
<li class="chapter" data-level="9.5" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#exploring-bivariate-relationships-using-wide-data"><i class="fa fa-check"></i><b>9.5</b> Exploring bivariate relationships using “wide” data</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#large-scale-patterns-of-correlations"><i class="fa fa-check"></i><b>9.5.1</b> Large scale patterns of correlations</a></li>
<li class="chapter" data-level="9.5.2" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#adding-new-columns-and-combining-filtered-data-frames"><i class="fa fa-check"></i><b>9.5.2</b> Adding new columns and combining filtered data frames</a></li>
<li class="chapter" data-level="9.5.3" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#a-heat-mapped-sorted-by-correlations"><i class="fa fa-check"></i><b>9.5.3</b> A heat mapped sorted by correlations</a></li>
<li class="chapter" data-level="9.5.4" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#a-fancy-figure"><i class="fa fa-check"></i><b>9.5.4</b> A “fancy” figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html"><i class="fa fa-check"></i><b>10</b> Functions and control flow statements</a>
<ul>
<li class="chapter" data-level="10.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#writing-your-own-functions"><i class="fa fa-check"></i><b>10.1</b> Writing your own functions</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#function-arguments"><i class="fa fa-check"></i><b>10.1.1</b> Function arguments</a></li>
<li class="chapter" data-level="10.1.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#writing-functions-with-optional-arguments"><i class="fa fa-check"></i><b>10.1.2</b> Writing functions with optional arguments</a></li>
<li class="chapter" data-level="10.1.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#putting-r-functions-in-scripts"><i class="fa fa-check"></i><b>10.1.3</b> Putting R functions in Scripts</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#control-flow-statements"><i class="fa fa-check"></i><b>10.2</b> Control flow statements</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#if-and-if-else-statements"><i class="fa fa-check"></i><b>10.2.1</b> <code>if</code> and <code>if-else</code> statements</a></li>
<li class="chapter" data-level="10.2.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#for-loops"><i class="fa fa-check"></i><b>10.2.2</b> for loops</a></li>
<li class="chapter" data-level="10.2.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#break-statement"><i class="fa fa-check"></i><b>10.2.3</b> <code>break</code> statement</a></li>
<li class="chapter" data-level="10.2.4" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#repeat-loops"><i class="fa fa-check"></i><b>10.2.4</b> <code>repeat</code> loops</a></li>
<li class="chapter" data-level="10.2.5" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#next-statement"><i class="fa fa-check"></i><b>10.2.5</b> <code>next</code> statement</a></li>
<li class="chapter" data-level="10.2.6" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#while-statements"><i class="fa fa-check"></i><b>10.2.6</b> while statements</a></li>
<li class="chapter" data-level="10.2.7" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#ifelse"><i class="fa fa-check"></i><b>10.2.7</b> <code>ifelse</code></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#map-and-related-tools"><i class="fa fa-check"></i><b>10.3</b> <code>map</code> and related tools</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#basic-map"><i class="fa fa-check"></i><b>10.3.1</b> basic <code>map</code></a></li>
<li class="chapter" data-level="10.3.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#map_if-and-map_at"><i class="fa fa-check"></i><b>10.3.2</b> <code>map_if</code> and <code>map_at</code></a></li>
<li class="chapter" data-level="10.3.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#mapping-in-parallel-using-map2"><i class="fa fa-check"></i><b>10.3.3</b> mapping in parallel using <code>map2</code></a></li>
<li class="chapter" data-level="10.3.4" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#map-variants-that-return-vectors"><i class="fa fa-check"></i><b>10.3.4</b> <code>map</code> variants that return vectors</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="frequency-distributions-and-descriptive-statistics.html"><a href="frequency-distributions-and-descriptive-statistics.html"><i class="fa fa-check"></i><b>11</b> Frequency Distributions and Descriptive Statistics</a></li>
<li class="chapter" data-level="12" data-path="joint-frequency-distributions-and-measures-of-association.html"><a href="joint-frequency-distributions-and-measures-of-association.html"><i class="fa fa-check"></i><b>12</b> Joint Frequency Distributions and Measures of Association</a></li>
<li class="chapter" data-level="13" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html"><i class="fa fa-check"></i><b>13</b> Introduction to Probability</a>
<ul>
<li class="chapter" data-level="13.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#terms"><i class="fa fa-check"></i><b>13.1</b> Terms</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#examples-of-random-trials-outcomes-and-events"><i class="fa fa-check"></i><b>13.1.1</b> Examples of random trials, outcomes, and events</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#frequentist-definition-of-probability"><i class="fa fa-check"></i><b>13.2</b> Frequentist definition of probability</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#examples-probability"><i class="fa fa-check"></i><b>13.2.1</b> Examples: Probability</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#probability-distribution"><i class="fa fa-check"></i><b>13.3</b> Probability distribution</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#discrete-probability-distribution"><i class="fa fa-check"></i><b>13.3.1</b> Discrete probability distribution</a></li>
<li class="chapter" data-level="13.3.2" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#continuous-probability-distribution"><i class="fa fa-check"></i><b>13.3.2</b> Continuous probability distribution</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#mutually-exclusive-events"><i class="fa fa-check"></i><b>13.4</b> Mutually exclusive events</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#addition-rule-mutually-exclusive-events"><i class="fa fa-check"></i><b>13.4.1</b> Addition rule, mutually exclusive events</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#independence"><i class="fa fa-check"></i><b>13.5</b> Independence</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#multiplication-rule-independent-events"><i class="fa fa-check"></i><b>13.5.1</b> Multiplication rule, independent events</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#general-addition-rule"><i class="fa fa-check"></i><b>13.6</b> General addition rule</a></li>
<li class="chapter" data-level="13.7" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#conditional-probability"><i class="fa fa-check"></i><b>13.7</b> Conditional probability</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#example-conditional-probability"><i class="fa fa-check"></i><b>13.7.1</b> Example: Conditional probability</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#general-multiplication-rule"><i class="fa fa-check"></i><b>13.8</b> General multiplication rule</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#example-general-multiplication-rule"><i class="fa fa-check"></i><b>13.8.1</b> Example: General multiplication rule</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#probability-trees"><i class="fa fa-check"></i><b>13.9</b> Probability trees</a></li>
<li class="chapter" data-level="13.10" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#law-of-total-probability"><i class="fa fa-check"></i><b>13.10</b> Law of total probability</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html"><i class="fa fa-check"></i><b>14</b> Introduction to Sampling Distributions, Part I</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#libraries-4"><i class="fa fa-check"></i><b>14.1</b> Libraries</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#in-class-experiments"><i class="fa fa-check"></i><b>14.2</b> In class experiments</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#experiment-1"><i class="fa fa-check"></i><b>14.2.1</b> Experiment 1</a></li>
<li class="chapter" data-level="14.2.2" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#experiment-2"><i class="fa fa-check"></i><b>14.2.2</b> Experiment 2</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#simulating-sampling-in-r"><i class="fa fa-check"></i><b>14.3</b> Simulating sampling in R</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#binomial-distribution"><i class="fa fa-check"></i><b>14.3.1</b> Binomial distribution</a></li>
<li class="chapter" data-level="14.3.2" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#sampling-from-the-binomial-distribution-in-r"><i class="fa fa-check"></i><b>14.3.2</b> Sampling from the binomial distribution in R</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#distribution-of-estimates-of-the-proportion"><i class="fa fa-check"></i><b>14.4</b> Distribution of estimates of the proportion</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html"><i class="fa fa-check"></i><b>15</b> Introduction to Sampling Distributions, Part II</a>
<ul>
<li class="chapter" data-level="15.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#libraries-5"><i class="fa fa-check"></i><b>15.1</b> Libraries</a></li>
<li class="chapter" data-level="15.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#data-set-simulated-male-heights"><i class="fa fa-check"></i><b>15.2</b> Data set: Simulated male heights</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#properties-of-the-underlying-population"><i class="fa fa-check"></i><b>15.2.1</b> Properties of the underlying population</a></li>
<li class="chapter" data-level="15.2.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#other-r-functions-related-to-the-normal-distribution"><i class="fa fa-check"></i><b>15.2.2</b> Other R functions related to the normal distribution</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#seeding-the-pseudo-random-number-generator"><i class="fa fa-check"></i><b>15.3</b> Seeding the pseudo-random number generator</a></li>
<li class="chapter" data-level="15.4" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#random-sampling-from-the-simulated-population"><i class="fa fa-check"></i><b>15.4</b> Random sampling from the simulated population</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#another-random-sample"><i class="fa fa-check"></i><b>15.4.1</b> Another random sample</a></li>
<li class="chapter" data-level="15.4.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#simulating-the-generation-of-many-random-samples"><i class="fa fa-check"></i><b>15.4.2</b> Simulating the generation of many random samples</a></li>
<li class="chapter" data-level="15.4.3" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#a-function-to-estimate-statistics-of-interest-in-a-random-sample"><i class="fa fa-check"></i><b>15.4.3</b> A function to estimate statistics of interest in a random sample</a></li>
<li class="chapter" data-level="15.4.4" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#generating-statistics-for-many-random-samples"><i class="fa fa-check"></i><b>15.4.4</b> Generating statistics for many random samples</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#simulated-sampling-distribution-of-the-mean"><i class="fa fa-check"></i><b>15.5</b> Simulated sampling distribution of the mean</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#differences-between-sampling-distribution-and-samplepopulation-distributions"><i class="fa fa-check"></i><b>15.5.1</b> Differences between sampling distribution and sample/population distributions</a></li>
<li class="chapter" data-level="15.5.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#use-sampling-distributions-to-understand-the-behavior-of-statistics-of-interest"><i class="fa fa-check"></i><b>15.5.2</b> Use sampling distributions to understand the behavior of statistics of interest</a></li>
<li class="chapter" data-level="15.5.3" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#sampling-distributions-for-different-sample-sizes"><i class="fa fa-check"></i><b>15.5.3</b> Sampling distributions for different sample sizes</a></li>
<li class="chapter" data-level="15.5.4" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#discussion-of-trends-for-sampling-distributions-of-different-sample-sizes"><i class="fa fa-check"></i><b>15.5.4</b> Discussion of trends for sampling distributions of different sample sizes</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>15.6</b> Standard Error of the Mean</a></li>
<li class="chapter" data-level="15.7" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#sampling-distribution-of-the-standard-deviation"><i class="fa fa-check"></i><b>15.7</b> Sampling Distribution of the Standard Deviation</a>
<ul>
<li class="chapter" data-level="15.7.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#standard-error-of-standard-deviations"><i class="fa fa-check"></i><b>15.7.1</b> Standard error of standard deviations</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#what-happens-to-the-sampling-distribution-of-the-mean-and-standard-deviation-when-our-sample-size-is-small"><i class="fa fa-check"></i><b>15.8</b> What happens to the sampling distribution of the mean and standard deviation when our sample size is small?</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#for-small-samples-sample-standard-deviations-systematically-underestimate-the-population-standard-deviation"><i class="fa fa-check"></i><b>15.8.1</b> For small samples, sample standard deviations systematically underestimate the population standard deviation</a></li>
<li class="chapter" data-level="15.8.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#underestimates-of-the-standard-deviation-given-small-n-lead-to-understimates-of-the-se-of-the-mean"><i class="fa fa-check"></i><b>15.8.2</b> Underestimates of the standard deviation given small <span class="math inline">\(n\)</span> lead to understimates of the SE of the mean</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html"><i class="fa fa-check"></i><b>16</b> Introduction to hypothesis testing</a>
<ul>
<li class="chapter" data-level="16.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#libraries-6"><i class="fa fa-check"></i><b>16.1</b> Libraries</a></li>
<li class="chapter" data-level="16.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses"><i class="fa fa-check"></i><b>16.2</b> Null and Alternative Hypotheses</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-hypotheses"><i class="fa fa-check"></i><b>16.2.1</b> Null hypotheses</a></li>
<li class="chapter" data-level="16.2.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#alternative-hypotheses"><i class="fa fa-check"></i><b>16.2.2</b> Alternative hypotheses</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#rejecting-failing-to-reject-null-hypotheses"><i class="fa fa-check"></i><b>16.3</b> Rejecting / failing to reject null hypotheses</a></li>
<li class="chapter" data-level="16.4" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#outcomes-of-hypothesis-tests"><i class="fa fa-check"></i><b>16.4</b> Outcomes of hypothesis tests</a></li>
<li class="chapter" data-level="16.5" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#using-p-values-to-assess-the-strength-of-evidence-against-the-null-hypothesis"><i class="fa fa-check"></i><b>16.5</b> Using p-values to assess the strength of evidence against the null hypothesis</a></li>
<li class="chapter" data-level="16.6" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#example-comparing-a-sample-mean-to-an-hypothesized-normal-distribution"><i class="fa fa-check"></i><b>16.6</b> Example: Comparing a sample mean to an hypothesized normal distribution</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses-1"><i class="fa fa-check"></i><b>16.6.1</b> Null and alternative hypotheses</a></li>
<li class="chapter" data-level="16.6.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#sampling-distribution-of-the-mean"><i class="fa fa-check"></i><b>16.6.2</b> Sampling distribution of the mean</a></li>
<li class="chapter" data-level="16.6.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#calculating-a-p-value"><i class="fa fa-check"></i><b>16.6.3</b> Calculating a p-value</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#example-handedness-of-toads"><i class="fa fa-check"></i><b>16.7</b> Example: Handedness of toads</a>
<ul>
<li class="chapter" data-level="16.7.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses-2"><i class="fa fa-check"></i><b>16.7.1</b> Null and alternative hypotheses</a></li>
<li class="chapter" data-level="16.7.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#data-2"><i class="fa fa-check"></i><b>16.7.2</b> Data</a></li>
<li class="chapter" data-level="16.7.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#sampling-distribution-for-proportions-binomial-distribution"><i class="fa fa-check"></i><b>16.7.3</b> Sampling distribution for proportions: Binomial distribution</a></li>
<li class="chapter" data-level="16.7.4" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#calculating-a-p-value-for-the-binomial-test"><i class="fa fa-check"></i><b>16.7.4</b> Calculating a p-value for the binomial test</a></li>
<li class="chapter" data-level="16.7.5" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#the-binom.test-function"><i class="fa fa-check"></i><b>16.7.5</b> The <code>binom.test()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#example-measuring-the-association-between-maternal-smoking-and-premature-births"><i class="fa fa-check"></i><b>16.8</b> Example: Measuring the association between maternal smoking and premature births</a>
<ul>
<li class="chapter" data-level="16.8.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses-3"><i class="fa fa-check"></i><b>16.8.1</b> Null and alternative hypotheses:</a></li>
<li class="chapter" data-level="16.8.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#contingency-table-analysis-using-the-chi2-statistic"><i class="fa fa-check"></i><b>16.8.2</b> Contingency table analysis using the <span class="math inline">\(\chi^2\)</span> statistic</a></li>
<li class="chapter" data-level="16.8.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#chi2-distribution"><i class="fa fa-check"></i><b>16.8.3</b> <span class="math inline">\(\chi^2\)</span>-distribution</a></li>
<li class="chapter" data-level="16.8.4" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#carrying-out-a-hypothesis-test-using-the-chi2"><i class="fa fa-check"></i><b>16.8.4</b> Carrying out a hypothesis test using the <span class="math inline">\(\chi^2\)</span></a></li>
<li class="chapter" data-level="16.8.5" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#interpretation-of-the-chi2-test"><i class="fa fa-check"></i><b>16.8.5</b> Interpretation of the <span class="math inline">\(\chi^2\)</span>-test</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html"><i class="fa fa-check"></i><b>17</b> Introduction to confidence intervals</a>
<ul>
<li class="chapter" data-level="17.1" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#confidence-intervals"><i class="fa fa-check"></i><b>17.1</b> Confidence Intervals</a></li>
<li class="chapter" data-level="17.2" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#generic-formulation-for-confidence-intervals"><i class="fa fa-check"></i><b>17.2</b> Generic formulation for confidence intervals</a></li>
<li class="chapter" data-level="17.3" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#example-confidence-intervals-for-the-mean"><i class="fa fa-check"></i><b>17.3</b> Example: Confidence intervals for the mean</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#simulation-of-means"><i class="fa fa-check"></i><b>17.3.1</b> Simulation of means</a></li>
<li class="chapter" data-level="17.3.2" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#distance-between-sample-means-and-true-means"><i class="fa fa-check"></i><b>17.3.2</b> Distance between sample means and true means</a></li>
<li class="chapter" data-level="17.3.3" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#calculating-a-ci"><i class="fa fa-check"></i><b>17.3.3</b> Calculating a CI</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#a-problem-arises"><i class="fa fa-check"></i><b>17.4</b> A problem arises!</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="normal-distributions.html"><a href="normal-distributions.html"><i class="fa fa-check"></i><b>18</b> Normal distributions</a>
<ul>
<li class="chapter" data-level="18.1" data-path="normal-distributions.html"><a href="normal-distributions.html#basics-about-normal-distributions"><i class="fa fa-check"></i><b>18.1</b> Basics about normal distributions</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="normal-distributions.html"><a href="normal-distributions.html#notation"><i class="fa fa-check"></i><b>18.1.1</b> Notation</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="normal-distributions.html"><a href="normal-distributions.html#normal-distribution-probability-density-function"><i class="fa fa-check"></i><b>18.2</b> Normal distribution, probability density function</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="normal-distributions.html"><a href="normal-distributions.html#dnorm-calculates-the-normal-pdf"><i class="fa fa-check"></i><b>18.2.1</b> <code>dnorm()</code> calculates the normal pdf</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="normal-distributions.html"><a href="normal-distributions.html#approximately-normal-distributions-are-very-common"><i class="fa fa-check"></i><b>18.3</b> Approximately normal distributions are very common</a></li>
<li class="chapter" data-level="18.4" data-path="normal-distributions.html"><a href="normal-distributions.html#central-limit-theorem"><i class="fa fa-check"></i><b>18.4</b> Central limit theorem</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="normal-distributions.html"><a href="normal-distributions.html#example-continuous-variation-from-discrete-loci"><i class="fa fa-check"></i><b>18.4.1</b> Example: Continuous variation from discrete loci</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="normal-distributions.html"><a href="normal-distributions.html#visualizing-normal-distributions"><i class="fa fa-check"></i><b>18.5</b> Visualizing normal distributions</a></li>
<li class="chapter" data-level="18.6" data-path="normal-distributions.html"><a href="normal-distributions.html#comparing-values-from-different-normal-distributions"><i class="fa fa-check"></i><b>18.6</b> Comparing values from different normal distributions</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="normal-distributions.html"><a href="normal-distributions.html#standardized-or-z-scores"><i class="fa fa-check"></i><b>18.6.1</b> Standardized or Z-scores</a></li>
</ul></li>
<li class="chapter" data-level="18.7" data-path="normal-distributions.html"><a href="normal-distributions.html#standard-normal-distribution"><i class="fa fa-check"></i><b>18.7</b> Standard normal distribution</a></li>
<li class="chapter" data-level="18.8" data-path="normal-distributions.html"><a href="normal-distributions.html#rule"><i class="fa fa-check"></i><b>18.8</b> 88-95-99.7 Rule</a></li>
<li class="chapter" data-level="18.9" data-path="normal-distributions.html"><a href="normal-distributions.html#percentiles"><i class="fa fa-check"></i><b>18.9</b> Percentiles</a></li>
<li class="chapter" data-level="18.10" data-path="normal-distributions.html"><a href="normal-distributions.html#cutoff-points"><i class="fa fa-check"></i><b>18.10</b> Cutoff points</a></li>
<li class="chapter" data-level="18.11" data-path="normal-distributions.html"><a href="normal-distributions.html#assessing-normality"><i class="fa fa-check"></i><b>18.11</b> Assessing normality</a>
<ul>
<li class="chapter" data-level="18.11.1" data-path="normal-distributions.html"><a href="normal-distributions.html#comparing-histograms-to-theoretical-normals"><i class="fa fa-check"></i><b>18.11.1</b> Comparing histograms to theoretical normals</a></li>
<li class="chapter" data-level="18.11.2" data-path="normal-distributions.html"><a href="normal-distributions.html#normal-probability-plot"><i class="fa fa-check"></i><b>18.11.2</b> Normal probability plot</a></li>
<li class="chapter" data-level="18.11.3" data-path="normal-distributions.html"><a href="normal-distributions.html#comparing-the-empirical-cdf-to-the-theoretical-cdf"><i class="fa fa-check"></i><b>18.11.3</b> Comparing the empirical CDF to the theoretical CDF</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html"><i class="fa fa-check"></i><b>19</b> Comparing sample means</a>
<ul>
<li class="chapter" data-level="19.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#hypothesis-test-for-the-mean-using-the-t-distribution"><i class="fa fa-check"></i><b>19.1</b> Hypothesis test for the mean using the t-distribution</a></li>
<li class="chapter" data-level="19.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#one-sample-t-test"><i class="fa fa-check"></i><b>19.2</b> One sample t-test</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#one-sample-t-test-test-statistic"><i class="fa fa-check"></i><b>19.2.1</b> One sample t-test, test statistic</a></li>
<li class="chapter" data-level="19.2.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#assumptions-of-one-sample-t-tests"><i class="fa fa-check"></i><b>19.2.2</b> Assumptions of one sample t-tests</a></li>
<li class="chapter" data-level="19.2.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#example-gene-expression-in-mice"><i class="fa fa-check"></i><b>19.2.3</b> Example: Gene expression in mice</a></li>
<li class="chapter" data-level="19.2.4" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#confidence-intervals-for-the-mean"><i class="fa fa-check"></i><b>19.2.4</b> Confidence intervals for the mean</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#the-t.test-function-in-r"><i class="fa fa-check"></i><b>19.3</b> The <code>t.test</code> function in R</a></li>
<li class="chapter" data-level="19.4" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#two-sample-t-test"><i class="fa fa-check"></i><b>19.4</b> Two sample t-test</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#standard-error-for-the-difference-in-means"><i class="fa fa-check"></i><b>19.4.1</b> Standard error for the difference in means</a></li>
<li class="chapter" data-level="19.4.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#two-sample-t-test-test-statistic"><i class="fa fa-check"></i><b>19.4.2</b> Two sample t-test, test statistic</a></li>
<li class="chapter" data-level="19.4.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#assumptions-of-two-sample-t-test"><i class="fa fa-check"></i><b>19.4.3</b> Assumptions of two sample t-test</a></li>
<li class="chapter" data-level="19.4.4" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#example-comparing-the-effects-of-two-drugs"><i class="fa fa-check"></i><b>19.4.4</b> Example: Comparing the effects of two drugs</a></li>
<li class="chapter" data-level="19.4.5" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#specifying-t.test-in-terms-of-a-formula"><i class="fa fa-check"></i><b>19.4.5</b> Specifying <code>t.test()</code> in terms of a formula</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#paired-t-test"><i class="fa fa-check"></i><b>19.5</b> Paired t-test</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#paired-t-test-test-statistic"><i class="fa fa-check"></i><b>19.5.1</b> Paired t-test, test statistic</a></li>
<li class="chapter" data-level="19.5.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#assumptions-of-paired-t-test"><i class="fa fa-check"></i><b>19.5.2</b> Assumptions of paired t-test</a></li>
<li class="chapter" data-level="19.5.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#paired-t-test-example"><i class="fa fa-check"></i><b>19.5.3</b> Paired t-test, example</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#the-fallacy-of-indirect-comparison"><i class="fa fa-check"></i><b>19.6</b> The fallacy of indirect comparison</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#interpreting-confidence-intervals-in-light-of-two-sample-t-tests"><i class="fa fa-check"></i><b>19.6.1</b> Interpreting confidence intervals in light of two sample t-tests</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#summary-table-for-different-t-tests"><i class="fa fa-check"></i><b>19.7</b> Summary table for different t-tests</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html"><i class="fa fa-check"></i><b>20</b> Analysis of Variance</a>
<ul>
<li class="chapter" data-level="20.1" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#hypotheses-for-anova"><i class="fa fa-check"></i><b>20.1</b> Hypotheses for ANOVA</a></li>
<li class="chapter" data-level="20.2" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-assumptions"><i class="fa fa-check"></i><b>20.2</b> ANOVA, assumptions</a></li>
<li class="chapter" data-level="20.3" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-key-idea"><i class="fa fa-check"></i><b>20.3</b> ANOVA, key idea</a></li>
<li class="chapter" data-level="20.4" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#partioning-of-sum-of-squares"><i class="fa fa-check"></i><b>20.4</b> Partioning of sum of squares</a></li>
<li class="chapter" data-level="20.5" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#mathematical-partitioning-of-sums-of-squares"><i class="fa fa-check"></i><b>20.5</b> Mathematical partitioning of sums of squares</a></li>
<li class="chapter" data-level="20.6" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-test-statistic-and-sampling-distribution"><i class="fa fa-check"></i><b>20.6</b> ANOVA test statistic and sampling distribution</a></li>
<li class="chapter" data-level="20.7" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-tables"><i class="fa fa-check"></i><b>20.7</b> ANOVA tables</a></li>
<li class="chapter" data-level="20.8" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#the-aov-function"><i class="fa fa-check"></i><b>20.8</b> The <code>aov()</code> function</a></li>
<li class="chapter" data-level="20.9" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#example-circadian-rythm-data"><i class="fa fa-check"></i><b>20.9</b> Example, circadian rythm data</a>
<ul>
<li class="chapter" data-level="20.9.1" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#visualizing-the-data"><i class="fa fa-check"></i><b>20.9.1</b> Visualizing the data</a></li>
<li class="chapter" data-level="20.9.2" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#carrying-out-the-anova"><i class="fa fa-check"></i><b>20.9.2</b> Carrying out the ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="20.10" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-calculations-step-by-step"><i class="fa fa-check"></i><b>20.10</b> ANOVA calculations: Step-by-step</a></li>
<li class="chapter" data-level="20.11" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#visualizing-the-partitioning-of-sum-of-squares"><i class="fa fa-check"></i><b>20.11</b> Visualizing the partitioning of sum-of-squares</a></li>
<li class="chapter" data-level="20.12" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#which-pairs-of-group-means-are-different"><i class="fa fa-check"></i><b>20.12</b> Which pairs of group means are different?</a>
<ul>
<li class="chapter" data-level="20.12.1" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#tukey-kramer-test"><i class="fa fa-check"></i><b>20.12.1</b> Tukey-Kramer test</a></li>
</ul></li>
<li class="chapter" data-level="20.13" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#repeatability"><i class="fa fa-check"></i><b>20.13</b> Repeatability</a>
<ul>
<li class="chapter" data-level="" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#estimating-repeatability-using-anova"><i class="fa fa-check"></i>Estimating Repeatability using ANOVA</a></li>
<li class="chapter" data-level="" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#repeatability-walking-stick-example"><i class="fa fa-check"></i>Repeatability: Walking stick example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html"><i class="fa fa-check"></i><b>21</b> Violations of Assumptions</a>
<ul>
<li class="chapter" data-level="21.1" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#graphical-methods-for-detecting-deviations-from-normality"><i class="fa fa-check"></i><b>21.1</b> Graphical methods for detecting deviations from normality</a>
<ul>
<li class="chapter" data-level="" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#normal-probability-plots"><i class="fa fa-check"></i>Normal probability plots</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#formal-test-of-normality"><i class="fa fa-check"></i><b>21.2</b> Formal test of normality</a></li>
<li class="chapter" data-level="21.3" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#when-to-ignore-violations-of-assumptions"><i class="fa fa-check"></i><b>21.3</b> When to ignore violations of assumptions</a></li>
<li class="chapter" data-level="21.4" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#data-transformations"><i class="fa fa-check"></i><b>21.4</b> Data transformations</a>
<ul>
<li class="chapter" data-level="" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#example-log-transformation-to-deal-with-skew"><i class="fa fa-check"></i>Example: log transformation to deal with skew</a></li>
<li class="chapter" data-level="21.4.1" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#confidence-intervals-and-data-transformations"><i class="fa fa-check"></i><b>21.4.1</b> Confidence intervals and data transformations</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#nonparametric-tests"><i class="fa fa-check"></i><b>21.5</b> Nonparametric tests</a>
<ul>
<li class="chapter" data-level="21.5.1" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#mann-whitney-u-test-non-parametric-alternative-to-t-test"><i class="fa fa-check"></i><b>21.5.1</b> Mann-Whitney U-test: non-parametric alternative to t-test</a></li>
<li class="chapter" data-level="21.5.2" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#kruskal-wallis-test-non-parametric-alternative-to-anova"><i class="fa fa-check"></i><b>21.5.2</b> Kruskal-Wallis test: non-parametric alternative to ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html"><i class="fa fa-check"></i><b>22</b> Bivariate Linear Regression</a>
<ul>
<li class="chapter" data-level="22.1" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#regression-terminology"><i class="fa fa-check"></i><b>22.1</b> Regression terminology</a></li>
<li class="chapter" data-level="22.2" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#the-optimality-criterion-for-least-squares-regression"><i class="fa fa-check"></i><b>22.2</b> The optimality criterion for least-squares regression</a></li>
<li class="chapter" data-level="22.3" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#solution-for-the-least-squares-criterion"><i class="fa fa-check"></i><b>22.3</b> Solution for the least-squares criterion</a></li>
<li class="chapter" data-level="22.4" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#libraries-7"><i class="fa fa-check"></i><b>22.4</b> Libraries</a></li>
<li class="chapter" data-level="22.5" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#illustrating-linear-regression-with-simulated-data"><i class="fa fa-check"></i><b>22.5</b> Illustrating linear regression with simulated data</a></li>
<li class="chapter" data-level="22.6" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#specifying-regression-models-in-r"><i class="fa fa-check"></i><b>22.6</b> Specifying Regression Models in R</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#fitted-values"><i class="fa fa-check"></i><b>22.6.1</b> Fitted values</a></li>
<li class="chapter" data-level="22.6.2" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#getting-the-model-coefficients"><i class="fa fa-check"></i><b>22.6.2</b> Getting the model coefficients</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#residuals"><i class="fa fa-check"></i><b>22.7</b> Residuals</a></li>
<li class="chapter" data-level="22.8" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#regression-as-sum-of-squares-decomposition"><i class="fa fa-check"></i><b>22.8</b> Regression as sum-of-squares decomposition</a></li>
<li class="chapter" data-level="22.9" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#variance-explained-by-a-regression-model"><i class="fa fa-check"></i><b>22.9</b> Variance “explained” by a regression model</a></li>
<li class="chapter" data-level="22.10" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broom-a-library-for-converting-model-results-into-data-frames"><i class="fa fa-check"></i><b>22.10</b> Broom: a library for converting model results into data frames</a>
<ul>
<li class="chapter" data-level="22.10.1" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broomtidy"><i class="fa fa-check"></i><b>22.10.1</b> <code>broom::tidy</code></a></li>
<li class="chapter" data-level="22.10.2" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broomaugment"><i class="fa fa-check"></i><b>22.10.2</b> <code>broom::augment</code></a></li>
<li class="chapter" data-level="22.10.3" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broomglance"><i class="fa fa-check"></i><b>22.10.3</b> <code>broom::glance</code></a></li>
</ul></li>
<li class="chapter" data-level="22.11" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#example-predicting-lion-age-based-on-nose-color"><i class="fa fa-check"></i><b>22.11</b> Example: Predicting lion age based on nose color</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="multiple-regression.html"><a href="multiple-regression.html"><i class="fa fa-check"></i><b>23</b> Multiple regression</a>
<ul>
<li class="chapter" data-level="23.1" data-path="multiple-regression.html"><a href="multiple-regression.html#libraries-to-install"><i class="fa fa-check"></i><b>23.1</b> Libraries to install</a></li>
<li class="chapter" data-level="23.2" data-path="multiple-regression.html"><a href="multiple-regression.html#review-of-bivariate-regression"><i class="fa fa-check"></i><b>23.2</b> Review of bivariate regression</a></li>
<li class="chapter" data-level="23.3" data-path="multiple-regression.html"><a href="multiple-regression.html#multiple-regression-1"><i class="fa fa-check"></i><b>23.3</b> Multiple regression</a>
<ul>
<li class="chapter" data-level="23.3.1" data-path="multiple-regression.html"><a href="multiple-regression.html#geometrical-interpretation"><i class="fa fa-check"></i><b>23.3.1</b> Geometrical interpretation</a></li>
<li class="chapter" data-level="23.3.2" data-path="multiple-regression.html"><a href="multiple-regression.html#coefficient-of-determination-for-multiple-regression"><i class="fa fa-check"></i><b>23.3.2</b> Coefficient of determination for multiple regression</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="multiple-regression.html"><a href="multiple-regression.html#interpretting-multiple-regression"><i class="fa fa-check"></i><b>23.4</b> Interpretting Multiple Regression</a></li>
<li class="chapter" data-level="23.5" data-path="multiple-regression.html"><a href="multiple-regression.html#libaries"><i class="fa fa-check"></i><b>23.5</b> Libaries</a></li>
<li class="chapter" data-level="23.6" data-path="multiple-regression.html"><a href="multiple-regression.html#example-data-set-mtcars"><i class="fa fa-check"></i><b>23.6</b> Example data set: <code>mtcars</code></a></li>
<li class="chapter" data-level="23.7" data-path="multiple-regression.html"><a href="multiple-regression.html#visualizing-and-summarizing-the-variables-of-interest"><i class="fa fa-check"></i><b>23.7</b> Visualizing and summarizing the variables of interest</a></li>
<li class="chapter" data-level="23.8" data-path="multiple-regression.html"><a href="multiple-regression.html#d-plots"><i class="fa fa-check"></i><b>23.8</b> 3D plots</a></li>
<li class="chapter" data-level="23.9" data-path="multiple-regression.html"><a href="multiple-regression.html#fitting-a-multiple-regression-model-in-r"><i class="fa fa-check"></i><b>23.9</b> Fitting a multiple regression model in R</a></li>
<li class="chapter" data-level="23.10" data-path="multiple-regression.html"><a href="multiple-regression.html#visualizing-the-regression-plane"><i class="fa fa-check"></i><b>23.10</b> Visualizing the regression plane</a></li>
<li class="chapter" data-level="23.11" data-path="multiple-regression.html"><a href="multiple-regression.html#interactive-3d-visualizations-using-opengl"><i class="fa fa-check"></i><b>23.11</b> Interactive 3D Visualizations Using OpenGL</a></li>
<li class="chapter" data-level="23.12" data-path="multiple-regression.html"><a href="multiple-regression.html#examining-the-residuals"><i class="fa fa-check"></i><b>23.12</b> Examining the residuals</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="logistic-regression.html"><a href="logistic-regression.html"><i class="fa fa-check"></i><b>24</b> Logistic regression</a>
<ul>
<li class="chapter" data-level="24.1" data-path="logistic-regression.html"><a href="logistic-regression.html#a-web-app-to-explore-the-logistic-regression-equation"><i class="fa fa-check"></i><b>24.1</b> A web app to explore the logistic regression equation</a></li>
<li class="chapter" data-level="24.2" data-path="logistic-regression.html"><a href="logistic-regression.html#titanic-data-set"><i class="fa fa-check"></i><b>24.2</b> Titanic data set</a></li>
<li class="chapter" data-level="24.3" data-path="logistic-regression.html"><a href="logistic-regression.html#subsetting-the-data"><i class="fa fa-check"></i><b>24.3</b> Subsetting the data</a></li>
<li class="chapter" data-level="24.4" data-path="logistic-regression.html"><a href="logistic-regression.html#visualizing-survival-as-a-function-of-age"><i class="fa fa-check"></i><b>24.4</b> Visualizing survival as a function of age</a></li>
<li class="chapter" data-level="24.5" data-path="logistic-regression.html"><a href="logistic-regression.html#fitting-the-logistic-regression-model"><i class="fa fa-check"></i><b>24.5</b> Fitting the logistic regression model</a></li>
<li class="chapter" data-level="24.6" data-path="logistic-regression.html"><a href="logistic-regression.html#visualizing-the-logistic-regression"><i class="fa fa-check"></i><b>24.6</b> Visualizing the logistic regression</a>
<ul>
<li class="chapter" data-level="24.6.1" data-path="logistic-regression.html"><a href="logistic-regression.html#quick-and-easy-visualization"><i class="fa fa-check"></i><b>24.6.1</b> Quick and easy visualization</a></li>
</ul></li>
<li class="chapter" data-level="24.7" data-path="logistic-regression.html"><a href="logistic-regression.html#impact-of-sex-and-passenger-class-on-the-models"><i class="fa fa-check"></i><b>24.7</b> Impact of sex and passenger class on the models</a></li>
<li class="chapter" data-level="24.8" data-path="logistic-regression.html"><a href="logistic-regression.html#fitting-multiple-models-based-on-groupings-use-dplyrdo"><i class="fa fa-check"></i><b>24.8</b> Fitting multiple models based on groupings use <code>dplyr::do</code></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biology 304: Biological Data Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analysis-of-variance" class="section level1 hasAnchor" number="20">
<h1><span class="header-section-number">Chapter 20</span> Analysis of Variance<a href="analysis-of-variance.html#analysis-of-variance" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><span class="math inline">\(t\)</span>-tests are the standard approach for comparing means between two groups. When you want to compare means between more than two groups a technique called “Analysis of Variance” (ANOVA) is used.</p>
<div id="hypotheses-for-anova" class="section level2 hasAnchor" number="20.1">
<h2><span class="header-section-number">20.1</span> Hypotheses for ANOVA<a href="analysis-of-variance.html#hypotheses-for-anova" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When using ANOVA to compare means, the null and alternative hypotheses are:</p>
<ul>
<li><span class="math inline">\(H_0\)</span>: The means of all the groups are equal</li>
<li><span class="math inline">\(H_A\)</span>: At least one of the means is different from the others</li>
</ul>
</div>
<div id="anova-assumptions" class="section level2 hasAnchor" number="20.2">
<h2><span class="header-section-number">20.2</span> ANOVA, assumptions<a href="analysis-of-variance.html#anova-assumptions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>ANOVA assumes:</p>
<ul>
<li>The measurements in every group represent a random sample from the corresponding population</li>
<li>The varaible of interest is normally distributed</li>
<li>The variance is approximately the same in all the groups</li>
</ul>
</div>
<div id="anova-key-idea" class="section level2 hasAnchor" number="20.3">
<h2><span class="header-section-number">20.3</span> ANOVA, key idea<a href="analysis-of-variance.html#anova-key-idea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The key idea behind ANOVA is that:</p>
<ul>
<li>If the observations in each group are drawn from populations with equal means (and variances) then the variation <em>between</em> group means should be similar to the inter-individual variation <em>within groups</em>.</li>
</ul>
</div>
<div id="partioning-of-sum-of-squares" class="section level2 hasAnchor" number="20.4">
<h2><span class="header-section-number">20.4</span> Partioning of sum of squares<a href="analysis-of-variance.html#partioning-of-sum-of-squares" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another way to think about ANOVA is as a “partitioning of variance”. The total variance among all the individuals across groups can be decomposed into:</p>
<ol style="list-style-type: decimal">
<li><p>variance of the group means around the “grand mean”;</p></li>
<li><p>variance of individuals around the group means.</p></li>
</ol>
<p>However, rather than using variance we use sums of square deviations around the respectives means (usually shortened to “sums of squares”).</p>
<p>This decomposition is represented visually in the figure below:</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-452"></span>
<img src="figures/whitlock_15.1-2.png" alt="Whitock and Schluter, Fig 15.1.2 -- Illustrating the partitioning of sum of squares into $MS_{group}$ and $MS_{error}$ components. " width="90%" />
<p class="caption">
Figure 20.1: Whitock and Schluter, Fig 15.1.2 – Illustrating the partitioning of sum of squares into <span class="math inline">\(MS_{group}\)</span> and <span class="math inline">\(MS_{error}\)</span> components.
</p>
</div>
</div>
<div id="mathematical-partitioning-of-sums-of-squares" class="section level2 hasAnchor" number="20.5">
<h2><span class="header-section-number">20.5</span> Mathematical partitioning of sums of squares<a href="analysis-of-variance.html#mathematical-partitioning-of-sums-of-squares" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Variable <span class="math inline">\(X\)</span> with a total sample of <span class="math inline">\(N\)</span> observations, partitioned ito <span class="math inline">\(k\)</span> groups. The sample size of the g-th group is <span class="math inline">\(n_g\)</span>, and thus <span class="math inline">\(N = \sum_{g=1}^{k} n_g\)</span>. Let <span class="math inline">\(\overline{X}\)</span> indicate the grand mean of <span class="math inline">\(X\)</span> and <span class="math inline">\(\overline{X}_g\)</span> indicate the mean of <span class="math inline">\(X\)</span> in the g-th group.</p>
<p><strong>Total sums of squares</strong></p>
<p>We call the sum of the squared deviations around the grand mean the “total sum of sqaures” (<span class="math inline">\(SS_\text{total}\)</span>).</p>
<p><span class="math display">\[
SS_\text{total} = \sum_{i=1}^N (x_i-\overline{X})^2
\]</span></p>
<ul>
<li>The total degrees of freedom is: <span class="math inline">\(df_\text{total} = N - 1\)</span></li>
</ul>
<p><strong>Group sum of squares and group mean square deviation</strong></p>
<p>The sum of squared deviations of the group means around the grand mean is called the “group sum of squares”:
<span class="math display">\[
SS_\text{group} = \sum_{g=1}^kn_g(\overline{X}_g - \overline{X})^2
\]</span></p>
<ul>
<li><p>The degrees of freedom associated with the group sum of squares is: <span class="math inline">\(df_\text{group} = k - 1\)</span></p></li>
<li><p>Define the “group mean squared deviation” as:
<span class="math display">\[
MS_\text{group} = \frac{SS_\text{group}}{k-1}
\]</span></p></li>
</ul>
<p><strong>Error sum of squares and error mean square deviation</strong></p>
<p>The sum of squared deviations of the individual observations about their respective group means is called the “error sum of squares”:
<span class="math display">\[
SS_\text{error} = \sum_{g=1}^k\sum_{i=1}^{n_g} (x_{i,g} - \overline{X}_g)^2
\]</span></p>
<ul>
<li><p>The degrees of freedom associated with the error sum of squares is: <span class="math inline">\(df_\text{error} = N - k\)</span></p></li>
<li><p>Define the “error mean squared deviation” as:
<span class="math display">\[
MS_\text{error} = \frac{SS_\text{error}}{N-k}
\]</span></p></li>
</ul>
<p><strong>Variation explained: <span class="math inline">\(R^2\)</span></strong></p>
<p>We can summarize the contribution of the group differences to the total variation in the data, using the <span class="math inline">\(R^2\)</span> <strong>value</strong>.</p>
<p>To do this we note that the total sum of squares is the sum of the group and error sum of squares:
<span class="math display">\[
SS_\text{total} = SS_\text{group} + SS_\text{error}
\]</span></p>
<p>The <span class="math inline">\(R^2\)</span>-value is thus defined as:
<span class="math display">\[
R^2 = \frac{SS_\text{groups}}{SS_\text{total}}
\]</span></p>
</div>
<div id="anova-test-statistic-and-sampling-distribution" class="section level2 hasAnchor" number="20.6">
<h2><span class="header-section-number">20.6</span> ANOVA test statistic and sampling distribution<a href="analysis-of-variance.html#anova-test-statistic-and-sampling-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>F-statistic</strong></p>
<p>The test statistic used in ANOVA is designated <span class="math inline">\(F\)</span>, and is defined as follows:</p>
<p><span class="math display">\[
F = \frac{\text{group mean square}}{\text{error mean square}} = \frac{\text{MS}_\text{group}}{\text{MS}_\text{error}}
\]</span></p>
<ul>
<li><p>Under the null hypothesis, the between group and within group variances are similar and thus the <span class="math inline">\(F\)</span> statistic should be approximately 1.</p></li>
<li><p>Large values of the <span class="math inline">\(F\)</span>-statistic means that the between group variance exceeds the within group variance, indicating that at least one of the means is different from the others</p></li>
</ul>
<p><strong>The F-distribution</strong></p>
<p>The sampling distribution of the <span class="math inline">\(F\)</span>-statistic is called the <span class="math inline">\(F\)</span>-distribution.</p>
<p>The <span class="math inline">\(F\)</span>-distribution depends on two parameters:</p>
<ol style="list-style-type: decimal">
<li><p>the degrees of freedom associated with the group sum of squares, <span class="math inline">\(df_\text{group} = k - 1\)</span>;</p></li>
<li><p>the degrees of freedom associated with the error sum of squares, <span class="math inline">\(df_\text{error} = N - k\)</span>;</p></li>
</ol>
<p>We designate a particular <span class="math inline">\(F\)</span>-distribution as <span class="math inline">\(F_{k-1,N-k}\)</span>. We will illustrate what an F-distribution looks like for particular parameters below.</p>
</div>
<div id="anova-tables" class="section level2 hasAnchor" number="20.7">
<h2><span class="header-section-number">20.7</span> ANOVA tables<a href="analysis-of-variance.html#anova-tables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The results of an analysis of variance test are often presented in the form of a table organized as follows:</p>
<table>
<colgroup>
<col width="22%" />
<col width="19%" />
<col width="19%" />
<col width="19%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="center"><span class="math inline">\(SS\)</span></th>
<th align="center"><span class="math inline">\(df\)</span></th>
<th align="center"><span class="math inline">\(MS\)</span></th>
<th align="center"><span class="math inline">\(F\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Group</td>
<td align="center"><span class="math inline">\(SS_\text{group}\)</span></td>
<td align="center"><span class="math inline">\(k-1\)</span></td>
<td align="center"><span class="math inline">\(MS_\text{group}\)</span></td>
<td align="center"><span class="math inline">\(MS_\text{group}/MS_\text{error}\)</span></td>
</tr>
<tr class="even">
<td align="left">Error</td>
<td align="center"><span class="math inline">\(SS_\text{error}\)</span></td>
<td align="center"><span class="math inline">\(N-k\)</span></td>
<td align="center"><span class="math inline">\(MS_\text{error}\)</span></td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="left">Total</td>
<td align="center"><span class="math inline">\(SS_\text{total}\)</span></td>
<td align="center"><span class="math inline">\(N-1\)</span></td>
<td align="center"></td>
<td align="center"></td>
</tr>
</tbody>
</table>
</div>
<div id="the-aov-function" class="section level2 hasAnchor" number="20.8">
<h2><span class="header-section-number">20.8</span> The <code>aov()</code> function<a href="analysis-of-variance.html#the-aov-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As you would suspect, there is a built in R function to carry out ANOVA. This function is designated <code>aov()</code>. <code>aov</code> takes a formula style argument where the variable of interest is on the left, and the grouping variable indicated on the right.</p>
<pre><code>aov(variable.of.interest ~ grouping.variable, data = df)</code></pre>
</div>
<div id="example-circadian-rythm-data" class="section level2 hasAnchor" number="20.9">
<h2><span class="header-section-number">20.9</span> Example, circadian rythm data<a href="analysis-of-variance.html#example-circadian-rythm-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Your textbook describes an examplar data set from a study designed to test the effects of light treatment on circadian rhythms (see Whitlock &amp; Schluter, Example 15.1).</p>
<ul>
<li>The investigators randomly assigned 22 individuals to one of three treatment groups and measured phase shifts in melatonin production. The treatment groups were:
<ul>
<li>control group (8 indiviuals)</li>
<li>light applied on the back of the knee (7 individuals)</li>
<li>light applied to the eyes (7 individuals)</li>
</ul></li>
</ul>
<p>These data are available at: <a href="https://raw.githubusercontent.com/bio304-class/bio304-course-notes/master/datasets/ABD-circadian-rythms.csv">ABD-circadian-rythms.csv</a></p>
<p>The null and alternative hypotheses associated with the ANOVA of these data are:</p>
<ul>
<li><span class="math inline">\(H_0\)</span>: the means of the treatments groups are the same</li>
<li><span class="math inline">\(H_1\)</span>: the mean of at least one of the treatment groups is different from the others</li>
</ul>
<p><strong>Libraries</strong></p>
<div class="sourceCode" id="cb436"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb436-1"><a href="analysis-of-variance.html#cb436-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb436-2"><a href="analysis-of-variance.html#cb436-2" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb436-3"><a href="analysis-of-variance.html#cb436-3" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb436-4"><a href="analysis-of-variance.html#cb436-4" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb436-5"><a href="analysis-of-variance.html#cb436-5" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">20180113</span>)</span></code></pre></div>
<p><strong>Load the data</strong></p>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="analysis-of-variance.html#cb437-1" tabindex="-1"></a>circadian <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;https://raw.githubusercontent.com/bio304-class/bio304-course-notes/master/datasets/ABD-circadian-rythms.csv&quot;</span>)</span></code></pre></div>
<p>The data is very simple: just two columns indicating treatment group and the variable of interest:</p>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="analysis-of-variance.html#cb438-1" tabindex="-1"></a><span class="fu">head</span>(circadian, <span class="at">n=</span><span class="dv">3</span>)</span>
<span id="cb438-2"><a href="analysis-of-variance.html#cb438-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 2</span></span>
<span id="cb438-3"><a href="analysis-of-variance.html#cb438-3" tabindex="-1"></a><span class="co">#&gt;   treatment shift</span></span>
<span id="cb438-4"><a href="analysis-of-variance.html#cb438-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;dbl&gt;</span></span>
<span id="cb438-5"><a href="analysis-of-variance.html#cb438-5" tabindex="-1"></a><span class="co">#&gt; 1 control    0.53</span></span>
<span id="cb438-6"><a href="analysis-of-variance.html#cb438-6" tabindex="-1"></a><span class="co">#&gt; 2 control    0.36</span></span>
<span id="cb438-7"><a href="analysis-of-variance.html#cb438-7" tabindex="-1"></a><span class="co">#&gt; 3 control    0.2</span></span></code></pre></div>
<div id="visualizing-the-data" class="section level3 hasAnchor" number="20.9.1">
<h3><span class="header-section-number">20.9.1</span> Visualizing the data<a href="analysis-of-variance.html#visualizing-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As is our standard practice, let’s start by visualizing the data. We’ll create a point plot depicting the observations colored by treatment group.</p>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="analysis-of-variance.html#cb439-1" tabindex="-1"></a><span class="co"># we&#39;re going to re-use our jittering across plots so</span></span>
<span id="cb439-2"><a href="analysis-of-variance.html#cb439-2" tabindex="-1"></a><span class="co"># assign it to a variable</span></span>
<span id="cb439-3"><a href="analysis-of-variance.html#cb439-3" tabindex="-1"></a>pd <span class="ot">&lt;-</span> <span class="fu">position_jitter</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">height=</span><span class="dv">0</span>)</span>
<span id="cb439-4"><a href="analysis-of-variance.html#cb439-4" tabindex="-1"></a></span>
<span id="cb439-5"><a href="analysis-of-variance.html#cb439-5" tabindex="-1"></a>point.plot <span class="ot">&lt;-</span></span>
<span id="cb439-6"><a href="analysis-of-variance.html#cb439-6" tabindex="-1"></a>  circadian <span class="sc">%&gt;%</span></span>
<span id="cb439-7"><a href="analysis-of-variance.html#cb439-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>treatment, <span class="at">y=</span>shift, </span>
<span id="cb439-8"><a href="analysis-of-variance.html#cb439-8" tabindex="-1"></a>             <span class="at">color=</span>treatment, <span class="at">group=</span><span class="fu">row.names</span>(circadian))) <span class="sc">+</span></span>
<span id="cb439-9"><a href="analysis-of-variance.html#cb439-9" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb439-10"><a href="analysis-of-variance.html#cb439-10" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">1</span>)<span class="sc">+</span> </span>
<span id="cb439-11"><a href="analysis-of-variance.html#cb439-11" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Treatment&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Phase shift (h)&quot;</span>)</span>
<span id="cb439-12"><a href="analysis-of-variance.html#cb439-12" tabindex="-1"></a></span>
<span id="cb439-13"><a href="analysis-of-variance.html#cb439-13" tabindex="-1"></a>point.plot</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-456-1.png" width="480" /></p>
</div>
<div id="carrying-out-the-anova" class="section level3 hasAnchor" number="20.9.2">
<h3><span class="header-section-number">20.9.2</span> Carrying out the ANOVA<a href="analysis-of-variance.html#carrying-out-the-anova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>From our visualization it certainly seems like there may be differences among the group (treatment) means. Let’s test this formally using the <code>aov()</code> function:</p>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="analysis-of-variance.html#cb440-1" tabindex="-1"></a>circadian.aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(shift <span class="sc">~</span> treatment, <span class="at">data =</span> circadian)</span></code></pre></div>
<p>The <code>summary</code> function applied to the <code>aov</code> fit will print out a typical ANOVA table and calculate the associated P-value for the <span class="math inline">\(F\)</span> test statistic:</p>
<div class="sourceCode" id="cb441"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb441-1"><a href="analysis-of-variance.html#cb441-1" tabindex="-1"></a><span class="fu">summary</span>(circadian.aov)</span>
<span id="cb441-2"><a href="analysis-of-variance.html#cb441-2" tabindex="-1"></a><span class="co">#&gt;             Df Sum Sq Mean Sq F value  Pr(&gt;F)   </span></span>
<span id="cb441-3"><a href="analysis-of-variance.html#cb441-3" tabindex="-1"></a><span class="co">#&gt; treatment    2  7.224   3.612   7.289 0.00447 **</span></span>
<span id="cb441-4"><a href="analysis-of-variance.html#cb441-4" tabindex="-1"></a><span class="co">#&gt; Residuals   19  9.415   0.496                   </span></span>
<span id="cb441-5"><a href="analysis-of-variance.html#cb441-5" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb441-6"><a href="analysis-of-variance.html#cb441-6" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span></code></pre></div>
<p>If you want the ANOVA table in a form you can compute with, the <code>broom::tidy</code> function we explored previously comes in handy:</p>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="analysis-of-variance.html#cb442-1" tabindex="-1"></a>circadian.aov.table <span class="ot">&lt;-</span> <span class="fu">tidy</span>(circadian.aov)</span>
<span id="cb442-2"><a href="analysis-of-variance.html#cb442-2" tabindex="-1"></a>circadian.aov.table</span>
<span id="cb442-3"><a href="analysis-of-variance.html#cb442-3" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2 × 6</span></span>
<span id="cb442-4"><a href="analysis-of-variance.html#cb442-4" tabindex="-1"></a><span class="co">#&gt;   term         df sumsq meansq statistic   p.value</span></span>
<span id="cb442-5"><a href="analysis-of-variance.html#cb442-5" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span></span>
<span id="cb442-6"><a href="analysis-of-variance.html#cb442-6" tabindex="-1"></a><span class="co">#&gt; 1 treatment     2 7.224 3.612      7.289  0.004472</span></span>
<span id="cb442-7"><a href="analysis-of-variance.html#cb442-7" tabindex="-1"></a><span class="co">#&gt; 2 Residuals    19 9.415 0.4955    NA     NA</span></span></code></pre></div>
<p>The table above tells us that the <span class="math inline">\(F\)</span>-statistic for this ANOVA is ~7.29. The table also tells us that the P-value associated with this F-statistic, is quite small, P-value &lt; 0.005.</p>
<p>The P-value can be calculated explicitly using the <code>pf()</code> function (similar to the <code>pnorm()</code> and <code>pt()</code> functions we’ve seen previously):</p>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="analysis-of-variance.html#cb443-1" tabindex="-1"></a>circadian.F.stat <span class="ot">&lt;-</span> circadian.aov.table<span class="sc">$</span>statistic[<span class="dv">1</span>]</span>
<span id="cb443-2"><a href="analysis-of-variance.html#cb443-2" tabindex="-1"></a><span class="fu">pf</span>(circadian.F.stat, <span class="at">df1 =</span> <span class="dv">2</span>, <span class="at">df2 =</span> <span class="dv">19</span>, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb443-3"><a href="analysis-of-variance.html#cb443-3" tabindex="-1"></a><span class="co">#&gt; [1] 0.004472271</span></span></code></pre></div>
<p>Note that we set <code>lower.tail = FALSE</code> to calculate the probability of getting an F-statistic this large or greater.</p>
<p><strong>Visualizing the F-distribution</strong></p>
<p>Let’s draw the corresponding F-distribution, <span class="math inline">\(F_{2,19}\)</span>, using the <code>df()</code> function (parallel to <code>dnorm()</code> and <code>dt()</code>, with the region corresponding to an F-statistic greater than 7.29 shaded red. Because the area in the right tail we’re interested in is quite small, in a second plot we’ve zoomed in on this region.</p>
<div class="sourceCode" id="cb444"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb444-1"><a href="analysis-of-variance.html#cb444-1" tabindex="-1"></a>fdist <span class="ot">&lt;-</span> <span class="fu">data_frame</span>(<span class="at">f =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="at">length.out =</span> <span class="dv">250</span>),</span>
<span id="cb444-2"><a href="analysis-of-variance.html#cb444-2" tabindex="-1"></a>                    <span class="at">density =</span> <span class="fu">df</span>(f, <span class="at">df1=</span><span class="dv">2</span>, <span class="at">df2=</span><span class="dv">19</span>))</span>
<span id="cb444-3"><a href="analysis-of-variance.html#cb444-3" tabindex="-1"></a></span>
<span id="cb444-4"><a href="analysis-of-variance.html#cb444-4" tabindex="-1"></a>plot.a <span class="ot">&lt;-</span> </span>
<span id="cb444-5"><a href="analysis-of-variance.html#cb444-5" tabindex="-1"></a>  <span class="fu">ggplot</span>(fdist, <span class="fu">aes</span>(f, density)) <span class="sc">+</span> </span>
<span id="cb444-6"><a href="analysis-of-variance.html#cb444-6" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span> </span>
<span id="cb444-7"><a href="analysis-of-variance.html#cb444-7" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="at">data =</span> <span class="fu">filter</span>(fdist, f <span class="sc">&gt;=</span> <span class="fl">7.29</span>), <span class="at">fill=</span><span class="st">&#39;red&#39;</span>, <span class="at">alpha=</span><span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb444-8"><a href="analysis-of-variance.html#cb444-8" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;F&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Density&quot;</span>, <span class="at">title=</span><span class="st">&quot;F(2,19)&quot;</span>)</span>
<span id="cb444-9"><a href="analysis-of-variance.html#cb444-9" tabindex="-1"></a></span>
<span id="cb444-10"><a href="analysis-of-variance.html#cb444-10" tabindex="-1"></a>plot.b <span class="ot">&lt;-</span> </span>
<span id="cb444-11"><a href="analysis-of-variance.html#cb444-11" tabindex="-1"></a>  plot.a <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">5</span>,<span class="dv">10</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="fl">0.02</span>) <span class="sc">+</span> </span>
<span id="cb444-12"><a href="analysis-of-variance.html#cb444-12" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Zoomed in view of right tail of F(2,19)&quot;</span>)</span>
<span id="cb444-13"><a href="analysis-of-variance.html#cb444-13" tabindex="-1"></a></span>
<span id="cb444-14"><a href="analysis-of-variance.html#cb444-14" tabindex="-1"></a><span class="fu">plot_grid</span>(plot.a, plot.b, <span class="at">labels=</span><span class="st">&quot;AUTO&quot;</span>)</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-461-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p><strong>Critical values of the F-distribution</strong></p>
<p>If we wanted to know what the critical F value is for a corresponding type I error rate we can use the <code>qf()</code> function:</p>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="analysis-of-variance.html#cb445-1" tabindex="-1"></a><span class="co"># the critical value of F for alpha = 0.05</span></span>
<span id="cb445-2"><a href="analysis-of-variance.html#cb445-2" tabindex="-1"></a><span class="fu">qf</span>(<span class="fl">0.05</span>, <span class="dv">2</span>, <span class="dv">19</span>, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb445-3"><a href="analysis-of-variance.html#cb445-3" tabindex="-1"></a><span class="co">#&gt; [1] 3.521893</span></span></code></pre></div>
</div>
</div>
<div id="anova-calculations-step-by-step" class="section level2 hasAnchor" number="20.10">
<h2><span class="header-section-number">20.10</span> ANOVA calculations: Step-by-step<a href="analysis-of-variance.html#anova-calculations-step-by-step" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>aov()</code> function carries out all the ANOVA calculations behind the scenes. It’s useful to pull back the curtain and see how the various quantities are calculated.</p>
<p><strong>Total SS</strong></p>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="analysis-of-variance.html#cb446-1" tabindex="-1"></a>grand.mean <span class="ot">&lt;-</span> <span class="fu">mean</span>(circadian<span class="sc">$</span>shift)</span>
<span id="cb446-2"><a href="analysis-of-variance.html#cb446-2" tabindex="-1"></a></span>
<span id="cb446-3"><a href="analysis-of-variance.html#cb446-3" tabindex="-1"></a><span class="co"># total sum of squares</span></span>
<span id="cb446-4"><a href="analysis-of-variance.html#cb446-4" tabindex="-1"></a>total.table <span class="ot">&lt;-</span> </span>
<span id="cb446-5"><a href="analysis-of-variance.html#cb446-5" tabindex="-1"></a>  circadian <span class="sc">%&gt;%</span></span>
<span id="cb446-6"><a href="analysis-of-variance.html#cb446-6" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SS =</span> <span class="fu">sum</span>((shift <span class="sc">-</span> grand.mean)<span class="sc">**</span><span class="dv">2</span>),</span>
<span id="cb446-7"><a href="analysis-of-variance.html#cb446-7" tabindex="-1"></a>            <span class="at">df =</span> <span class="fu">n</span>() <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb446-8"><a href="analysis-of-variance.html#cb446-8" tabindex="-1"></a></span>
<span id="cb446-9"><a href="analysis-of-variance.html#cb446-9" tabindex="-1"></a>total.table</span>
<span id="cb446-10"><a href="analysis-of-variance.html#cb446-10" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 2</span></span>
<span id="cb446-11"><a href="analysis-of-variance.html#cb446-11" tabindex="-1"></a><span class="co">#&gt;      SS    df</span></span>
<span id="cb446-12"><a href="analysis-of-variance.html#cb446-12" tabindex="-1"></a><span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt;</span></span>
<span id="cb446-13"><a href="analysis-of-variance.html#cb446-13" tabindex="-1"></a><span class="co">#&gt; 1 16.64    21</span></span></code></pre></div>
<p><strong>Group SS and MS</strong></p>
<p>We use <code>group_by</code> and <code>summarize</code> to calculates group means and the group deviates (the difference between the group means and the grand mean):</p>
<div class="sourceCode" id="cb447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb447-1"><a href="analysis-of-variance.html#cb447-1" tabindex="-1"></a>group.df <span class="ot">&lt;-</span></span>
<span id="cb447-2"><a href="analysis-of-variance.html#cb447-2" tabindex="-1"></a>  circadian <span class="sc">%&gt;%</span> </span>
<span id="cb447-3"><a href="analysis-of-variance.html#cb447-3" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb447-4"><a href="analysis-of-variance.html#cb447-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">n</span>(), </span>
<span id="cb447-5"><a href="analysis-of-variance.html#cb447-5" tabindex="-1"></a>            <span class="at">group.mean =</span> <span class="fu">mean</span>(shift), </span>
<span id="cb447-6"><a href="analysis-of-variance.html#cb447-6" tabindex="-1"></a>            <span class="at">grand.mean =</span> grand.mean,</span>
<span id="cb447-7"><a href="analysis-of-variance.html#cb447-7" tabindex="-1"></a>            <span class="at">group.deviates =</span> group.mean <span class="sc">-</span> grand.mean)</span>
<span id="cb447-8"><a href="analysis-of-variance.html#cb447-8" tabindex="-1"></a></span>
<span id="cb447-9"><a href="analysis-of-variance.html#cb447-9" tabindex="-1"></a>group.df</span>
<span id="cb447-10"><a href="analysis-of-variance.html#cb447-10" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 5</span></span>
<span id="cb447-11"><a href="analysis-of-variance.html#cb447-11" tabindex="-1"></a><span class="co">#&gt;   treatment     n group.mean grand.mean group.deviates</span></span>
<span id="cb447-12"><a href="analysis-of-variance.html#cb447-12" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;int&gt;      &lt;dbl&gt;      &lt;dbl&gt;          &lt;dbl&gt;</span></span>
<span id="cb447-13"><a href="analysis-of-variance.html#cb447-13" tabindex="-1"></a><span class="co">#&gt; 1 control       8    -0.3088    -0.7127         0.4040</span></span>
<span id="cb447-14"><a href="analysis-of-variance.html#cb447-14" tabindex="-1"></a><span class="co">#&gt; 2 eyes          7    -1.551     -0.7127        -0.8387</span></span>
<span id="cb447-15"><a href="analysis-of-variance.html#cb447-15" tabindex="-1"></a><span class="co">#&gt; 3 knee          7    -0.3357    -0.7127         0.3770</span></span></code></pre></div>
<p>Having calculated the group deviates, we calculate the group sum of squares and related quantities:</p>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="analysis-of-variance.html#cb448-1" tabindex="-1"></a>group.table <span class="ot">&lt;-</span></span>
<span id="cb448-2"><a href="analysis-of-variance.html#cb448-2" tabindex="-1"></a>  group.df <span class="sc">%&gt;%</span></span>
<span id="cb448-3"><a href="analysis-of-variance.html#cb448-3" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SS =</span> <span class="fu">sum</span>(n <span class="sc">*</span> group.deviates<span class="sc">**</span><span class="dv">2</span>),</span>
<span id="cb448-4"><a href="analysis-of-variance.html#cb448-4" tabindex="-1"></a>            <span class="at">k =</span> <span class="fu">n</span>(),</span>
<span id="cb448-5"><a href="analysis-of-variance.html#cb448-5" tabindex="-1"></a>            <span class="at">df =</span> k<span class="dv">-1</span>,</span>
<span id="cb448-6"><a href="analysis-of-variance.html#cb448-6" tabindex="-1"></a>            <span class="at">MS =</span> SS<span class="sc">/</span>df)</span>
<span id="cb448-7"><a href="analysis-of-variance.html#cb448-7" tabindex="-1"></a></span>
<span id="cb448-8"><a href="analysis-of-variance.html#cb448-8" tabindex="-1"></a>group.table</span>
<span id="cb448-9"><a href="analysis-of-variance.html#cb448-9" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 4</span></span>
<span id="cb448-10"><a href="analysis-of-variance.html#cb448-10" tabindex="-1"></a><span class="co">#&gt;      SS     k    df    MS</span></span>
<span id="cb448-11"><a href="analysis-of-variance.html#cb448-11" tabindex="-1"></a><span class="co">#&gt;   &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt;</span></span>
<span id="cb448-12"><a href="analysis-of-variance.html#cb448-12" tabindex="-1"></a><span class="co">#&gt; 1 7.224     3     2 3.612</span></span></code></pre></div>
<p><strong>Error SS and MS</strong></p>
<p>Next we turn to variation of the individual observations around the group means, which is the basis of the error sum of squares and mean square. Again we calculate this in two steps:</p>
<div class="sourceCode" id="cb449"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb449-1"><a href="analysis-of-variance.html#cb449-1" tabindex="-1"></a>error.df <span class="ot">&lt;-</span></span>
<span id="cb449-2"><a href="analysis-of-variance.html#cb449-2" tabindex="-1"></a>  circadian <span class="sc">%&gt;%</span> </span>
<span id="cb449-3"><a href="analysis-of-variance.html#cb449-3" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb449-4"><a href="analysis-of-variance.html#cb449-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group.mean =</span> <span class="fu">mean</span>(shift),</span>
<span id="cb449-5"><a href="analysis-of-variance.html#cb449-5" tabindex="-1"></a>         <span class="at">error.deviates =</span> shift <span class="sc">-</span> group.mean) <span class="sc">%&gt;%</span></span>
<span id="cb449-6"><a href="analysis-of-variance.html#cb449-6" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SS =</span> <span class="fu">sum</span>(error.deviates<span class="sc">**</span><span class="dv">2</span>),</span>
<span id="cb449-7"><a href="analysis-of-variance.html#cb449-7" tabindex="-1"></a>            <span class="at">n =</span> <span class="fu">n</span>())</span>
<span id="cb449-8"><a href="analysis-of-variance.html#cb449-8" tabindex="-1"></a></span>
<span id="cb449-9"><a href="analysis-of-variance.html#cb449-9" tabindex="-1"></a>error.df</span>
<span id="cb449-10"><a href="analysis-of-variance.html#cb449-10" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 3</span></span>
<span id="cb449-11"><a href="analysis-of-variance.html#cb449-11" tabindex="-1"></a><span class="co">#&gt;   treatment    SS     n</span></span>
<span id="cb449-12"><a href="analysis-of-variance.html#cb449-12" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;int&gt;</span></span>
<span id="cb449-13"><a href="analysis-of-variance.html#cb449-13" tabindex="-1"></a><span class="co">#&gt; 1 control   2.670     8</span></span>
<span id="cb449-14"><a href="analysis-of-variance.html#cb449-14" tabindex="-1"></a><span class="co">#&gt; 2 eyes      2.993     7</span></span>
<span id="cb449-15"><a href="analysis-of-variance.html#cb449-15" tabindex="-1"></a><span class="co">#&gt; 3 knee      3.752     7</span></span></code></pre></div>
<p>Now we calculate the error sum of squares and related quantities:</p>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="analysis-of-variance.html#cb450-1" tabindex="-1"></a>error.table <span class="ot">&lt;-</span></span>
<span id="cb450-2"><a href="analysis-of-variance.html#cb450-2" tabindex="-1"></a>  error.df <span class="sc">%&gt;%</span></span>
<span id="cb450-3"><a href="analysis-of-variance.html#cb450-3" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SS =</span> <span class="fu">sum</span>(SS),</span>
<span id="cb450-4"><a href="analysis-of-variance.html#cb450-4" tabindex="-1"></a>            <span class="at">k =</span> <span class="fu">n</span>(),</span>
<span id="cb450-5"><a href="analysis-of-variance.html#cb450-5" tabindex="-1"></a>            <span class="at">N =</span> <span class="fu">sum</span>(n),</span>
<span id="cb450-6"><a href="analysis-of-variance.html#cb450-6" tabindex="-1"></a>            <span class="at">df =</span> N <span class="sc">-</span> k,</span>
<span id="cb450-7"><a href="analysis-of-variance.html#cb450-7" tabindex="-1"></a>            <span class="at">MS =</span> SS<span class="sc">/</span>df)</span>
<span id="cb450-8"><a href="analysis-of-variance.html#cb450-8" tabindex="-1"></a></span>
<span id="cb450-9"><a href="analysis-of-variance.html#cb450-9" tabindex="-1"></a>error.table</span>
<span id="cb450-10"><a href="analysis-of-variance.html#cb450-10" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 5</span></span>
<span id="cb450-11"><a href="analysis-of-variance.html#cb450-11" tabindex="-1"></a><span class="co">#&gt;      SS     k     N    df     MS</span></span>
<span id="cb450-12"><a href="analysis-of-variance.html#cb450-12" tabindex="-1"></a><span class="co">#&gt;   &lt;dbl&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;  &lt;dbl&gt;</span></span>
<span id="cb450-13"><a href="analysis-of-variance.html#cb450-13" tabindex="-1"></a><span class="co">#&gt; 1 9.415     3    22    19 0.4955</span></span></code></pre></div>
<p><strong>Calculating the F-statistic and <span class="math inline">\(R^2\)</span></strong></p>
<p>Having calculated our estimates of between group variance and within group variance (<span class="math inline">\(MS_\text{group}\)</span> and $MS_) we’re now ready to calculate the <span class="math inline">\(F\)</span> test statistic.</p>
<div class="sourceCode" id="cb451"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb451-1"><a href="analysis-of-variance.html#cb451-1" tabindex="-1"></a>F.stat <span class="ot">&lt;-</span> group.table<span class="sc">$</span>MS<span class="sc">/</span>error.table<span class="sc">$</span>MS</span>
<span id="cb451-2"><a href="analysis-of-variance.html#cb451-2" tabindex="-1"></a>F.stat</span>
<span id="cb451-3"><a href="analysis-of-variance.html#cb451-3" tabindex="-1"></a><span class="co">#&gt; [1] 7.289449</span></span></code></pre></div>
<p>The variation “explained” by the group is:</p>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="analysis-of-variance.html#cb452-1" tabindex="-1"></a>R2 <span class="ot">&lt;-</span> group.table<span class="sc">$</span>SS<span class="sc">/</span>total.table<span class="sc">$</span>SS</span>
<span id="cb452-2"><a href="analysis-of-variance.html#cb452-2" tabindex="-1"></a>R2</span>
<span id="cb452-3"><a href="analysis-of-variance.html#cb452-3" tabindex="-1"></a><span class="co">#&gt; [1] 0.4341684</span></span></code></pre></div>
<p>Thus about 43% of the total sum of squared deviation among subjects, with respect to the phase shift variable, is explained by differences in light treatment.</p>
</div>
<div id="visualizing-the-partitioning-of-sum-of-squares" class="section level2 hasAnchor" number="20.11">
<h2><span class="header-section-number">20.11</span> Visualizing the partitioning of sum-of-squares<a href="analysis-of-variance.html#visualizing-the-partitioning-of-sum-of-squares" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Total SS</strong></p>
<div class="sourceCode" id="cb453"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb453-1"><a href="analysis-of-variance.html#cb453-1" tabindex="-1"></a>total.plot <span class="ot">&lt;-</span></span>
<span id="cb453-2"><a href="analysis-of-variance.html#cb453-2" tabindex="-1"></a>  circadian <span class="sc">%&gt;%</span></span>
<span id="cb453-3"><a href="analysis-of-variance.html#cb453-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>treatment, <span class="at">y=</span>shift, </span>
<span id="cb453-4"><a href="analysis-of-variance.html#cb453-4" tabindex="-1"></a>             <span class="at">color=</span>treatment, <span class="at">group=</span><span class="fu">row.names</span>(circadian))) <span class="sc">+</span></span>
<span id="cb453-5"><a href="analysis-of-variance.html#cb453-5" tabindex="-1"></a>    <span class="fu">geom_linerange</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> grand.mean, <span class="at">ymax =</span> shift), <span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb453-6"><a href="analysis-of-variance.html#cb453-6" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> grand.mean, <span class="at">linetype=</span><span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span> </span>
<span id="cb453-7"><a href="analysis-of-variance.html#cb453-7" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">1</span>)<span class="sc">+</span> </span>
<span id="cb453-8"><a href="analysis-of-variance.html#cb453-8" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Treatment&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Phase shift (h)&quot;</span>,</span>
<span id="cb453-9"><a href="analysis-of-variance.html#cb453-9" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">&quot;Deviation of observations around the grand mean (dashed line)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb453-10"><a href="analysis-of-variance.html#cb453-10" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>))</span>
<span id="cb453-11"><a href="analysis-of-variance.html#cb453-11" tabindex="-1"></a></span>
<span id="cb453-12"><a href="analysis-of-variance.html#cb453-12" tabindex="-1"></a>total.plot</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-470-1.png" width="576" /></p>
<p><strong>Group SS</strong></p>
<p>Let’s visualize the difference of the group means from the grand mean:</p>
<div class="sourceCode" id="cb454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb454-1"><a href="analysis-of-variance.html#cb454-1" tabindex="-1"></a>group.plot <span class="ot">&lt;-</span></span>
<span id="cb454-2"><a href="analysis-of-variance.html#cb454-2" tabindex="-1"></a>  group.df <span class="sc">%&gt;%</span></span>
<span id="cb454-3"><a href="analysis-of-variance.html#cb454-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> group.mean, <span class="at">color=</span>treatment)) <span class="sc">+</span> </span>
<span id="cb454-4"><a href="analysis-of-variance.html#cb454-4" tabindex="-1"></a>    <span class="fu">geom_linerange</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> grand.mean, <span class="at">ymax =</span> group.mean), <span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb454-5"><a href="analysis-of-variance.html#cb454-5" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb454-6"><a href="analysis-of-variance.html#cb454-6" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> grand.mean, <span class="at">linetype=</span><span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span> </span>
<span id="cb454-7"><a href="analysis-of-variance.html#cb454-7" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb454-8"><a href="analysis-of-variance.html#cb454-8" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Treatment&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Phase shift (h)&quot;</span>,</span>
<span id="cb454-9"><a href="analysis-of-variance.html#cb454-9" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">&quot;Deviation of group means around the grand mean&quot;</span>) <span class="sc">+</span> </span>
<span id="cb454-10"><a href="analysis-of-variance.html#cb454-10" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>))</span>
<span id="cb454-11"><a href="analysis-of-variance.html#cb454-11" tabindex="-1"></a></span>
<span id="cb454-12"><a href="analysis-of-variance.html#cb454-12" tabindex="-1"></a>group.plot</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-471-1.png" width="576" /></p>
<p><strong>Error SS</strong></p>
<p>We can visualize the individual deviates around the group means as so:</p>
<div class="sourceCode" id="cb455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb455-1"><a href="analysis-of-variance.html#cb455-1" tabindex="-1"></a>error.plot <span class="ot">&lt;-</span></span>
<span id="cb455-2"><a href="analysis-of-variance.html#cb455-2" tabindex="-1"></a>  circadian <span class="sc">%&gt;%</span> </span>
<span id="cb455-3"><a href="analysis-of-variance.html#cb455-3" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb455-4"><a href="analysis-of-variance.html#cb455-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group.mean =</span> <span class="fu">mean</span>(shift)) <span class="sc">%&gt;%</span></span>
<span id="cb455-5"><a href="analysis-of-variance.html#cb455-5" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> shift, <span class="at">color =</span> treatment)) <span class="sc">+</span> </span>
<span id="cb455-6"><a href="analysis-of-variance.html#cb455-6" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> group.mean),<span class="at">size=</span><span class="dv">3</span>,<span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb455-7"><a href="analysis-of-variance.html#cb455-7" tabindex="-1"></a>    <span class="fu">geom_linerange</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> group.mean, <span class="at">ymax =</span> shift), <span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb455-8"><a href="analysis-of-variance.html#cb455-8" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb455-9"><a href="analysis-of-variance.html#cb455-9" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Treatment&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Phase shift (h)&quot;</span>,</span>
<span id="cb455-10"><a href="analysis-of-variance.html#cb455-10" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">&quot;Deviation of observations around the groups means&quot;</span>) <span class="sc">+</span> </span>
<span id="cb455-11"><a href="analysis-of-variance.html#cb455-11" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>))</span>
<span id="cb455-12"><a href="analysis-of-variance.html#cb455-12" tabindex="-1"></a></span>
<span id="cb455-13"><a href="analysis-of-variance.html#cb455-13" tabindex="-1"></a>error.plot</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-472-1.png" width="576" /></p>
<p><strong>Combined visualization</strong></p>
<p>We can combine our three plots created above into a single figure using <code>cowplot::plot_grid</code>:</p>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="analysis-of-variance.html#cb456-1" tabindex="-1"></a>combined.plot <span class="ot">&lt;-</span><span class="fu">plot_grid</span>(total.plot, group.plot, error.plot, </span>
<span id="cb456-2"><a href="analysis-of-variance.html#cb456-2" tabindex="-1"></a>                          <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>,<span class="st">&quot;C&quot;</span>), <span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb456-3"><a href="analysis-of-variance.html#cb456-3" tabindex="-1"></a>combined.plot</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-473-1.png" width="1440" /></p>
</div>
<div id="which-pairs-of-group-means-are-different" class="section level2 hasAnchor" number="20.12">
<h2><span class="header-section-number">20.12</span> Which pairs of group means are different?<a href="analysis-of-variance.html#which-pairs-of-group-means-are-different" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If an ANOVA indicates that at least one of the group means is different than the others, the next question is usually “which pairs are different?”. There are slighly different tests for what are called “planned” versus “unplanned” comparisons. Your textbook discusses the differences between these two types</p>
<p>Here we focus on a common test for unplanned comparisons, called the Tukey Honest Significant Differences test (referred to as the Tukey-Kramer test in your textbook). The Tukey HSD test controls for the “family-wise error rate”, meaning it tries to keep the overall false positive (Type I error) rate at a specified value.</p>
<div id="tukey-kramer-test" class="section level3 hasAnchor" number="20.12.1">
<h3><span class="header-section-number">20.12.1</span> Tukey-Kramer test<a href="analysis-of-variance.html#tukey-kramer-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The function <code>TukeyHSD</code> implements the Tukey-Kramer test. The input to <code>TukeyHSD</code> is the fit from <code>aov</code>:</p>
<div class="sourceCode" id="cb457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb457-1"><a href="analysis-of-variance.html#cb457-1" tabindex="-1"></a><span class="fu">TukeyHSD</span>(circadian.aov)</span>
<span id="cb457-2"><a href="analysis-of-variance.html#cb457-2" tabindex="-1"></a><span class="co">#&gt;   Tukey multiple comparisons of means</span></span>
<span id="cb457-3"><a href="analysis-of-variance.html#cb457-3" tabindex="-1"></a><span class="co">#&gt;     95% family-wise confidence level</span></span>
<span id="cb457-4"><a href="analysis-of-variance.html#cb457-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb457-5"><a href="analysis-of-variance.html#cb457-5" tabindex="-1"></a><span class="co">#&gt; Fit: aov(formula = shift ~ treatment, data = circadian)</span></span>
<span id="cb457-6"><a href="analysis-of-variance.html#cb457-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb457-7"><a href="analysis-of-variance.html#cb457-7" tabindex="-1"></a><span class="co">#&gt; $treatment</span></span>
<span id="cb457-8"><a href="analysis-of-variance.html#cb457-8" tabindex="-1"></a><span class="co">#&gt;                     diff        lwr        upr     p adj</span></span>
<span id="cb457-9"><a href="analysis-of-variance.html#cb457-9" tabindex="-1"></a><span class="co">#&gt; eyes-control -1.24267857 -2.1682364 -0.3171207 0.0078656</span></span>
<span id="cb457-10"><a href="analysis-of-variance.html#cb457-10" tabindex="-1"></a><span class="co">#&gt; knee-control -0.02696429 -0.9525222  0.8985936 0.9969851</span></span>
<span id="cb457-11"><a href="analysis-of-variance.html#cb457-11" tabindex="-1"></a><span class="co">#&gt; knee-eyes     1.21571429  0.2598022  2.1716263 0.0116776</span></span></code></pre></div>
<p>Here again, the <code>broom::tidy</code> function comes in handy:</p>
<div class="sourceCode" id="cb458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb458-1"><a href="analysis-of-variance.html#cb458-1" tabindex="-1"></a><span class="fu">tidy</span>(<span class="fu">TukeyHSD</span>(circadian.aov))</span>
<span id="cb458-2"><a href="analysis-of-variance.html#cb458-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 7</span></span>
<span id="cb458-3"><a href="analysis-of-variance.html#cb458-3" tabindex="-1"></a><span class="co">#&gt;   term      contrast     null.value estimate conf.low conf.high adj.p.value</span></span>
<span id="cb458-4"><a href="analysis-of-variance.html#cb458-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;             &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;</span></span>
<span id="cb458-5"><a href="analysis-of-variance.html#cb458-5" tabindex="-1"></a><span class="co">#&gt; 1 treatment eyes-control          0 -1.243    -2.168    -0.3171    0.007866</span></span>
<span id="cb458-6"><a href="analysis-of-variance.html#cb458-6" tabindex="-1"></a><span class="co">#&gt; 2 treatment knee-control          0 -0.02696  -0.9525    0.8986    0.9970  </span></span>
<span id="cb458-7"><a href="analysis-of-variance.html#cb458-7" tabindex="-1"></a><span class="co">#&gt; 3 treatment knee-eyes             0  1.216     0.2598    2.172     0.01168</span></span></code></pre></div>
<p>The Tukey HSD test by default give us 95% confidence intervals for the differences in means between each pair of groups, and an associated P-value for the null hypothesis of equal means between pairs. Interpretting the results above, we see that we fail to reject the null hypothesis of equal means for the knee and control treatment groups (i.e. we have no statistical support to conclude they are different). However, we reject the null hypothesis for equality of means between control and eye treatments and between knee and eye treatements. We have evidence that light treatments applied to the eye cause a mean negative shift in the phase of melatonin production relative to control and knee treatment groups.</p>
</div>
</div>
<div id="repeatability" class="section level2 hasAnchor" number="20.13">
<h2><span class="header-section-number">20.13</span> Repeatability<a href="analysis-of-variance.html#repeatability" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>Nearly every measure of a continuous measurement or assay we apply in biology (and other sicences) has associated with it some <strong>measurement error</strong>.</p></li>
<li><p>Measurement error is usually not “biological variation” of interest, but rather “technical variation” associated with our ability to measure quantities of interest precisely</p>
<ul>
<li><strong>Example</strong>: Have three people independently measure the length of a human femur to the nearest millimeter. Unless the values are aggressively rounded, there is a high likelihood that you’ll get three different values.</li>
</ul></li>
</ul>
<div id="estimating-repeatability-using-anova" class="section level3 unnumbered hasAnchor">
<h3>Estimating Repeatability using ANOVA<a href="analysis-of-variance.html#estimating-repeatability-using-anova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>ANOVA can be used to estimate the <strong>Repeatability</strong> of a measure, which provides a way to quantify how much of the variance we observe between individuals is due to measurement error.</p>
<p><strong>Estimating Repeatability: Experimental steps</strong></p>
<ul>
<li>Repeatedly, but independently, measure the same variable in the same individual or other unit of observation</li>
<li>Carry out the same repeated measurements across individuals</li>
</ul>
<p><strong>Estimating Repeatability: Statistical steps</strong></p>
<ul>
<li>Calculate <span class="math inline">\(MS_\text{groups}\)</span> and <span class="math inline">\(MS_\text{error}\)</span> where individuals are the grouping unit.</li>
<li>Estimate the variance among groups, <span class="math inline">\(s_A^2\)</span> as:
<span class="math display">\[
s_A^2 = \frac{MS_\text{groups} - MS_\text{error}}{n}
\]</span></li>
</ul>
<p>where <span class="math inline">\(n\)</span> is the number of replicate measures per individual.</p>
<ul>
<li><p>Estimate the repeatability as:
<span class="math display">\[
\text{Repeatability} = \frac{s_A^2}{s_A^2 + MS_\text{error}}
\]</span></p></li>
<li><p><span class="math inline">\(0 \leq \text{Repeatability} \leq 1\)</span>; values near zero indicate nearly all variance is due to measurement error, values near one indicate small fraction of variance due to measurement error</p></li>
</ul>
</div>
<div id="repeatability-walking-stick-example" class="section level3 unnumbered hasAnchor">
<h3>Repeatability: Walking stick example<a href="analysis-of-variance.html#repeatability-walking-stick-example" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Nosil and Crespi measured various morphological features of walking stick insects from digital photographs. For each specimen they took two independent photographs and measured femur length (in cm) on each photograph.</p>
<div class="sourceCode" id="cb459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb459-1"><a href="analysis-of-variance.html#cb459-1" tabindex="-1"></a>walking.sticks <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;https://github.com/bio304-class/bio304-course-notes/raw/master/datasets/ABD-walking-sticks.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="analysis-of-variance.html#cb460-1" tabindex="-1"></a>by.specimen <span class="ot">&lt;-</span></span>
<span id="cb460-2"><a href="analysis-of-variance.html#cb460-2" tabindex="-1"></a>  walking.sticks <span class="sc">%&gt;%</span></span>
<span id="cb460-3"><a href="analysis-of-variance.html#cb460-3" tabindex="-1"></a>  <span class="fu">group_by</span>(specimen) <span class="sc">%&gt;%</span></span>
<span id="cb460-4"><a href="analysis-of-variance.html#cb460-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">min.fl =</span> <span class="fu">min</span>(femurLength), <span class="at">max.fl =</span> <span class="fu">max</span>(femurLength), <span class="at">avg.fl =</span> <span class="fl">0.5</span><span class="sc">*</span>(min.fl <span class="sc">+</span> max.fl)) <span class="sc">%&gt;%</span></span>
<span id="cb460-5"><a href="analysis-of-variance.html#cb460-5" tabindex="-1"></a>  <span class="fu">arrange</span>(avg.fl) <span class="sc">%&gt;%</span></span>
<span id="cb460-6"><a href="analysis-of-variance.html#cb460-6" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb460-7"><a href="analysis-of-variance.html#cb460-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank.fl =</span> <span class="fu">rank</span>(avg.fl, <span class="at">ties.method =</span> <span class="st">&quot;first&quot;</span>))</span>
<span id="cb460-8"><a href="analysis-of-variance.html#cb460-8" tabindex="-1"></a></span>
<span id="cb460-9"><a href="analysis-of-variance.html#cb460-9" tabindex="-1"></a><span class="fu">ggplot</span>(by.specimen, <span class="fu">aes</span>(<span class="at">x =</span> rank.fl)) <span class="sc">+</span> </span>
<span id="cb460-10"><a href="analysis-of-variance.html#cb460-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> min.fl), <span class="at">color=</span><span class="st">&quot;firebrick&quot;</span>) <span class="sc">+</span> </span>
<span id="cb460-11"><a href="analysis-of-variance.html#cb460-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> max.fl <span class="sc">+</span> <span class="fl">0.0025</span>), <span class="at">color=</span><span class="st">&quot;firebrick&quot;</span>) <span class="sc">+</span></span>
<span id="cb460-12"><a href="analysis-of-variance.html#cb460-12" tabindex="-1"></a>  <span class="fu">geom_linerange</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> min.fl, <span class="at">ymax =</span> max.fl <span class="sc">+</span> <span class="fl">0.0025</span>)) <span class="sc">+</span></span>
<span id="cb460-13"><a href="analysis-of-variance.html#cb460-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Individual walking sticks&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Femur length (cm)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb460-14"><a href="analysis-of-variance.html#cb460-14" tabindex="-1"></a>  <span class="fu">theme_classic</span>()<span class="sc">+</span></span>
<span id="cb460-15"><a href="analysis-of-variance.html#cb460-15" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb460-16"><a href="analysis-of-variance.html#cb460-16" tabindex="-1"></a>        <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>())  </span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-477-1.png" width="312" style="display: block; margin: auto;" /></p>
<p>First we carry out the ANOVA in the standard way, using the specimen variable as the grouping variable:</p>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="analysis-of-variance.html#cb461-1" tabindex="-1"></a>sticks.aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(femurLength <span class="sc">~</span> <span class="fu">as.factor</span>(specimen), </span>
<span id="cb461-2"><a href="analysis-of-variance.html#cb461-2" tabindex="-1"></a>                  <span class="at">data =</span> walking.sticks)</span>
<span id="cb461-3"><a href="analysis-of-variance.html#cb461-3" tabindex="-1"></a>sticks.table <span class="ot">&lt;-</span> <span class="fu">tidy</span>(sticks.aov)</span>
<span id="cb461-4"><a href="analysis-of-variance.html#cb461-4" tabindex="-1"></a>sticks.table</span>
<span id="cb461-5"><a href="analysis-of-variance.html#cb461-5" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2 × 6</span></span>
<span id="cb461-6"><a href="analysis-of-variance.html#cb461-6" tabindex="-1"></a><span class="co">#&gt;   term                   df    sumsq    meansq statistic      p.value</span></span>
<span id="cb461-7"><a href="analysis-of-variance.html#cb461-7" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;               &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;        &lt;dbl&gt;</span></span>
<span id="cb461-8"><a href="analysis-of-variance.html#cb461-8" tabindex="-1"></a><span class="co">#&gt; 1 as.factor(specimen)    24 0.05913  0.002464      6.921  0.000004077</span></span>
<span id="cb461-9"><a href="analysis-of-variance.html#cb461-9" tabindex="-1"></a><span class="co">#&gt; 2 Residuals              25 0.008900 0.0003560    NA     NA</span></span></code></pre></div>
<p>From the ANOVA table, use <span class="math inline">\(MS_\text{group}\)</span> and <span class="math inline">\(MS_\text{error}\)</span> to calculate <span class="math inline">\(s_A\)</span>:</p>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="analysis-of-variance.html#cb462-1" tabindex="-1"></a>repeatability.table <span class="ot">&lt;-</span> <span class="fu">data_frame</span>(</span>
<span id="cb462-2"><a href="analysis-of-variance.html#cb462-2" tabindex="-1"></a>  <span class="at">MS.groups =</span> sticks.table<span class="sc">$</span>meansq[<span class="dv">1</span>],</span>
<span id="cb462-3"><a href="analysis-of-variance.html#cb462-3" tabindex="-1"></a>  <span class="at">MS.error =</span> sticks.table<span class="sc">$</span>meansq[<span class="dv">2</span>],</span>
<span id="cb462-4"><a href="analysis-of-variance.html#cb462-4" tabindex="-1"></a>  <span class="at">sA =</span> (MS.groups <span class="sc">-</span> MS.error)<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb462-5"><a href="analysis-of-variance.html#cb462-5" tabindex="-1"></a>  <span class="at">repeatability =</span> sA<span class="sc">/</span>(sA <span class="sc">+</span> MS.error))</span>
<span id="cb462-6"><a href="analysis-of-variance.html#cb462-6" tabindex="-1"></a></span>
<span id="cb462-7"><a href="analysis-of-variance.html#cb462-7" tabindex="-1"></a>repeatability.table</span>
<span id="cb462-8"><a href="analysis-of-variance.html#cb462-8" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 4</span></span>
<span id="cb462-9"><a href="analysis-of-variance.html#cb462-9" tabindex="-1"></a><span class="co">#&gt;   MS.groups  MS.error       sA repeatability</span></span>
<span id="cb462-10"><a href="analysis-of-variance.html#cb462-10" tabindex="-1"></a><span class="co">#&gt;       &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span id="cb462-11"><a href="analysis-of-variance.html#cb462-11" tabindex="-1"></a><span class="co">#&gt; 1  0.002464 0.0003560 0.001054        0.7475</span></span></code></pre></div>
<p>From the table we calculated:</p>
<ul>
<li><span class="math inline">\(s_A^2 =\)</span> 0.0010539</li>
<li><span class="math inline">\(\text{Repeatability} =\)</span> 0.7475028</li>
</ul>
<p>A repeatability of 0.75 indicates that we would estimate that ~75% of the total variance in femure length measurements in our population of walking stick insects is the result of true differences in femur length between individuals, while roughly 25% of the variance is attributable to measurement error.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="comparing-sample-means.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="violations-of-assumptions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["bio304-book.pdf"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  },
  "toolbar": {
    "position": "fixed"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
