<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 23 Multiple regression | Biology 304: Biological Data Analysis</title>
  <meta name="description" content="Hands-on materials for Bio 304 course at Duke University." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 23 Multiple regression | Biology 304: Biological Data Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Hands-on materials for Bio 304 course at Duke University." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 23 Multiple regression | Biology 304: Biological Data Analysis" />
  
  <meta name="twitter:description" content="Hands-on materials for Bio 304 course at Duke University." />
  

<meta name="author" content="Paul M. Magwene" />


<meta name="date" content="2025-06-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bivariate-linear-regression.html"/>
<link rel="next" href="logistic-regression.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="libs/rglWebGL-binding-1.3.18/rglWebGL.js"></script>
<link href="libs/rglwidgetClass-1.3.18/rgl.css" rel="stylesheet" />
<script src="libs/rglwidgetClass-1.3.18/rglClass.min.js"></script>
<script src="libs/CanvasMatrix4-1.3.18/CanvasMatrix.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#how-to-use-these-lecture-notes"><i class="fa fa-check"></i><b>1.1</b> How to use these lecture notes</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html"><i class="fa fa-check"></i><b>2</b> Data story: Women and children first on the Titanic?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#background"><i class="fa fa-check"></i><b>2.1</b> Background</a></li>
<li class="chapter" data-level="2.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#dataset"><i class="fa fa-check"></i><b>2.2</b> Dataset</a></li>
<li class="chapter" data-level="2.3" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#libraries"><i class="fa fa-check"></i><b>2.3</b> Libraries</a></li>
<li class="chapter" data-level="2.4" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#read-data"><i class="fa fa-check"></i><b>2.4</b> Read data</a></li>
<li class="chapter" data-level="2.5" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#whats-in-the-data"><i class="fa fa-check"></i><b>2.5</b> What’s in the data?</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#simple-data-wrangling"><i class="fa fa-check"></i><b>2.5.1</b> Simple data wrangling</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#categorizing-passengers"><i class="fa fa-check"></i><b>2.6</b> Categorizing passengers</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#how-many-people-survived"><i class="fa fa-check"></i><b>2.6.1</b> How many people survived?</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#women-first"><i class="fa fa-check"></i><b>2.6.2</b> Women first?</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#a-bar-plot-using-proportions-rather-than-counts"><i class="fa fa-check"></i><b>2.6.3</b> A bar plot using proportions rather than counts</a></li>
<li class="chapter" data-level="2.6.4" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#mosaic-plots"><i class="fa fa-check"></i><b>2.6.4</b> Mosaic plots</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#passenger-ages"><i class="fa fa-check"></i><b>2.7</b> Passenger ages</a></li>
<li class="chapter" data-level="2.8" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#how-does-age-relate-to-survival"><i class="fa fa-check"></i><b>2.8</b> How does age relate to survival?</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#strip-charts"><i class="fa fa-check"></i><b>2.8.1</b> Strip charts</a></li>
<li class="chapter" data-level="2.8.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#box-plots"><i class="fa fa-check"></i><b>2.8.2</b> Box plots</a></li>
<li class="chapter" data-level="2.8.3" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#fitting-a-model-relating-survival-to-age-and-sex"><i class="fa fa-check"></i><b>2.8.3</b> Fitting a model relating survival to age and sex</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#how-does-class-affect-survival"><i class="fa fa-check"></i><b>2.9</b> How does class affect survival?</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#combining-sex-and-class"><i class="fa fa-check"></i><b>2.9.1</b> Combining Sex and Class</a></li>
<li class="chapter" data-level="2.9.2" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#extending-the-logistic-regression-model-to-consider-class-age-sex-and-survival"><i class="fa fa-check"></i><b>2.9.2</b> Extending the logistic regression model to consider class, age, sex, and survival</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="data-story-women-and-children-first-on-the-titanic.html"><a href="data-story-women-and-children-first-on-the-titanic.html#conclusion"><i class="fa fa-check"></i><b>2.10</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>3</b> Getting Started with R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#what-is-r"><i class="fa fa-check"></i><b>3.1</b> What is R?</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#what-is-rstudio"><i class="fa fa-check"></i><b>3.2</b> What is RStudio?</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#entering-commands-in-the-console"><i class="fa fa-check"></i><b>3.3</b> Entering commands in the console</a></li>
<li class="chapter" data-level="3.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#comments"><i class="fa fa-check"></i><b>3.4</b> Comments</a></li>
<li class="chapter" data-level="3.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#using-r-as-a-calculator"><i class="fa fa-check"></i><b>3.5</b> Using R as a Calculator</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#common-mathematical-functions"><i class="fa fa-check"></i><b>3.5.1</b> Common mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#the-r-help-system"><i class="fa fa-check"></i><b>3.6</b> The R Help System</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#getting-help-from-the-console"><i class="fa fa-check"></i><b>3.6.1</b> Getting help from the console</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#variable-assignment-in-r"><i class="fa fa-check"></i><b>3.7</b> Variable assignment in R</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#valid-variable-names"><i class="fa fa-check"></i><b>3.7.1</b> Valid variable names</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#data-types"><i class="fa fa-check"></i><b>3.8</b> Data types</a></li>
<li class="chapter" data-level="3.9" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#logical-values"><i class="fa fa-check"></i><b>3.9</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#logical-operators"><i class="fa fa-check"></i><b>3.9.1</b> Logical operators</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#character-strings"><i class="fa fa-check"></i><b>3.10</b> Character strings</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#joining-strings"><i class="fa fa-check"></i><b>3.10.1</b> Joining strings</a></li>
<li class="chapter" data-level="3.10.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#splitting-strings"><i class="fa fa-check"></i><b>3.10.2</b> Splitting strings</a></li>
<li class="chapter" data-level="3.10.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#substrings"><i class="fa fa-check"></i><b>3.10.3</b> Substrings</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#packages"><i class="fa fa-check"></i><b>3.11</b> Packages</a>
<ul>
<li class="chapter" data-level="3.11.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-packages-from-the-console"><i class="fa fa-check"></i><b>3.11.1</b> Installing packages from the console</a></li>
<li class="chapter" data-level="3.11.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#install-the-tidyverse-package"><i class="fa fa-check"></i><b>3.11.2</b> Install the tidyverse package</a></li>
<li class="chapter" data-level="3.11.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-packages-from-the-rstudio-dialog"><i class="fa fa-check"></i><b>3.11.3</b> Installing packages from the RStudio dialog</a></li>
<li class="chapter" data-level="3.11.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#loading-packages-with-the-library-function"><i class="fa fa-check"></i><b>3.11.4</b> Loading packages with the <code>library()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="3.12" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#reading-data-from-a-file"><i class="fa fa-check"></i><b>3.12</b> Reading data from a file</a></li>
<li class="chapter" data-level="3.13" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#exploring-the-possums-data-set"><i class="fa fa-check"></i><b>3.13</b> Exploring the “possums” data set</a></li>
<li class="chapter" data-level="3.14" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#simple-tables"><i class="fa fa-check"></i><b>3.14</b> Simple tables</a></li>
<li class="chapter" data-level="3.15" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#simple-figures"><i class="fa fa-check"></i><b>3.15</b> Simple figures</a></li>
<li class="chapter" data-level="3.16" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#bar-plots"><i class="fa fa-check"></i><b>3.16</b> Bar plots</a></li>
<li class="chapter" data-level="3.17" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#histograms"><i class="fa fa-check"></i><b>3.17</b> Histograms</a></li>
<li class="chapter" data-level="3.18" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#scatter-plots"><i class="fa fa-check"></i><b>3.18</b> Scatter plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html"><i class="fa fa-check"></i><b>4</b> R Markdown and R Notebooks</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#r-notebooks"><i class="fa fa-check"></i><b>4.1</b> R Notebooks</a></li>
<li class="chapter" data-level="4.2" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#creating-an-r-notebook"><i class="fa fa-check"></i><b>4.2</b> Creating an R Notebook</a></li>
<li class="chapter" data-level="4.3" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#the-default-r-notebook-template"><i class="fa fa-check"></i><b>4.3</b> The default R Notebook template</a></li>
<li class="chapter" data-level="4.4" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#code-and-non-code-blocks"><i class="fa fa-check"></i><b>4.4</b> Code and Non-code blocks</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#non-code-blocks"><i class="fa fa-check"></i><b>4.4.1</b> Non-code blocks</a></li>
<li class="chapter" data-level="4.4.2" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#code-blocks"><i class="fa fa-check"></i><b>4.4.2</b> Code blocks</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#running-a-code-chunk"><i class="fa fa-check"></i><b>4.5</b> Running a code chunk</a></li>
<li class="chapter" data-level="4.6" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#running-all-code-chunks-above"><i class="fa fa-check"></i><b>4.6</b> Running all code chunks above</a></li>
<li class="chapter" data-level="4.7" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#knitting-an-r-markdown-to-html"><i class="fa fa-check"></i><b>4.7</b> “Knitting an” R Markdown to HTML</a></li>
<li class="chapter" data-level="4.8" data-path="r-markdown-and-r-notebooks.html"><a href="r-markdown-and-r-notebooks.html#sharing-your-reproducible-r-notebook"><i class="fa fa-check"></i><b>4.8</b> Sharing your reproducible R Notebook</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html"><i class="fa fa-check"></i><b>5</b> More R Basics: Data structures</a>
<ul>
<li class="chapter" data-level="5.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vectors"><i class="fa fa-check"></i><b>5.1</b> Vectors</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vector-arithmetic"><i class="fa fa-check"></i><b>5.1.1</b> Vector Arithmetic</a></li>
<li class="chapter" data-level="5.1.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vector-recycling"><i class="fa fa-check"></i><b>5.1.2</b> Vector recycling</a></li>
<li class="chapter" data-level="5.1.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#simple-statistical-functions-for-numeric-vectors"><i class="fa fa-check"></i><b>5.1.3</b> Simple statistical functions for numeric vectors</a></li>
<li class="chapter" data-level="5.1.4" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#indexing-vectors"><i class="fa fa-check"></i><b>5.1.4</b> Indexing Vectors</a></li>
<li class="chapter" data-level="5.1.5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#comparison-operators-applied-to-vectors"><i class="fa fa-check"></i><b>5.1.5</b> Comparison operators applied to vectors</a></li>
<li class="chapter" data-level="5.1.6" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#combining-indexing-and-comparison-of-vectors"><i class="fa fa-check"></i><b>5.1.6</b> Combining Indexing and Comparison of Vectors</a></li>
<li class="chapter" data-level="5.1.7" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vector-manipulation"><i class="fa fa-check"></i><b>5.1.7</b> Vector manipulation</a></li>
<li class="chapter" data-level="5.1.8" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#vectors-from-regular-sequences"><i class="fa fa-check"></i><b>5.1.8</b> Vectors from regular sequences</a></li>
<li class="chapter" data-level="5.1.9" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#additional-functions-for-working-with-vectors"><i class="fa fa-check"></i><b>5.1.9</b> Additional functions for working with vectors</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#lists"><i class="fa fa-check"></i><b>5.2</b> Lists</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#length-and-type-of-lists"><i class="fa fa-check"></i><b>5.2.1</b> Length and type of lists</a></li>
<li class="chapter" data-level="5.2.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#indexing-lists"><i class="fa fa-check"></i><b>5.2.2</b> Indexing lists</a></li>
<li class="chapter" data-level="5.2.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#naming-list-elements"><i class="fa fa-check"></i><b>5.2.3</b> Naming list elements</a></li>
<li class="chapter" data-level="5.2.4" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#the-operator"><i class="fa fa-check"></i><b>5.2.4</b> The <code>$</code> operator</a></li>
<li class="chapter" data-level="5.2.5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#changing-and-adding-lists-items"><i class="fa fa-check"></i><b>5.2.5</b> Changing and adding lists items</a></li>
<li class="chapter" data-level="5.2.6" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#combining-lists"><i class="fa fa-check"></i><b>5.2.6</b> Combining lists</a></li>
<li class="chapter" data-level="5.2.7" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#converting-lists-to-vectors"><i class="fa fa-check"></i><b>5.2.7</b> Converting lists to vectors</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#data-frames"><i class="fa fa-check"></i><b>5.3</b> Data frames</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#creating-a-data-frame"><i class="fa fa-check"></i><b>5.3.1</b> Creating a data frame</a></li>
<li class="chapter" data-level="5.3.2" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#type-and-class-for-data-frames"><i class="fa fa-check"></i><b>5.3.2</b> Type and class for data frames</a></li>
<li class="chapter" data-level="5.3.3" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#length-and-dimension-for-data-frames"><i class="fa fa-check"></i><b>5.3.3</b> Length and dimension for data frames</a></li>
<li class="chapter" data-level="5.3.4" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#indexing-and-accessing-data-frames"><i class="fa fa-check"></i><b>5.3.4</b> Indexing and accessing data frames</a></li>
<li class="chapter" data-level="5.3.5" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#logical-indexing-of-data-frames"><i class="fa fa-check"></i><b>5.3.5</b> Logical indexing of data frames</a></li>
<li class="chapter" data-level="5.3.6" data-path="more-r-basics-data-structures.html"><a href="more-r-basics-data-structures.html#adding-columns-to-a-data-frame"><i class="fa fa-check"></i><b>5.3.6</b> Adding columns to a data frame</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html"><i class="fa fa-check"></i><b>6</b> Introduction to ggplot2</a>
<ul>
<li class="chapter" data-level="6.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#loading-ggplot2"><i class="fa fa-check"></i><b>6.1</b> Loading ggplot2</a></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#example-data-set-andersons-iris-data"><i class="fa fa-check"></i><b>6.2</b> Example data set: Anderson’s Iris Data</a></li>
<li class="chapter" data-level="6.3" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#template-for-single-layer-plots-in-ggplot2"><i class="fa fa-check"></i><b>6.3</b> Template for single layer plots in ggplot2</a></li>
<li class="chapter" data-level="6.4" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#an-aside-about-function-arguments"><i class="fa fa-check"></i><b>6.4</b> An aside about function arguments</a></li>
<li class="chapter" data-level="6.5" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#strip-plots"><i class="fa fa-check"></i><b>6.5</b> Strip plots</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#jittering-data"><i class="fa fa-check"></i><b>6.5.1</b> Jittering data</a></li>
<li class="chapter" data-level="6.5.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#adding-categorical-information"><i class="fa fa-check"></i><b>6.5.2</b> Adding categorical information</a></li>
<li class="chapter" data-level="6.5.3" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#rotating-plot-coordinates"><i class="fa fa-check"></i><b>6.5.3</b> Rotating plot coordinates</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#histograms-1"><i class="fa fa-check"></i><b>6.6</b> Histograms</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#variations-on-histograms-when-considering-categorical-data"><i class="fa fa-check"></i><b>6.6.1</b> Variations on histograms when considering categorical data</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#faceting-to-depict-categorical-information"><i class="fa fa-check"></i><b>6.7</b> Faceting to depict categorical information</a></li>
<li class="chapter" data-level="6.8" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#density-plots"><i class="fa fa-check"></i><b>6.8</b> Density plots</a></li>
<li class="chapter" data-level="6.9" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#violin-or-beanplot"><i class="fa fa-check"></i><b>6.9</b> Violin or Beanplot</a></li>
<li class="chapter" data-level="6.10" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#boxplots"><i class="fa fa-check"></i><b>6.10</b> Boxplots</a></li>
<li class="chapter" data-level="6.11" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#building-complex-visualizations-with-layers"><i class="fa fa-check"></i><b>6.11</b> Building complex visualizations with layers</a></li>
<li class="chapter" data-level="6.12" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#useful-combination-plots"><i class="fa fa-check"></i><b>6.12</b> Useful combination plots</a>
<ul>
<li class="chapter" data-level="6.12.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#boxplot-plus-strip-plot"><i class="fa fa-check"></i><b>6.12.1</b> Boxplot plus strip plot</a></li>
<li class="chapter" data-level="6.12.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#setting-shared-aesthetics"><i class="fa fa-check"></i><b>6.12.2</b> Setting shared aesthetics</a></li>
</ul></li>
<li class="chapter" data-level="6.13" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#ggplot-layers-can-be-assigned-to-variables"><i class="fa fa-check"></i><b>6.13</b> ggplot layers can be assigned to variables</a>
<ul>
<li class="chapter" data-level="6.13.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#violin-plot-plus-strip-plot"><i class="fa fa-check"></i><b>6.13.1</b> Violin plot plus strip plot</a></li>
</ul></li>
<li class="chapter" data-level="6.14" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#adding-titles-and-tweaking-axis-labels"><i class="fa fa-check"></i><b>6.14</b> Adding titles and tweaking axis labels</a></li>
<li class="chapter" data-level="6.15" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#ggplot2-themes"><i class="fa fa-check"></i><b>6.15</b> ggplot2 themes</a>
<ul>
<li class="chapter" data-level="6.15.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#further-customization-with-ggplot2theme"><i class="fa fa-check"></i><b>6.15.1</b> Further customization with <code>ggplot2::theme</code></a></li>
</ul></li>
<li class="chapter" data-level="6.16" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#other-aspects-of-ggplots-can-be-assigned-to-variables"><i class="fa fa-check"></i><b>6.16</b> Other aspects of ggplots can be assigned to variables</a></li>
<li class="chapter" data-level="6.17" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#bivariate-plots"><i class="fa fa-check"></i><b>6.17</b> Bivariate plots</a>
<ul>
<li class="chapter" data-level="6.17.1" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#scatter-plots-1"><i class="fa fa-check"></i><b>6.17.1</b> Scatter plots</a></li>
<li class="chapter" data-level="6.17.2" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#adding-a-trend-line-to-a-scatter-plot"><i class="fa fa-check"></i><b>6.17.2</b> Adding a trend line to a scatter plot</a></li>
</ul></li>
<li class="chapter" data-level="6.18" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#bivariate-density-plots"><i class="fa fa-check"></i><b>6.18</b> Bivariate density plots</a></li>
<li class="chapter" data-level="6.19" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#combining-scatter-plots-and-density-plots-with-categorical-information"><i class="fa fa-check"></i><b>6.19</b> Combining Scatter Plots and Density Plots with Categorical Information</a></li>
<li class="chapter" data-level="6.20" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#density-plots-with-fill"><i class="fa fa-check"></i><b>6.20</b> Density plots with fill</a></li>
<li class="chapter" data-level="6.21" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#d-bin-and-hex-plots"><i class="fa fa-check"></i><b>6.21</b> 2D bin and hex plots</a></li>
<li class="chapter" data-level="6.22" data-path="introduction-to-ggplot2.html"><a href="introduction-to-ggplot2.html#the-cowplot-package"><i class="fa fa-check"></i><b>6.22</b> The <code>cowplot</code> package</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html"><i class="fa fa-check"></i><b>7</b> Introduction to the <code>dplyr</code> package</a>
<ul>
<li class="chapter" data-level="7.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#libraries-1"><i class="fa fa-check"></i><b>7.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#reading-data-with-the-readr-package"><i class="fa fa-check"></i><b>7.2</b> Reading data with the <code>readr</code> package</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#example-data-nc-births"><i class="fa fa-check"></i><b>7.2.1</b> Example data: NC Births</a></li>
<li class="chapter" data-level="7.2.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#reading-excel-files"><i class="fa fa-check"></i><b>7.2.2</b> Reading Excel files</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#a-note-on-tibbles"><i class="fa fa-check"></i><b>7.3</b> A note on “tibbles”</a></li>
<li class="chapter" data-level="7.4" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#data-filtering-and-transformation-with-dplyr"><i class="fa fa-check"></i><b>7.4</b> Data filtering and transformation with <code>dplyr</code></a></li>
<li class="chapter" data-level="7.5" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#dplyrs-verbs"><i class="fa fa-check"></i><b>7.5</b> dplyr’s “verbs”</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#select"><i class="fa fa-check"></i><b>7.5.1</b> <code>select</code></a></li>
<li class="chapter" data-level="7.5.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#filter"><i class="fa fa-check"></i><b>7.5.2</b> <code>filter</code></a></li>
<li class="chapter" data-level="7.5.3" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#mutate"><i class="fa fa-check"></i><b>7.5.3</b> <code>mutate</code></a></li>
<li class="chapter" data-level="7.5.4" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#arrange"><i class="fa fa-check"></i><b>7.5.4</b> <code>arrange</code></a></li>
<li class="chapter" data-level="7.5.5" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#summarize"><i class="fa fa-check"></i><b>7.5.5</b> <code>summarize</code></a></li>
<li class="chapter" data-level="7.5.6" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#group_by"><i class="fa fa-check"></i><b>7.5.6</b> <code>group_by</code></a></li>
<li class="chapter" data-level="7.5.7" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#combining-grouping-and-summarizing"><i class="fa fa-check"></i><b>7.5.7</b> Combining grouping and summarizing</a></li>
<li class="chapter" data-level="7.5.8" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#scoped-variants-of-mutate-and-summarize"><i class="fa fa-check"></i><b>7.5.8</b> Scoped variants of <code>mutate</code> and <code>summarize</code></a></li>
<li class="chapter" data-level="7.5.9" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#combining-summarize-with-grouping-aesthetics-in-ggplot2"><i class="fa fa-check"></i><b>7.5.9</b> Combining summarize with grouping aesthetics in <code>ggplot2</code></a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#pipes"><i class="fa fa-check"></i><b>7.6</b> Pipes</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#install-and-load-magrittr"><i class="fa fa-check"></i><b>7.6.1</b> Install and load <code>magrittr</code></a></li>
<li class="chapter" data-level="7.6.2" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-basic-pipe-operator"><i class="fa fa-check"></i><b>7.6.2</b> The basic pipe operator</a></li>
<li class="chapter" data-level="7.6.3" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#an-example-without-pipes"><i class="fa fa-check"></i><b>7.6.3</b> An example without pipes</a></li>
<li class="chapter" data-level="7.6.4" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-same-example-using-pipes"><i class="fa fa-check"></i><b>7.6.4</b> The same example using pipes</a></li>
<li class="chapter" data-level="7.6.5" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#assigning-the-output-of-a-statement-involving-pipes-to-a-variable"><i class="fa fa-check"></i><b>7.6.5</b> Assigning the output of a statement involving pipes to a variable</a></li>
<li class="chapter" data-level="7.6.6" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#compound-assignment-pipe-operator"><i class="fa fa-check"></i><b>7.6.6</b> Compound assignment pipe operator</a></li>
<li class="chapter" data-level="7.6.7" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-dot-operator-with-pipes"><i class="fa fa-check"></i><b>7.6.7</b> The dot operator with pipes</a></li>
<li class="chapter" data-level="7.6.8" data-path="introduction-to-the-dplyr-package.html"><a href="introduction-to-the-dplyr-package.html#the-exposition-pipe-operator"><i class="fa fa-check"></i><b>7.6.8</b> The exposition pipe operator</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html"><i class="fa fa-check"></i><b>8</b> Data wrangling, Part I</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#libraries-2"><i class="fa fa-check"></i><b>8.1</b> Libraries</a></li>
<li class="chapter" data-level="8.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#data"><i class="fa fa-check"></i><b>8.2</b> Data</a></li>
<li class="chapter" data-level="8.3" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#renaming-data-frame-columms"><i class="fa fa-check"></i><b>8.3</b> Renaming data frame columms</a></li>
<li class="chapter" data-level="8.4" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#dropping-unneeded-columns"><i class="fa fa-check"></i><b>8.4</b> Dropping unneeded columns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#finding-all-empty-columns"><i class="fa fa-check"></i><b>8.4.1</b> Finding all empty columns</a></li>
<li class="chapter" data-level="8.4.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#dropping-columns-by-matching-names"><i class="fa fa-check"></i><b>8.4.2</b> Dropping columns by matching names</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#merging-data-frames"><i class="fa fa-check"></i><b>8.5</b> Merging data frames</a></li>
<li class="chapter" data-level="8.6" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#reshaping-data-with-tidyr"><i class="fa fa-check"></i><b>8.6</b> Reshaping data with tidyr</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#wide-to-long-conversions-using-tidyrgather"><i class="fa fa-check"></i><b>8.6.1</b> Wide to long conversions using <code>tidyr::gather</code></a></li>
<li class="chapter" data-level="8.6.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#extracting-information-from-combined-variables-using-tidyrextract"><i class="fa fa-check"></i><b>8.6.2</b> Extracting information from combined variables using <code>tidyr::extract</code></a></li>
<li class="chapter" data-level="8.6.3" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#subsetting-rows"><i class="fa fa-check"></i><b>8.6.3</b> Subsetting rows</a></li>
<li class="chapter" data-level="8.6.4" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#splitting-columns"><i class="fa fa-check"></i><b>8.6.4</b> Splitting columns</a></li>
<li class="chapter" data-level="8.6.5" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#combining-data-frames"><i class="fa fa-check"></i><b>8.6.5</b> Combining data frames</a></li>
<li class="chapter" data-level="8.6.6" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#sorting-data-frame-rows"><i class="fa fa-check"></i><b>8.6.6</b> Sorting data frame rows</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#using-your-tidy-data"><i class="fa fa-check"></i><b>8.7</b> Using your tidy data</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#visualizing-gene-expression-time-series"><i class="fa fa-check"></i><b>8.7.1</b> Visualizing gene expression time series</a></li>
<li class="chapter" data-level="8.7.2" data-path="data-wrangling-part-i.html"><a href="data-wrangling-part-i.html#finding-the-most-variable-genes"><i class="fa fa-check"></i><b>8.7.2</b> Finding the most variable genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html"><i class="fa fa-check"></i><b>9</b> Data wrangling, Part II</a>
<ul>
<li class="chapter" data-level="9.1" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#libraries-3"><i class="fa fa-check"></i><b>9.1</b> Libraries</a></li>
<li class="chapter" data-level="9.2" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#data-1"><i class="fa fa-check"></i><b>9.2</b> Data</a></li>
<li class="chapter" data-level="9.3" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#heat-maps"><i class="fa fa-check"></i><b>9.3</b> Heat maps</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#better-color-schemes-with-rcolorbrewer"><i class="fa fa-check"></i><b>9.3.1</b> Better color schemes with RColorBrewer</a></li>
<li class="chapter" data-level="9.3.2" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#looking-for-patterns-using-sorted-data-and-heat-maps"><i class="fa fa-check"></i><b>9.3.2</b> Looking for patterns using sorted data and heat maps</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#long-to-wide-conversion-using-tidyrspread"><i class="fa fa-check"></i><b>9.4</b> Long-to-wide conversion using <code>tidyr::spread</code></a></li>
<li class="chapter" data-level="9.5" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#exploring-bivariate-relationships-using-wide-data"><i class="fa fa-check"></i><b>9.5</b> Exploring bivariate relationships using “wide” data</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#large-scale-patterns-of-correlations"><i class="fa fa-check"></i><b>9.5.1</b> Large scale patterns of correlations</a></li>
<li class="chapter" data-level="9.5.2" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#adding-new-columns-and-combining-filtered-data-frames"><i class="fa fa-check"></i><b>9.5.2</b> Adding new columns and combining filtered data frames</a></li>
<li class="chapter" data-level="9.5.3" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#a-heat-mapped-sorted-by-correlations"><i class="fa fa-check"></i><b>9.5.3</b> A heat mapped sorted by correlations</a></li>
<li class="chapter" data-level="9.5.4" data-path="data-wrangling-part-ii.html"><a href="data-wrangling-part-ii.html#a-fancy-figure"><i class="fa fa-check"></i><b>9.5.4</b> A “fancy” figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html"><i class="fa fa-check"></i><b>10</b> Functions and control flow statements</a>
<ul>
<li class="chapter" data-level="10.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#writing-your-own-functions"><i class="fa fa-check"></i><b>10.1</b> Writing your own functions</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#function-arguments"><i class="fa fa-check"></i><b>10.1.1</b> Function arguments</a></li>
<li class="chapter" data-level="10.1.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#writing-functions-with-optional-arguments"><i class="fa fa-check"></i><b>10.1.2</b> Writing functions with optional arguments</a></li>
<li class="chapter" data-level="10.1.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#putting-r-functions-in-scripts"><i class="fa fa-check"></i><b>10.1.3</b> Putting R functions in Scripts</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#control-flow-statements"><i class="fa fa-check"></i><b>10.2</b> Control flow statements</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#if-and-if-else-statements"><i class="fa fa-check"></i><b>10.2.1</b> <code>if</code> and <code>if-else</code> statements</a></li>
<li class="chapter" data-level="10.2.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#for-loops"><i class="fa fa-check"></i><b>10.2.2</b> for loops</a></li>
<li class="chapter" data-level="10.2.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#break-statement"><i class="fa fa-check"></i><b>10.2.3</b> <code>break</code> statement</a></li>
<li class="chapter" data-level="10.2.4" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#repeat-loops"><i class="fa fa-check"></i><b>10.2.4</b> <code>repeat</code> loops</a></li>
<li class="chapter" data-level="10.2.5" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#next-statement"><i class="fa fa-check"></i><b>10.2.5</b> <code>next</code> statement</a></li>
<li class="chapter" data-level="10.2.6" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#while-statements"><i class="fa fa-check"></i><b>10.2.6</b> while statements</a></li>
<li class="chapter" data-level="10.2.7" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#ifelse"><i class="fa fa-check"></i><b>10.2.7</b> <code>ifelse</code></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#map-and-related-tools"><i class="fa fa-check"></i><b>10.3</b> <code>map</code> and related tools</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#basic-map"><i class="fa fa-check"></i><b>10.3.1</b> basic <code>map</code></a></li>
<li class="chapter" data-level="10.3.2" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#map_if-and-map_at"><i class="fa fa-check"></i><b>10.3.2</b> <code>map_if</code> and <code>map_at</code></a></li>
<li class="chapter" data-level="10.3.3" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#mapping-in-parallel-using-map2"><i class="fa fa-check"></i><b>10.3.3</b> mapping in parallel using <code>map2</code></a></li>
<li class="chapter" data-level="10.3.4" data-path="functions-and-control-flow-statements.html"><a href="functions-and-control-flow-statements.html#map-variants-that-return-vectors"><i class="fa fa-check"></i><b>10.3.4</b> <code>map</code> variants that return vectors</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="frequency-distributions-and-descriptive-statistics.html"><a href="frequency-distributions-and-descriptive-statistics.html"><i class="fa fa-check"></i><b>11</b> Frequency Distributions and Descriptive Statistics</a></li>
<li class="chapter" data-level="12" data-path="joint-frequency-distributions-and-measures-of-association.html"><a href="joint-frequency-distributions-and-measures-of-association.html"><i class="fa fa-check"></i><b>12</b> Joint Frequency Distributions and Measures of Association</a></li>
<li class="chapter" data-level="13" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html"><i class="fa fa-check"></i><b>13</b> Introduction to Probability</a>
<ul>
<li class="chapter" data-level="13.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#terms"><i class="fa fa-check"></i><b>13.1</b> Terms</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#examples-of-random-trials-outcomes-and-events"><i class="fa fa-check"></i><b>13.1.1</b> Examples of random trials, outcomes, and events</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#frequentist-definition-of-probability"><i class="fa fa-check"></i><b>13.2</b> Frequentist definition of probability</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#examples-probability"><i class="fa fa-check"></i><b>13.2.1</b> Examples: Probability</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#probability-distribution"><i class="fa fa-check"></i><b>13.3</b> Probability distribution</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#discrete-probability-distribution"><i class="fa fa-check"></i><b>13.3.1</b> Discrete probability distribution</a></li>
<li class="chapter" data-level="13.3.2" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#continuous-probability-distribution"><i class="fa fa-check"></i><b>13.3.2</b> Continuous probability distribution</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#mutually-exclusive-events"><i class="fa fa-check"></i><b>13.4</b> Mutually exclusive events</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#addition-rule-mutually-exclusive-events"><i class="fa fa-check"></i><b>13.4.1</b> Addition rule, mutually exclusive events</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#independence"><i class="fa fa-check"></i><b>13.5</b> Independence</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#multiplication-rule-independent-events"><i class="fa fa-check"></i><b>13.5.1</b> Multiplication rule, independent events</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#general-addition-rule"><i class="fa fa-check"></i><b>13.6</b> General addition rule</a></li>
<li class="chapter" data-level="13.7" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#conditional-probability"><i class="fa fa-check"></i><b>13.7</b> Conditional probability</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#example-conditional-probability"><i class="fa fa-check"></i><b>13.7.1</b> Example: Conditional probability</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#general-multiplication-rule"><i class="fa fa-check"></i><b>13.8</b> General multiplication rule</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#example-general-multiplication-rule"><i class="fa fa-check"></i><b>13.8.1</b> Example: General multiplication rule</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#probability-trees"><i class="fa fa-check"></i><b>13.9</b> Probability trees</a></li>
<li class="chapter" data-level="13.10" data-path="introduction-to-probability.html"><a href="introduction-to-probability.html#law-of-total-probability"><i class="fa fa-check"></i><b>13.10</b> Law of total probability</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html"><i class="fa fa-check"></i><b>14</b> Introduction to Sampling Distributions, Part I</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#libraries-4"><i class="fa fa-check"></i><b>14.1</b> Libraries</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#in-class-experiments"><i class="fa fa-check"></i><b>14.2</b> In class experiments</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#experiment-1"><i class="fa fa-check"></i><b>14.2.1</b> Experiment 1</a></li>
<li class="chapter" data-level="14.2.2" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#experiment-2"><i class="fa fa-check"></i><b>14.2.2</b> Experiment 2</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#simulating-sampling-in-r"><i class="fa fa-check"></i><b>14.3</b> Simulating sampling in R</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#binomial-distribution"><i class="fa fa-check"></i><b>14.3.1</b> Binomial distribution</a></li>
<li class="chapter" data-level="14.3.2" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#sampling-from-the-binomial-distribution-in-r"><i class="fa fa-check"></i><b>14.3.2</b> Sampling from the binomial distribution in R</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-sampling-distributions-part-i.html"><a href="introduction-to-sampling-distributions-part-i.html#distribution-of-estimates-of-the-proportion"><i class="fa fa-check"></i><b>14.4</b> Distribution of estimates of the proportion</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html"><i class="fa fa-check"></i><b>15</b> Introduction to Sampling Distributions, Part II</a>
<ul>
<li class="chapter" data-level="15.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#libraries-5"><i class="fa fa-check"></i><b>15.1</b> Libraries</a></li>
<li class="chapter" data-level="15.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#data-set-simulated-male-heights"><i class="fa fa-check"></i><b>15.2</b> Data set: Simulated male heights</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#properties-of-the-underlying-population"><i class="fa fa-check"></i><b>15.2.1</b> Properties of the underlying population</a></li>
<li class="chapter" data-level="15.2.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#other-r-functions-related-to-the-normal-distribution"><i class="fa fa-check"></i><b>15.2.2</b> Other R functions related to the normal distribution</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#seeding-the-pseudo-random-number-generator"><i class="fa fa-check"></i><b>15.3</b> Seeding the pseudo-random number generator</a></li>
<li class="chapter" data-level="15.4" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#random-sampling-from-the-simulated-population"><i class="fa fa-check"></i><b>15.4</b> Random sampling from the simulated population</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#another-random-sample"><i class="fa fa-check"></i><b>15.4.1</b> Another random sample</a></li>
<li class="chapter" data-level="15.4.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#simulating-the-generation-of-many-random-samples"><i class="fa fa-check"></i><b>15.4.2</b> Simulating the generation of many random samples</a></li>
<li class="chapter" data-level="15.4.3" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#a-function-to-estimate-statistics-of-interest-in-a-random-sample"><i class="fa fa-check"></i><b>15.4.3</b> A function to estimate statistics of interest in a random sample</a></li>
<li class="chapter" data-level="15.4.4" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#generating-statistics-for-many-random-samples"><i class="fa fa-check"></i><b>15.4.4</b> Generating statistics for many random samples</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#simulated-sampling-distribution-of-the-mean"><i class="fa fa-check"></i><b>15.5</b> Simulated sampling distribution of the mean</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#differences-between-sampling-distribution-and-samplepopulation-distributions"><i class="fa fa-check"></i><b>15.5.1</b> Differences between sampling distribution and sample/population distributions</a></li>
<li class="chapter" data-level="15.5.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#use-sampling-distributions-to-understand-the-behavior-of-statistics-of-interest"><i class="fa fa-check"></i><b>15.5.2</b> Use sampling distributions to understand the behavior of statistics of interest</a></li>
<li class="chapter" data-level="15.5.3" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#sampling-distributions-for-different-sample-sizes"><i class="fa fa-check"></i><b>15.5.3</b> Sampling distributions for different sample sizes</a></li>
<li class="chapter" data-level="15.5.4" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#discussion-of-trends-for-sampling-distributions-of-different-sample-sizes"><i class="fa fa-check"></i><b>15.5.4</b> Discussion of trends for sampling distributions of different sample sizes</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>15.6</b> Standard Error of the Mean</a></li>
<li class="chapter" data-level="15.7" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#sampling-distribution-of-the-standard-deviation"><i class="fa fa-check"></i><b>15.7</b> Sampling Distribution of the Standard Deviation</a>
<ul>
<li class="chapter" data-level="15.7.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#standard-error-of-standard-deviations"><i class="fa fa-check"></i><b>15.7.1</b> Standard error of standard deviations</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#what-happens-to-the-sampling-distribution-of-the-mean-and-standard-deviation-when-our-sample-size-is-small"><i class="fa fa-check"></i><b>15.8</b> What happens to the sampling distribution of the mean and standard deviation when our sample size is small?</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#for-small-samples-sample-standard-deviations-systematically-underestimate-the-population-standard-deviation"><i class="fa fa-check"></i><b>15.8.1</b> For small samples, sample standard deviations systematically underestimate the population standard deviation</a></li>
<li class="chapter" data-level="15.8.2" data-path="introduction-to-sampling-distributions-part-ii.html"><a href="introduction-to-sampling-distributions-part-ii.html#underestimates-of-the-standard-deviation-given-small-n-lead-to-understimates-of-the-se-of-the-mean"><i class="fa fa-check"></i><b>15.8.2</b> Underestimates of the standard deviation given small <span class="math inline">\(n\)</span> lead to understimates of the SE of the mean</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html"><i class="fa fa-check"></i><b>16</b> Introduction to hypothesis testing</a>
<ul>
<li class="chapter" data-level="16.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#libraries-6"><i class="fa fa-check"></i><b>16.1</b> Libraries</a></li>
<li class="chapter" data-level="16.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses"><i class="fa fa-check"></i><b>16.2</b> Null and Alternative Hypotheses</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-hypotheses"><i class="fa fa-check"></i><b>16.2.1</b> Null hypotheses</a></li>
<li class="chapter" data-level="16.2.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#alternative-hypotheses"><i class="fa fa-check"></i><b>16.2.2</b> Alternative hypotheses</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#rejecting-failing-to-reject-null-hypotheses"><i class="fa fa-check"></i><b>16.3</b> Rejecting / failing to reject null hypotheses</a></li>
<li class="chapter" data-level="16.4" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#outcomes-of-hypothesis-tests"><i class="fa fa-check"></i><b>16.4</b> Outcomes of hypothesis tests</a></li>
<li class="chapter" data-level="16.5" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#using-p-values-to-assess-the-strength-of-evidence-against-the-null-hypothesis"><i class="fa fa-check"></i><b>16.5</b> Using p-values to assess the strength of evidence against the null hypothesis</a></li>
<li class="chapter" data-level="16.6" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#example-comparing-a-sample-mean-to-an-hypothesized-normal-distribution"><i class="fa fa-check"></i><b>16.6</b> Example: Comparing a sample mean to an hypothesized normal distribution</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses-1"><i class="fa fa-check"></i><b>16.6.1</b> Null and alternative hypotheses</a></li>
<li class="chapter" data-level="16.6.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#sampling-distribution-of-the-mean"><i class="fa fa-check"></i><b>16.6.2</b> Sampling distribution of the mean</a></li>
<li class="chapter" data-level="16.6.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#calculating-a-p-value"><i class="fa fa-check"></i><b>16.6.3</b> Calculating a p-value</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#example-handedness-of-toads"><i class="fa fa-check"></i><b>16.7</b> Example: Handedness of toads</a>
<ul>
<li class="chapter" data-level="16.7.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses-2"><i class="fa fa-check"></i><b>16.7.1</b> Null and alternative hypotheses</a></li>
<li class="chapter" data-level="16.7.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#data-2"><i class="fa fa-check"></i><b>16.7.2</b> Data</a></li>
<li class="chapter" data-level="16.7.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#sampling-distribution-for-proportions-binomial-distribution"><i class="fa fa-check"></i><b>16.7.3</b> Sampling distribution for proportions: Binomial distribution</a></li>
<li class="chapter" data-level="16.7.4" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#calculating-a-p-value-for-the-binomial-test"><i class="fa fa-check"></i><b>16.7.4</b> Calculating a p-value for the binomial test</a></li>
<li class="chapter" data-level="16.7.5" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#the-binom.test-function"><i class="fa fa-check"></i><b>16.7.5</b> The <code>binom.test()</code> function</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#example-measuring-the-association-between-maternal-smoking-and-premature-births"><i class="fa fa-check"></i><b>16.8</b> Example: Measuring the association between maternal smoking and premature births</a>
<ul>
<li class="chapter" data-level="16.8.1" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#null-and-alternative-hypotheses-3"><i class="fa fa-check"></i><b>16.8.1</b> Null and alternative hypotheses:</a></li>
<li class="chapter" data-level="16.8.2" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#contingency-table-analysis-using-the-chi2-statistic"><i class="fa fa-check"></i><b>16.8.2</b> Contingency table analysis using the <span class="math inline">\(\chi^2\)</span> statistic</a></li>
<li class="chapter" data-level="16.8.3" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#chi2-distribution"><i class="fa fa-check"></i><b>16.8.3</b> <span class="math inline">\(\chi^2\)</span>-distribution</a></li>
<li class="chapter" data-level="16.8.4" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#carrying-out-a-hypothesis-test-using-the-chi2"><i class="fa fa-check"></i><b>16.8.4</b> Carrying out a hypothesis test using the <span class="math inline">\(\chi^2\)</span></a></li>
<li class="chapter" data-level="16.8.5" data-path="introduction-to-hypothesis-testing.html"><a href="introduction-to-hypothesis-testing.html#interpretation-of-the-chi2-test"><i class="fa fa-check"></i><b>16.8.5</b> Interpretation of the <span class="math inline">\(\chi^2\)</span>-test</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html"><i class="fa fa-check"></i><b>17</b> Introduction to confidence intervals</a>
<ul>
<li class="chapter" data-level="17.1" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#confidence-intervals"><i class="fa fa-check"></i><b>17.1</b> Confidence Intervals</a></li>
<li class="chapter" data-level="17.2" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#generic-formulation-for-confidence-intervals"><i class="fa fa-check"></i><b>17.2</b> Generic formulation for confidence intervals</a></li>
<li class="chapter" data-level="17.3" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#example-confidence-intervals-for-the-mean"><i class="fa fa-check"></i><b>17.3</b> Example: Confidence intervals for the mean</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#simulation-of-means"><i class="fa fa-check"></i><b>17.3.1</b> Simulation of means</a></li>
<li class="chapter" data-level="17.3.2" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#distance-between-sample-means-and-true-means"><i class="fa fa-check"></i><b>17.3.2</b> Distance between sample means and true means</a></li>
<li class="chapter" data-level="17.3.3" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#calculating-a-ci"><i class="fa fa-check"></i><b>17.3.3</b> Calculating a CI</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="introduction-to-confidence-intervals.html"><a href="introduction-to-confidence-intervals.html#a-problem-arises"><i class="fa fa-check"></i><b>17.4</b> A problem arises!</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="normal-distributions.html"><a href="normal-distributions.html"><i class="fa fa-check"></i><b>18</b> Normal distributions</a>
<ul>
<li class="chapter" data-level="18.1" data-path="normal-distributions.html"><a href="normal-distributions.html#basics-about-normal-distributions"><i class="fa fa-check"></i><b>18.1</b> Basics about normal distributions</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="normal-distributions.html"><a href="normal-distributions.html#notation"><i class="fa fa-check"></i><b>18.1.1</b> Notation</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="normal-distributions.html"><a href="normal-distributions.html#normal-distribution-probability-density-function"><i class="fa fa-check"></i><b>18.2</b> Normal distribution, probability density function</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="normal-distributions.html"><a href="normal-distributions.html#dnorm-calculates-the-normal-pdf"><i class="fa fa-check"></i><b>18.2.1</b> <code>dnorm()</code> calculates the normal pdf</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="normal-distributions.html"><a href="normal-distributions.html#approximately-normal-distributions-are-very-common"><i class="fa fa-check"></i><b>18.3</b> Approximately normal distributions are very common</a></li>
<li class="chapter" data-level="18.4" data-path="normal-distributions.html"><a href="normal-distributions.html#central-limit-theorem"><i class="fa fa-check"></i><b>18.4</b> Central limit theorem</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="normal-distributions.html"><a href="normal-distributions.html#example-continuous-variation-from-discrete-loci"><i class="fa fa-check"></i><b>18.4.1</b> Example: Continuous variation from discrete loci</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="normal-distributions.html"><a href="normal-distributions.html#visualizing-normal-distributions"><i class="fa fa-check"></i><b>18.5</b> Visualizing normal distributions</a></li>
<li class="chapter" data-level="18.6" data-path="normal-distributions.html"><a href="normal-distributions.html#comparing-values-from-different-normal-distributions"><i class="fa fa-check"></i><b>18.6</b> Comparing values from different normal distributions</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="normal-distributions.html"><a href="normal-distributions.html#standardized-or-z-scores"><i class="fa fa-check"></i><b>18.6.1</b> Standardized or Z-scores</a></li>
</ul></li>
<li class="chapter" data-level="18.7" data-path="normal-distributions.html"><a href="normal-distributions.html#standard-normal-distribution"><i class="fa fa-check"></i><b>18.7</b> Standard normal distribution</a></li>
<li class="chapter" data-level="18.8" data-path="normal-distributions.html"><a href="normal-distributions.html#rule"><i class="fa fa-check"></i><b>18.8</b> 88-95-99.7 Rule</a></li>
<li class="chapter" data-level="18.9" data-path="normal-distributions.html"><a href="normal-distributions.html#percentiles"><i class="fa fa-check"></i><b>18.9</b> Percentiles</a></li>
<li class="chapter" data-level="18.10" data-path="normal-distributions.html"><a href="normal-distributions.html#cutoff-points"><i class="fa fa-check"></i><b>18.10</b> Cutoff points</a></li>
<li class="chapter" data-level="18.11" data-path="normal-distributions.html"><a href="normal-distributions.html#assessing-normality"><i class="fa fa-check"></i><b>18.11</b> Assessing normality</a>
<ul>
<li class="chapter" data-level="18.11.1" data-path="normal-distributions.html"><a href="normal-distributions.html#comparing-histograms-to-theoretical-normals"><i class="fa fa-check"></i><b>18.11.1</b> Comparing histograms to theoretical normals</a></li>
<li class="chapter" data-level="18.11.2" data-path="normal-distributions.html"><a href="normal-distributions.html#normal-probability-plot"><i class="fa fa-check"></i><b>18.11.2</b> Normal probability plot</a></li>
<li class="chapter" data-level="18.11.3" data-path="normal-distributions.html"><a href="normal-distributions.html#comparing-the-empirical-cdf-to-the-theoretical-cdf"><i class="fa fa-check"></i><b>18.11.3</b> Comparing the empirical CDF to the theoretical CDF</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html"><i class="fa fa-check"></i><b>19</b> Comparing sample means</a>
<ul>
<li class="chapter" data-level="19.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#hypothesis-test-for-the-mean-using-the-t-distribution"><i class="fa fa-check"></i><b>19.1</b> Hypothesis test for the mean using the t-distribution</a></li>
<li class="chapter" data-level="19.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#one-sample-t-test"><i class="fa fa-check"></i><b>19.2</b> One sample t-test</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#one-sample-t-test-test-statistic"><i class="fa fa-check"></i><b>19.2.1</b> One sample t-test, test statistic</a></li>
<li class="chapter" data-level="19.2.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#assumptions-of-one-sample-t-tests"><i class="fa fa-check"></i><b>19.2.2</b> Assumptions of one sample t-tests</a></li>
<li class="chapter" data-level="19.2.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#example-gene-expression-in-mice"><i class="fa fa-check"></i><b>19.2.3</b> Example: Gene expression in mice</a></li>
<li class="chapter" data-level="19.2.4" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#confidence-intervals-for-the-mean"><i class="fa fa-check"></i><b>19.2.4</b> Confidence intervals for the mean</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#the-t.test-function-in-r"><i class="fa fa-check"></i><b>19.3</b> The <code>t.test</code> function in R</a></li>
<li class="chapter" data-level="19.4" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#two-sample-t-test"><i class="fa fa-check"></i><b>19.4</b> Two sample t-test</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#standard-error-for-the-difference-in-means"><i class="fa fa-check"></i><b>19.4.1</b> Standard error for the difference in means</a></li>
<li class="chapter" data-level="19.4.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#two-sample-t-test-test-statistic"><i class="fa fa-check"></i><b>19.4.2</b> Two sample t-test, test statistic</a></li>
<li class="chapter" data-level="19.4.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#assumptions-of-two-sample-t-test"><i class="fa fa-check"></i><b>19.4.3</b> Assumptions of two sample t-test</a></li>
<li class="chapter" data-level="19.4.4" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#example-comparing-the-effects-of-two-drugs"><i class="fa fa-check"></i><b>19.4.4</b> Example: Comparing the effects of two drugs</a></li>
<li class="chapter" data-level="19.4.5" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#specifying-t.test-in-terms-of-a-formula"><i class="fa fa-check"></i><b>19.4.5</b> Specifying <code>t.test()</code> in terms of a formula</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#paired-t-test"><i class="fa fa-check"></i><b>19.5</b> Paired t-test</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#paired-t-test-test-statistic"><i class="fa fa-check"></i><b>19.5.1</b> Paired t-test, test statistic</a></li>
<li class="chapter" data-level="19.5.2" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#assumptions-of-paired-t-test"><i class="fa fa-check"></i><b>19.5.2</b> Assumptions of paired t-test</a></li>
<li class="chapter" data-level="19.5.3" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#paired-t-test-example"><i class="fa fa-check"></i><b>19.5.3</b> Paired t-test, example</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#the-fallacy-of-indirect-comparison"><i class="fa fa-check"></i><b>19.6</b> The fallacy of indirect comparison</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#interpreting-confidence-intervals-in-light-of-two-sample-t-tests"><i class="fa fa-check"></i><b>19.6.1</b> Interpreting confidence intervals in light of two sample t-tests</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="comparing-sample-means.html"><a href="comparing-sample-means.html#summary-table-for-different-t-tests"><i class="fa fa-check"></i><b>19.7</b> Summary table for different t-tests</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html"><i class="fa fa-check"></i><b>20</b> Analysis of Variance</a>
<ul>
<li class="chapter" data-level="20.1" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#hypotheses-for-anova"><i class="fa fa-check"></i><b>20.1</b> Hypotheses for ANOVA</a></li>
<li class="chapter" data-level="20.2" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-assumptions"><i class="fa fa-check"></i><b>20.2</b> ANOVA, assumptions</a></li>
<li class="chapter" data-level="20.3" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-key-idea"><i class="fa fa-check"></i><b>20.3</b> ANOVA, key idea</a></li>
<li class="chapter" data-level="20.4" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#partioning-of-sum-of-squares"><i class="fa fa-check"></i><b>20.4</b> Partioning of sum of squares</a></li>
<li class="chapter" data-level="20.5" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#mathematical-partitioning-of-sums-of-squares"><i class="fa fa-check"></i><b>20.5</b> Mathematical partitioning of sums of squares</a></li>
<li class="chapter" data-level="20.6" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-test-statistic-and-sampling-distribution"><i class="fa fa-check"></i><b>20.6</b> ANOVA test statistic and sampling distribution</a></li>
<li class="chapter" data-level="20.7" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-tables"><i class="fa fa-check"></i><b>20.7</b> ANOVA tables</a></li>
<li class="chapter" data-level="20.8" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#the-aov-function"><i class="fa fa-check"></i><b>20.8</b> The <code>aov()</code> function</a></li>
<li class="chapter" data-level="20.9" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#example-circadian-rythm-data"><i class="fa fa-check"></i><b>20.9</b> Example, circadian rythm data</a>
<ul>
<li class="chapter" data-level="20.9.1" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#visualizing-the-data"><i class="fa fa-check"></i><b>20.9.1</b> Visualizing the data</a></li>
<li class="chapter" data-level="20.9.2" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#carrying-out-the-anova"><i class="fa fa-check"></i><b>20.9.2</b> Carrying out the ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="20.10" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#anova-calculations-step-by-step"><i class="fa fa-check"></i><b>20.10</b> ANOVA calculations: Step-by-step</a></li>
<li class="chapter" data-level="20.11" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#visualizing-the-partitioning-of-sum-of-squares"><i class="fa fa-check"></i><b>20.11</b> Visualizing the partitioning of sum-of-squares</a></li>
<li class="chapter" data-level="20.12" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#which-pairs-of-group-means-are-different"><i class="fa fa-check"></i><b>20.12</b> Which pairs of group means are different?</a>
<ul>
<li class="chapter" data-level="20.12.1" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#tukey-kramer-test"><i class="fa fa-check"></i><b>20.12.1</b> Tukey-Kramer test</a></li>
</ul></li>
<li class="chapter" data-level="20.13" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#repeatability"><i class="fa fa-check"></i><b>20.13</b> Repeatability</a>
<ul>
<li class="chapter" data-level="" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#estimating-repeatability-using-anova"><i class="fa fa-check"></i>Estimating Repeatability using ANOVA</a></li>
<li class="chapter" data-level="" data-path="analysis-of-variance.html"><a href="analysis-of-variance.html#repeatability-walking-stick-example"><i class="fa fa-check"></i>Repeatability: Walking stick example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html"><i class="fa fa-check"></i><b>21</b> Violations of Assumptions</a>
<ul>
<li class="chapter" data-level="21.1" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#graphical-methods-for-detecting-deviations-from-normality"><i class="fa fa-check"></i><b>21.1</b> Graphical methods for detecting deviations from normality</a>
<ul>
<li class="chapter" data-level="" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#normal-probability-plots"><i class="fa fa-check"></i>Normal probability plots</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#formal-test-of-normality"><i class="fa fa-check"></i><b>21.2</b> Formal test of normality</a></li>
<li class="chapter" data-level="21.3" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#when-to-ignore-violations-of-assumptions"><i class="fa fa-check"></i><b>21.3</b> When to ignore violations of assumptions</a></li>
<li class="chapter" data-level="21.4" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#data-transformations"><i class="fa fa-check"></i><b>21.4</b> Data transformations</a>
<ul>
<li class="chapter" data-level="" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#example-log-transformation-to-deal-with-skew"><i class="fa fa-check"></i>Example: log transformation to deal with skew</a></li>
<li class="chapter" data-level="21.4.1" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#confidence-intervals-and-data-transformations"><i class="fa fa-check"></i><b>21.4.1</b> Confidence intervals and data transformations</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#nonparametric-tests"><i class="fa fa-check"></i><b>21.5</b> Nonparametric tests</a>
<ul>
<li class="chapter" data-level="21.5.1" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#mann-whitney-u-test-non-parametric-alternative-to-t-test"><i class="fa fa-check"></i><b>21.5.1</b> Mann-Whitney U-test: non-parametric alternative to t-test</a></li>
<li class="chapter" data-level="21.5.2" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#kruskal-wallis-test-non-parametric-alternative-to-anova"><i class="fa fa-check"></i><b>21.5.2</b> Kruskal-Wallis test: non-parametric alternative to ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="violations-of-assumptions.html"><a href="violations-of-assumptions.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html"><i class="fa fa-check"></i><b>22</b> Bivariate Linear Regression</a>
<ul>
<li class="chapter" data-level="22.1" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#regression-terminology"><i class="fa fa-check"></i><b>22.1</b> Regression terminology</a></li>
<li class="chapter" data-level="22.2" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#the-optimality-criterion-for-least-squares-regression"><i class="fa fa-check"></i><b>22.2</b> The optimality criterion for least-squares regression</a></li>
<li class="chapter" data-level="22.3" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#solution-for-the-least-squares-criterion"><i class="fa fa-check"></i><b>22.3</b> Solution for the least-squares criterion</a></li>
<li class="chapter" data-level="22.4" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#libraries-7"><i class="fa fa-check"></i><b>22.4</b> Libraries</a></li>
<li class="chapter" data-level="22.5" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#illustrating-linear-regression-with-simulated-data"><i class="fa fa-check"></i><b>22.5</b> Illustrating linear regression with simulated data</a></li>
<li class="chapter" data-level="22.6" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#specifying-regression-models-in-r"><i class="fa fa-check"></i><b>22.6</b> Specifying Regression Models in R</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#fitted-values"><i class="fa fa-check"></i><b>22.6.1</b> Fitted values</a></li>
<li class="chapter" data-level="22.6.2" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#getting-the-model-coefficients"><i class="fa fa-check"></i><b>22.6.2</b> Getting the model coefficients</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#residuals"><i class="fa fa-check"></i><b>22.7</b> Residuals</a></li>
<li class="chapter" data-level="22.8" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#regression-as-sum-of-squares-decomposition"><i class="fa fa-check"></i><b>22.8</b> Regression as sum-of-squares decomposition</a></li>
<li class="chapter" data-level="22.9" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#variance-explained-by-a-regression-model"><i class="fa fa-check"></i><b>22.9</b> Variance “explained” by a regression model</a></li>
<li class="chapter" data-level="22.10" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broom-a-library-for-converting-model-results-into-data-frames"><i class="fa fa-check"></i><b>22.10</b> Broom: a library for converting model results into data frames</a>
<ul>
<li class="chapter" data-level="22.10.1" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broomtidy"><i class="fa fa-check"></i><b>22.10.1</b> <code>broom::tidy</code></a></li>
<li class="chapter" data-level="22.10.2" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broomaugment"><i class="fa fa-check"></i><b>22.10.2</b> <code>broom::augment</code></a></li>
<li class="chapter" data-level="22.10.3" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#broomglance"><i class="fa fa-check"></i><b>22.10.3</b> <code>broom::glance</code></a></li>
</ul></li>
<li class="chapter" data-level="22.11" data-path="bivariate-linear-regression.html"><a href="bivariate-linear-regression.html#example-predicting-lion-age-based-on-nose-color"><i class="fa fa-check"></i><b>22.11</b> Example: Predicting lion age based on nose color</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="multiple-regression.html"><a href="multiple-regression.html"><i class="fa fa-check"></i><b>23</b> Multiple regression</a>
<ul>
<li class="chapter" data-level="23.1" data-path="multiple-regression.html"><a href="multiple-regression.html#libraries-to-install"><i class="fa fa-check"></i><b>23.1</b> Libraries to install</a></li>
<li class="chapter" data-level="23.2" data-path="multiple-regression.html"><a href="multiple-regression.html#review-of-bivariate-regression"><i class="fa fa-check"></i><b>23.2</b> Review of bivariate regression</a></li>
<li class="chapter" data-level="23.3" data-path="multiple-regression.html"><a href="multiple-regression.html#multiple-regression-1"><i class="fa fa-check"></i><b>23.3</b> Multiple regression</a>
<ul>
<li class="chapter" data-level="23.3.1" data-path="multiple-regression.html"><a href="multiple-regression.html#geometrical-interpretation"><i class="fa fa-check"></i><b>23.3.1</b> Geometrical interpretation</a></li>
<li class="chapter" data-level="23.3.2" data-path="multiple-regression.html"><a href="multiple-regression.html#coefficient-of-determination-for-multiple-regression"><i class="fa fa-check"></i><b>23.3.2</b> Coefficient of determination for multiple regression</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="multiple-regression.html"><a href="multiple-regression.html#interpretting-multiple-regression"><i class="fa fa-check"></i><b>23.4</b> Interpretting Multiple Regression</a></li>
<li class="chapter" data-level="23.5" data-path="multiple-regression.html"><a href="multiple-regression.html#libaries"><i class="fa fa-check"></i><b>23.5</b> Libaries</a></li>
<li class="chapter" data-level="23.6" data-path="multiple-regression.html"><a href="multiple-regression.html#example-data-set-mtcars"><i class="fa fa-check"></i><b>23.6</b> Example data set: <code>mtcars</code></a></li>
<li class="chapter" data-level="23.7" data-path="multiple-regression.html"><a href="multiple-regression.html#visualizing-and-summarizing-the-variables-of-interest"><i class="fa fa-check"></i><b>23.7</b> Visualizing and summarizing the variables of interest</a></li>
<li class="chapter" data-level="23.8" data-path="multiple-regression.html"><a href="multiple-regression.html#d-plots"><i class="fa fa-check"></i><b>23.8</b> 3D plots</a></li>
<li class="chapter" data-level="23.9" data-path="multiple-regression.html"><a href="multiple-regression.html#fitting-a-multiple-regression-model-in-r"><i class="fa fa-check"></i><b>23.9</b> Fitting a multiple regression model in R</a></li>
<li class="chapter" data-level="23.10" data-path="multiple-regression.html"><a href="multiple-regression.html#visualizing-the-regression-plane"><i class="fa fa-check"></i><b>23.10</b> Visualizing the regression plane</a></li>
<li class="chapter" data-level="23.11" data-path="multiple-regression.html"><a href="multiple-regression.html#interactive-3d-visualizations-using-opengl"><i class="fa fa-check"></i><b>23.11</b> Interactive 3D Visualizations Using OpenGL</a></li>
<li class="chapter" data-level="23.12" data-path="multiple-regression.html"><a href="multiple-regression.html#examining-the-residuals"><i class="fa fa-check"></i><b>23.12</b> Examining the residuals</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="logistic-regression.html"><a href="logistic-regression.html"><i class="fa fa-check"></i><b>24</b> Logistic regression</a>
<ul>
<li class="chapter" data-level="24.1" data-path="logistic-regression.html"><a href="logistic-regression.html#a-web-app-to-explore-the-logistic-regression-equation"><i class="fa fa-check"></i><b>24.1</b> A web app to explore the logistic regression equation</a></li>
<li class="chapter" data-level="24.2" data-path="logistic-regression.html"><a href="logistic-regression.html#titanic-data-set"><i class="fa fa-check"></i><b>24.2</b> Titanic data set</a></li>
<li class="chapter" data-level="24.3" data-path="logistic-regression.html"><a href="logistic-regression.html#subsetting-the-data"><i class="fa fa-check"></i><b>24.3</b> Subsetting the data</a></li>
<li class="chapter" data-level="24.4" data-path="logistic-regression.html"><a href="logistic-regression.html#visualizing-survival-as-a-function-of-age"><i class="fa fa-check"></i><b>24.4</b> Visualizing survival as a function of age</a></li>
<li class="chapter" data-level="24.5" data-path="logistic-regression.html"><a href="logistic-regression.html#fitting-the-logistic-regression-model"><i class="fa fa-check"></i><b>24.5</b> Fitting the logistic regression model</a></li>
<li class="chapter" data-level="24.6" data-path="logistic-regression.html"><a href="logistic-regression.html#visualizing-the-logistic-regression"><i class="fa fa-check"></i><b>24.6</b> Visualizing the logistic regression</a>
<ul>
<li class="chapter" data-level="24.6.1" data-path="logistic-regression.html"><a href="logistic-regression.html#quick-and-easy-visualization"><i class="fa fa-check"></i><b>24.6.1</b> Quick and easy visualization</a></li>
</ul></li>
<li class="chapter" data-level="24.7" data-path="logistic-regression.html"><a href="logistic-regression.html#impact-of-sex-and-passenger-class-on-the-models"><i class="fa fa-check"></i><b>24.7</b> Impact of sex and passenger class on the models</a></li>
<li class="chapter" data-level="24.8" data-path="logistic-regression.html"><a href="logistic-regression.html#fitting-multiple-models-based-on-groupings-use-dplyrdo"><i class="fa fa-check"></i><b>24.8</b> Fitting multiple models based on groupings use <code>dplyr::do</code></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biology 304: Biological Data Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multiple-regression" class="section level1 hasAnchor" number="23">
<h1><span class="header-section-number">Chapter 23</span> Multiple regression<a href="multiple-regression.html#multiple-regression" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="libraries-to-install" class="section level2 hasAnchor" number="23.1">
<h2><span class="header-section-number">23.1</span> Libraries to install<a href="multiple-regression.html#libraries-to-install" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll be using several new packages for this class session. Install the following packages via one of the standard install mechanisms:</p>
<ul>
<li><code>HistData</code> – provides the <code>GaltonFamilies</code> example data sets we’ll work with</li>
<li><code>plot3D</code> – for generating 3D plots</li>
<li><code>rgl</code> – NOTE: On OS X, <code>rgl</code> requires you to install a program called XQuartz. XQuartz can be downloaded from the <a href="https://www.xquartz.org">XQuartz Home Page</a>. If you’re on a Mac, install XQuartz before installing <code>rgl</code>. You may have to reboot your computer after installing XQuartz.</li>
</ul>
</div>
<div id="review-of-bivariate-regression" class="section level2 hasAnchor" number="23.2">
<h2><span class="header-section-number">23.2</span> Review of bivariate regression<a href="multiple-regression.html#review-of-bivariate-regression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Recall the model for bivariate least-squares regression. When we regress <span class="math inline">\(Y\)</span> and <span class="math inline">\(X\)</span> we’re looking for a linear function, <span class="math inline">\(f(X)\)</span>, for which the following sum-of-squared deviations is minimized:</p>
<p><span class="math display">\[
\sum_{i=1}^n (y_i - f(x_i))^2
\]</span></p>
<p>The general form a linear function of one variable is a line,</p>
<p><span class="math display">\[
\widehat{Y} = f(x)  = a + bX
\]</span></p>
<p>where <span class="math inline">\(b\)</span> is the slope of the line and <span class="math inline">\(a\)</span> is the intercept.</p>
</div>
<div id="multiple-regression-1" class="section level2 hasAnchor" number="23.3">
<h2><span class="header-section-number">23.3</span> Multiple regression<a href="multiple-regression.html#multiple-regression-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The idea behind multiple regression is almost exactly the same as bivariate regression, except now we try and fit a linear model for <span class="math inline">\(Y\)</span> using multiple explanatory variables, <span class="math inline">\(X_1, X_2,\ldots, X_m\)</span>. That is we’re looking for a linear function, <span class="math inline">\(f(X_1, X_2,\ldots,X_m)\)</span> that minimizes:</p>
<p><span class="math display">\[
\sum_{i=1}^n(y_i - f(x_1, x_2,\ldots, x_m))^2
\]</span></p>
<p>A linear function of more than one variable is written as:</p>
<p><span class="math display">\[
\widehat{Y} = f(X_1, X_2,\ldots,X_m) = a + b_1X_1 + b_2X_2 + \cdots + b_mX_m
\]</span></p>
<p>Where <span class="math inline">\(a\)</span> is the intercept and <span class="math inline">\(b_1, b_2,\ldots,b_m\)</span> are the <strong>regression coefficients</strong>.</p>
<div id="geometrical-interpretation" class="section level3 hasAnchor" number="23.3.1">
<h3><span class="header-section-number">23.3.1</span> Geometrical interpretation<a href="multiple-regression.html#geometrical-interpretation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Geometrically the regression coefficients have the same interpretation as in the bivariate case – slopes with respect to the corresponding variable. When there are two predictor variables, the linear regression is geometrically a plane in 3-space, as shown in the figure below. When there are more than two predictor variables, the regression solution is a hyper-plane.</p>
<div class="float">
<img src="figures/fig-regression-variable-space.png" alt="Multiple regression, two predictor variables" />
<div class="figcaption">Multiple regression, two predictor variables</div>
</div>
<p>Mathematically, the best fitting regression coefficients, <span class="math inline">\(b_1, b_2,\ldots,b_m\)</span>, are found using linear algebra. Since we haven’t covered linear algebra in this course, I will omit the details. Conceptually the thing to remember is that the regression coefficients are related to the magnitude of the standard deviations of the the predictor variables and the covariances between the predictor and outcome variables.</p>
</div>
<div id="coefficient-of-determination-for-multiple-regression" class="section level3 hasAnchor" number="23.3.2">
<h3><span class="header-section-number">23.3.2</span> Coefficient of determination for multiple regression<a href="multiple-regression.html#coefficient-of-determination-for-multiple-regression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As in bivariate regression, the coefficient of determination (<span class="math inline">\(R^2\)</span>) provides a measure of the proportion of variance in the outcome variable (<span class="math inline">\(Y\)</span>) “explained” by the predictor variables (<span class="math inline">\(X_1, X_2, \ldots\)</span>).</p>
</div>
</div>
<div id="interpretting-multiple-regression" class="section level2 hasAnchor" number="23.4">
<h2><span class="header-section-number">23.4</span> Interpretting Multiple Regression<a href="multiple-regression.html#interpretting-multiple-regression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here are some things to keep in mind when interpretting a multple regression:</p>
<ul>
<li><p>In most cases of regression, causal interpretation of the model is not justified.</p></li>
<li><p>Standard bivariate and multiple regression assumes that the predictor variables ( (<span class="math inline">\(X_1, X_2, \ldots\)</span>) are observed without error. That is, uncertainty in the regression model is only associated with the outcome variable, not the predictors.</p></li>
<li><p>Comparing the size of regression coefficients only makes sense if all the predictor (explanatory) variables have the same scale</p></li>
<li><p>If the explanatory variables (<span class="math inline">\(X_1, X_2,\ldots,X_m\)</span>) are highly correlated, then the regression solution can be “unstable” – a small change in the data could lead to a large change in the regression model.</p></li>
</ul>
</div>
<div id="libaries" class="section level2 hasAnchor" number="23.5">
<h2><span class="header-section-number">23.5</span> Libaries<a href="multiple-regression.html#libaries" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb501-1"><a href="multiple-regression.html#cb501-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb501-2"><a href="multiple-regression.html#cb501-2" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb501-3"><a href="multiple-regression.html#cb501-3" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb501-4"><a href="multiple-regression.html#cb501-4" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb501-5"><a href="multiple-regression.html#cb501-5" tabindex="-1"></a><span class="fu">library</span>(plot3D)</span>
<span id="cb501-6"><a href="multiple-regression.html#cb501-6" tabindex="-1"></a><span class="fu">library</span>(rgl)</span></code></pre></div>
</div>
<div id="example-data-set-mtcars" class="section level2 hasAnchor" number="23.6">
<h2><span class="header-section-number">23.6</span> Example data set: <code>mtcars</code><a href="multiple-regression.html#example-data-set-mtcars" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>mtcars</code> dataset contains information on fuel consumption and ten other aspects of car design (see <code>?mtcars</code> for more info). We’ll use multiple regression to model the relationship between fuel consumption (<code>mpg</code>) and a vehicles weight (<code>wt</code>) and horsepower (<code>hp</code>).</p>
</div>
<div id="visualizing-and-summarizing-the-variables-of-interest" class="section level2 hasAnchor" number="23.7">
<h2><span class="header-section-number">23.7</span> Visualizing and summarizing the variables of interest<a href="multiple-regression.html#visualizing-and-summarizing-the-variables-of-interest" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before carrying out any regression modle it’s always a good idea to start out with visualizations of the individual variables first.</p>
<div class="sourceCode" id="cb502"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb502-1"><a href="multiple-regression.html#cb502-1" tabindex="-1"></a>hist.wt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mtcars, <span class="fu">aes</span>(wt)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">8</span>)</span>
<span id="cb502-2"><a href="multiple-regression.html#cb502-2" tabindex="-1"></a>hist.hp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mtcars, <span class="fu">aes</span>(hp)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">8</span>)</span>
<span id="cb502-3"><a href="multiple-regression.html#cb502-3" tabindex="-1"></a>hist.mpg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mtcars, <span class="fu">aes</span>(mpg)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">8</span>)</span>
<span id="cb502-4"><a href="multiple-regression.html#cb502-4" tabindex="-1"></a></span>
<span id="cb502-5"><a href="multiple-regression.html#cb502-5" tabindex="-1"></a><span class="fu">plot_grid</span>(hist.wt, hist.hp, hist.mpg, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;C&quot;</span>))</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-520-1.png" width="864" /></p>
<p>Let’s also create some quick data summaries for our variables:</p>
<div class="sourceCode" id="cb503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb503-1"><a href="multiple-regression.html#cb503-1" tabindex="-1"></a>mtcars.subset <span class="ot">&lt;-</span> </span>
<span id="cb503-2"><a href="multiple-regression.html#cb503-2" tabindex="-1"></a>  mtcars <span class="sc">%&gt;%</span></span>
<span id="cb503-3"><a href="multiple-regression.html#cb503-3" tabindex="-1"></a>  <span class="fu">select</span>(wt, hp, mpg)</span>
<span id="cb503-4"><a href="multiple-regression.html#cb503-4" tabindex="-1"></a></span>
<span id="cb503-5"><a href="multiple-regression.html#cb503-5" tabindex="-1"></a><span class="fu">summary</span>(mtcars.subset)</span>
<span id="cb503-6"><a href="multiple-regression.html#cb503-6" tabindex="-1"></a><span class="co">#&gt;        wt              hp             mpg       </span></span>
<span id="cb503-7"><a href="multiple-regression.html#cb503-7" tabindex="-1"></a><span class="co">#&gt;  Min.   :1.513   Min.   : 52.0   Min.   :10.40  </span></span>
<span id="cb503-8"><a href="multiple-regression.html#cb503-8" tabindex="-1"></a><span class="co">#&gt;  1st Qu.:2.581   1st Qu.: 96.5   1st Qu.:15.43  </span></span>
<span id="cb503-9"><a href="multiple-regression.html#cb503-9" tabindex="-1"></a><span class="co">#&gt;  Median :3.325   Median :123.0   Median :19.20  </span></span>
<span id="cb503-10"><a href="multiple-regression.html#cb503-10" tabindex="-1"></a><span class="co">#&gt;  Mean   :3.217   Mean   :146.7   Mean   :20.09  </span></span>
<span id="cb503-11"><a href="multiple-regression.html#cb503-11" tabindex="-1"></a><span class="co">#&gt;  3rd Qu.:3.610   3rd Qu.:180.0   3rd Qu.:22.80  </span></span>
<span id="cb503-12"><a href="multiple-regression.html#cb503-12" tabindex="-1"></a><span class="co">#&gt;  Max.   :5.424   Max.   :335.0   Max.   :33.90</span></span></code></pre></div>
<p>And a correlation matrix to summarize the bivariate associations between the variables:</p>
<div class="sourceCode" id="cb504"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb504-1"><a href="multiple-regression.html#cb504-1" tabindex="-1"></a><span class="fu">cor</span>(mtcars.subset)</span>
<span id="cb504-2"><a href="multiple-regression.html#cb504-2" tabindex="-1"></a><span class="co">#&gt;             wt         hp        mpg</span></span>
<span id="cb504-3"><a href="multiple-regression.html#cb504-3" tabindex="-1"></a><span class="co">#&gt; wt   1.0000000  0.6587479 -0.8676594</span></span>
<span id="cb504-4"><a href="multiple-regression.html#cb504-4" tabindex="-1"></a><span class="co">#&gt; hp   0.6587479  1.0000000 -0.7761684</span></span>
<span id="cb504-5"><a href="multiple-regression.html#cb504-5" tabindex="-1"></a><span class="co">#&gt; mpg -0.8676594 -0.7761684  1.0000000</span></span></code></pre></div>
<p>We can use the <code>GGally::ggpairs()</code> function, which we’ve seen previously, to create a visualization of the bivariate relationships:</p>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="multiple-regression.html#cb505-1" tabindex="-1"></a><span class="fu">ggpairs</span>(mtcars.subset)</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-523-1.png" width="576" /></p>
<p>From the scatter plots and correlation matrix we see that weight and horsepower are positively correlated, but both are negatively correlated with fuel economy. This jives with our intuition – bigger cars with more powerful engines generally get lower gas mileage.</p>
</div>
<div id="d-plots" class="section level2 hasAnchor" number="23.8">
<h2><span class="header-section-number">23.8</span> 3D plots<a href="multiple-regression.html#d-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Since we’re building a model that involves three variables, it makes sense to look at at 3D plot. ggplot2 has no built in facilities for 3D scatter plots so we’ll use a package called <a href="http://www.rforscience.com/rpackages/visualisation/plot3d/"><code>plot3D</code></a>. <code>plot3D</code> follows the plotting conventions of the base R-graphics capabilities, so we can’t build up figures in layers in the same way we do in ggplot. Instead we pass all the formatting arguments to a single function call.</p>
<p>To create a 3D scatter plot we can use the <code>plot3D::points3D</code> function. The argument <code>pch</code> sets the type of plotting character to use in the plot (for a graphical key of the available plotting characters see <a href="https://www.statmethods.net/advgraphs/parameters.html">this link</a>).</p>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="multiple-regression.html#cb506-1" tabindex="-1"></a><span class="fu">library</span>(plot3D)</span>
<span id="cb506-2"><a href="multiple-regression.html#cb506-2" tabindex="-1"></a></span>
<span id="cb506-3"><a href="multiple-regression.html#cb506-3" tabindex="-1"></a><span class="co"># create short variable names for convenience</span></span>
<span id="cb506-4"><a href="multiple-regression.html#cb506-4" tabindex="-1"></a>wt <span class="ot">&lt;-</span> mtcars<span class="sc">$</span>wt  </span>
<span id="cb506-5"><a href="multiple-regression.html#cb506-5" tabindex="-1"></a>hp <span class="ot">&lt;-</span> mtcars<span class="sc">$</span>hp</span>
<span id="cb506-6"><a href="multiple-regression.html#cb506-6" tabindex="-1"></a>mpg <span class="ot">&lt;-</span> mtcars<span class="sc">$</span>mpg</span>
<span id="cb506-7"><a href="multiple-regression.html#cb506-7" tabindex="-1"></a></span>
<span id="cb506-8"><a href="multiple-regression.html#cb506-8" tabindex="-1"></a><span class="fu">points3D</span>(wt, hp, mpg,</span>
<span id="cb506-9"><a href="multiple-regression.html#cb506-9" tabindex="-1"></a>         <span class="at">xlab =</span> <span class="st">&quot;Weight&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Horse Power&quot;</span>, <span class="at">zlab =</span> <span class="st">&quot;MPG&quot;</span>,</span>
<span id="cb506-10"><a href="multiple-regression.html#cb506-10" tabindex="-1"></a>         <span class="at">pch =</span> <span class="dv">20</span>)  </span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-524-1.png" width="576" /></p>
<p>We can change the angle of the 3D plot using the arguments <code>theta</code> and <code>phi</code> which change the “azimuthal direction” and “colatitude” (inclination angle). See the wikipedia page on <a href="https://en.wikipedia.org/wiki/Spherical_coordinate_system">spherical coordinate systems</a> for more explanation of this values.</p>
<div class="sourceCode" id="cb507"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb507-1"><a href="multiple-regression.html#cb507-1" tabindex="-1"></a><span class="fu">points3D</span>(wt, hp, mpg,</span>
<span id="cb507-2"><a href="multiple-regression.html#cb507-2" tabindex="-1"></a>         <span class="at">xlab =</span> <span class="st">&quot;Weight&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Horse Power&quot;</span>, <span class="at">zlab =</span> <span class="st">&quot;MPG&quot;</span>,</span>
<span id="cb507-3"><a href="multiple-regression.html#cb507-3" tabindex="-1"></a>         <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb507-4"><a href="multiple-regression.html#cb507-4" tabindex="-1"></a>         <span class="at">theta =</span> <span class="dv">20</span>, <span class="at">phi =</span> <span class="dv">20</span>  <span class="co"># these set viewing angle</span></span>
<span id="cb507-5"><a href="multiple-regression.html#cb507-5" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-525-1.png" width="576" /></p>
<p>If you want the points to have a uniform color specify a single color in the <code>col</code> argument. Here we also add vertical lines to the plot using the <code>type</code> argument and show</p>
<div class="sourceCode" id="cb508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb508-1"><a href="multiple-regression.html#cb508-1" tabindex="-1"></a><span class="fu">points3D</span>(wt, hp, mpg,</span>
<span id="cb508-2"><a href="multiple-regression.html#cb508-2" tabindex="-1"></a>         <span class="at">xlab =</span> <span class="st">&quot;Weight&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Horse Power&quot;</span>, <span class="at">zlab =</span> <span class="st">&quot;MPG&quot;</span>,</span>
<span id="cb508-3"><a href="multiple-regression.html#cb508-3" tabindex="-1"></a>         <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb508-4"><a href="multiple-regression.html#cb508-4" tabindex="-1"></a>         <span class="at">theta =</span> <span class="dv">45</span>, <span class="at">phi =</span> <span class="dv">25</span>,</span>
<span id="cb508-5"><a href="multiple-regression.html#cb508-5" tabindex="-1"></a>         <span class="at">type =</span> <span class="st">&quot;h&quot;</span></span>
<span id="cb508-6"><a href="multiple-regression.html#cb508-6" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-526-1.png" width="576" /></p>
<p>For more examples of how you can modify plots generated with the <code>plot3D</code> package see <a href="http://www.sthda.com/english/wiki/impressive-package-for-3d-and-4d-graph-r-software-and-data-visualization">this web page</a>.</p>
</div>
<div id="fitting-a-multiple-regression-model-in-r" class="section level2 hasAnchor" number="23.9">
<h2><span class="header-section-number">23.9</span> Fitting a multiple regression model in R<a href="multiple-regression.html#fitting-a-multiple-regression-model-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using the <code>lm()</code> function, fitting multiple regression models is a straightforward extension of fitting a bivariate regression model.</p>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="multiple-regression.html#cb509-1" tabindex="-1"></a>fit.mpg <span class="ot">&lt;-</span> <span class="fu">lm</span>(mpg <span class="sc">~</span> wt <span class="sc">+</span> hp, <span class="at">data =</span> mtcars.subset)</span>
<span id="cb509-2"><a href="multiple-regression.html#cb509-2" tabindex="-1"></a><span class="fu">summary</span>(fit.mpg)</span>
<span id="cb509-3"><a href="multiple-regression.html#cb509-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb509-4"><a href="multiple-regression.html#cb509-4" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb509-5"><a href="multiple-regression.html#cb509-5" tabindex="-1"></a><span class="co">#&gt; lm(formula = mpg ~ wt + hp, data = mtcars.subset)</span></span>
<span id="cb509-6"><a href="multiple-regression.html#cb509-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb509-7"><a href="multiple-regression.html#cb509-7" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb509-8"><a href="multiple-regression.html#cb509-8" tabindex="-1"></a><span class="co">#&gt;    Min     1Q Median     3Q    Max </span></span>
<span id="cb509-9"><a href="multiple-regression.html#cb509-9" tabindex="-1"></a><span class="co">#&gt; -3.941 -1.600 -0.182  1.050  5.854 </span></span>
<span id="cb509-10"><a href="multiple-regression.html#cb509-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb509-11"><a href="multiple-regression.html#cb509-11" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb509-12"><a href="multiple-regression.html#cb509-12" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb509-13"><a href="multiple-regression.html#cb509-13" tabindex="-1"></a><span class="co">#&gt; (Intercept) 37.22727    1.59879  23.285  &lt; 2e-16 ***</span></span>
<span id="cb509-14"><a href="multiple-regression.html#cb509-14" tabindex="-1"></a><span class="co">#&gt; wt          -3.87783    0.63273  -6.129 1.12e-06 ***</span></span>
<span id="cb509-15"><a href="multiple-regression.html#cb509-15" tabindex="-1"></a><span class="co">#&gt; hp          -0.03177    0.00903  -3.519  0.00145 ** </span></span>
<span id="cb509-16"><a href="multiple-regression.html#cb509-16" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb509-17"><a href="multiple-regression.html#cb509-17" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb509-18"><a href="multiple-regression.html#cb509-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb509-19"><a href="multiple-regression.html#cb509-19" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 2.593 on 29 degrees of freedom</span></span>
<span id="cb509-20"><a href="multiple-regression.html#cb509-20" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.8268, Adjusted R-squared:  0.8148 </span></span>
<span id="cb509-21"><a href="multiple-regression.html#cb509-21" tabindex="-1"></a><span class="co">#&gt; F-statistic: 69.21 on 2 and 29 DF,  p-value: 9.109e-12</span></span></code></pre></div>
<p>As was the case for bivariate regression, the <code>broom</code> package functions <code>tidy</code>, <code>glance</code>, and <code>augment</code> can be useful for working with the results from fitting the mode.</p>
<div class="sourceCode" id="cb510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb510-1"><a href="multiple-regression.html#cb510-1" tabindex="-1"></a><span class="fu">tidy</span>(fit.mpg)</span>
<span id="cb510-2"><a href="multiple-regression.html#cb510-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 5</span></span>
<span id="cb510-3"><a href="multiple-regression.html#cb510-3" tabindex="-1"></a><span class="co">#&gt;   term        estimate std.error statistic   p.value</span></span>
<span id="cb510-4"><a href="multiple-regression.html#cb510-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span></span>
<span id="cb510-5"><a href="multiple-regression.html#cb510-5" tabindex="-1"></a><span class="co">#&gt; 1 (Intercept) 37.23     1.599       23.28  2.565e-20</span></span>
<span id="cb510-6"><a href="multiple-regression.html#cb510-6" tabindex="-1"></a><span class="co">#&gt; 2 wt          -3.878    0.6327      -6.129 1.120e- 6</span></span>
<span id="cb510-7"><a href="multiple-regression.html#cb510-7" tabindex="-1"></a><span class="co">#&gt; 3 hp          -0.03177  0.009030    -3.519 1.451e- 3</span></span></code></pre></div>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="multiple-regression.html#cb511-1" tabindex="-1"></a><span class="fu">glance</span>(fit.mpg)</span>
<span id="cb511-2"><a href="multiple-regression.html#cb511-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 12</span></span>
<span id="cb511-3"><a href="multiple-regression.html#cb511-3" tabindex="-1"></a><span class="co">#&gt;   r.squared adj.r.squared sigma statistic   p.value    df logLik   AIC   BIC</span></span>
<span id="cb511-4"><a href="multiple-regression.html#cb511-4" tabindex="-1"></a><span class="co">#&gt;       &lt;dbl&gt;         &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></span>
<span id="cb511-5"><a href="multiple-regression.html#cb511-5" tabindex="-1"></a><span class="co">#&gt; 1    0.8268        0.8148 2.593     69.21 9.109e-12     2 -74.33 156.7 162.5</span></span>
<span id="cb511-6"><a href="multiple-regression.html#cb511-6" tabindex="-1"></a><span class="co">#&gt; # ℹ 3 more variables: deviance &lt;dbl&gt;, df.residual &lt;int&gt;, nobs &lt;int&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb512-1"><a href="multiple-regression.html#cb512-1" tabindex="-1"></a>mtcars.subset.augmented <span class="ot">&lt;-</span></span>
<span id="cb512-2"><a href="multiple-regression.html#cb512-2" tabindex="-1"></a>  <span class="fu">augment</span>(fit.mpg, mtcars.subset)</span></code></pre></div>
</div>
<div id="visualizing-the-regression-plane" class="section level2 hasAnchor" number="23.10">
<h2><span class="header-section-number">23.10</span> Visualizing the regression plane<a href="multiple-regression.html#visualizing-the-regression-plane" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For multiple regression on two predictor variables we can visualize the plane of best fit but adding it as a surface to our 3D plot.</p>
<div class="sourceCode" id="cb513"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb513-1"><a href="multiple-regression.html#cb513-1" tabindex="-1"></a><span class="co"># Create a regular grid over the range of wt and hp values</span></span>
<span id="cb513-2"><a href="multiple-regression.html#cb513-2" tabindex="-1"></a>grid.lines <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb513-3"><a href="multiple-regression.html#cb513-3" tabindex="-1"></a></span>
<span id="cb513-4"><a href="multiple-regression.html#cb513-4" tabindex="-1"></a>wt.grid <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">min</span>(wt), <span class="fu">max</span>(wt), <span class="at">length.out =</span> grid.lines)</span>
<span id="cb513-5"><a href="multiple-regression.html#cb513-5" tabindex="-1"></a>hp.grid <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">min</span>(hp), <span class="fu">max</span>(hp), <span class="at">length.out =</span> grid.lines)</span>
<span id="cb513-6"><a href="multiple-regression.html#cb513-6" tabindex="-1"></a>wthp.grid <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">x =</span> wt.grid, <span class="at">y =</span> hp.grid)</span>
<span id="cb513-7"><a href="multiple-regression.html#cb513-7" tabindex="-1"></a></span>
<span id="cb513-8"><a href="multiple-regression.html#cb513-8" tabindex="-1"></a>grid.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">wt =</span> wthp.grid[,<span class="dv">1</span>], <span class="at">hp =</span> wthp.grid[,<span class="dv">2</span>])</span>
<span id="cb513-9"><a href="multiple-regression.html#cb513-9" tabindex="-1"></a></span>
<span id="cb513-10"><a href="multiple-regression.html#cb513-10" tabindex="-1"></a><span class="co"># Predicted mpg at each point in grid</span></span>
<span id="cb513-11"><a href="multiple-regression.html#cb513-11" tabindex="-1"></a>mpg.grid <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">predict</span>(fit.mpg, <span class="at">newdata =</span> grid.df), </span>
<span id="cb513-12"><a href="multiple-regression.html#cb513-12" tabindex="-1"></a>                 <span class="at">nrow =</span> grid.lines, <span class="at">ncol =</span> grid.lines)</span>
<span id="cb513-13"><a href="multiple-regression.html#cb513-13" tabindex="-1"></a></span>
<span id="cb513-14"><a href="multiple-regression.html#cb513-14" tabindex="-1"></a><span class="co"># Predicted mpg at observed </span></span>
<span id="cb513-15"><a href="multiple-regression.html#cb513-15" tabindex="-1"></a>mpg.predicted <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit.mpg)</span>
<span id="cb513-16"><a href="multiple-regression.html#cb513-16" tabindex="-1"></a></span>
<span id="cb513-17"><a href="multiple-regression.html#cb513-17" tabindex="-1"></a><span class="co"># scatter plot with regression plane</span></span>
<span id="cb513-18"><a href="multiple-regression.html#cb513-18" tabindex="-1"></a><span class="fu">points3D</span>(wt, hp, mpg, </span>
<span id="cb513-19"><a href="multiple-regression.html#cb513-19" tabindex="-1"></a>    <span class="at">pch =</span> <span class="dv">16</span>, <span class="at">theta =</span> <span class="dv">30</span>, <span class="at">phi =</span> <span class="dv">5</span>, </span>
<span id="cb513-20"><a href="multiple-regression.html#cb513-20" tabindex="-1"></a>    <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">alpha=</span><span class="fl">0.9</span>,</span>
<span id="cb513-21"><a href="multiple-regression.html#cb513-21" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Weight&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Horsepower&quot;</span>, <span class="at">zlab =</span> <span class="st">&quot;MPG&quot;</span>,  </span>
<span id="cb513-22"><a href="multiple-regression.html#cb513-22" tabindex="-1"></a>    <span class="at">surf =</span> <span class="fu">list</span>(<span class="at">x =</span> wt.grid,</span>
<span id="cb513-23"><a href="multiple-regression.html#cb513-23" tabindex="-1"></a>                <span class="at">y =</span> hp.grid, </span>
<span id="cb513-24"><a href="multiple-regression.html#cb513-24" tabindex="-1"></a>                <span class="at">z =</span> mpg.grid, </span>
<span id="cb513-25"><a href="multiple-regression.html#cb513-25" tabindex="-1"></a>                <span class="at">facets =</span> <span class="cn">NA</span>, </span>
<span id="cb513-26"><a href="multiple-regression.html#cb513-26" tabindex="-1"></a>                <span class="at">fit =</span> mpg.predicted,</span>
<span id="cb513-27"><a href="multiple-regression.html#cb513-27" tabindex="-1"></a>                <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb513-28"><a href="multiple-regression.html#cb513-28" tabindex="-1"></a>    )</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-531-1.png" width="576" /></p>
</div>
<div id="interactive-3d-visualizations-using-opengl" class="section level2 hasAnchor" number="23.11">
<h2><span class="header-section-number">23.11</span> Interactive 3D Visualizations Using OpenGL<a href="multiple-regression.html#interactive-3d-visualizations-using-opengl" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The package <code>rgl</code> is another package that we can use for 3D visualization. <code>rgl</code> is powerful because it lets us create interactive plots we can rotate and zoom in/out on.</p>
<p>Once you’ve installed and loaded <code>rgl</code> try the following code.</p>
<div class="sourceCode" id="cb514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb514-1"><a href="multiple-regression.html#cb514-1" tabindex="-1"></a><span class="co"># create 3D scatter, using spheres to draw points</span></span>
<span id="cb514-2"><a href="multiple-regression.html#cb514-2" tabindex="-1"></a><span class="fu">plot3d</span>(wt, hp, mpg, </span>
<span id="cb514-3"><a href="multiple-regression.html#cb514-3" tabindex="-1"></a>       <span class="at">type =</span> <span class="st">&quot;s&quot;</span>, </span>
<span id="cb514-4"><a href="multiple-regression.html#cb514-4" tabindex="-1"></a>       <span class="at">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb514-5"><a href="multiple-regression.html#cb514-5" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb514-6"><a href="multiple-regression.html#cb514-6" tabindex="-1"></a></span>
<span id="cb514-7"><a href="multiple-regression.html#cb514-7" tabindex="-1"></a><span class="co"># only need to include this line if using in a markdown document</span></span>
<span id="cb514-8"><a href="multiple-regression.html#cb514-8" tabindex="-1"></a><span class="fu">rglwidget</span>() </span></code></pre></div>
<div class="figure">
<div id="rgl87074" style="width:576px;height:384px;" class="rglWebGL html-widget" role="img" aria-labelledby="rgl87074-aria"></div>
<script type="application/json" data-for="rgl87074">{"x":{"material":{"color":"#000000","alpha":1,"lit":true,"ambient":"#000000","specular":"#FFFFFF","emission":"#000000","shininess":50,"smooth":true,"front":"filled","back":"filled","size":3,"lwd":1,"fog":true,"point_antialias":false,"line_antialias":false,"texture":null,"textype":"rgb","texmode":"modulate","texmipmap":false,"texminfilter":"linear","texmagfilter":"linear","texenvmap":false,"depth_mask":true,"depth_test":"less","isTransparent":false,"polygon_offset":[0,0],"margin":"","floating":false,"tag":"","blend":["src_alpha","one_minus_src_alpha"]},"rootSubscene":6,"objects":{"13":{"id":13,"type":"spheres","material":{},"vertices":"0","colors":"1","radii":[[4.09920072555542]],"centers":"2","ignoreExtent":false,"fastTransparency":true,"flags":32771},"15":{"id":15,"type":"text","material":{"lit":false,"margin":0,"floating":true,"edge":[0,1,1]},"vertices":"3","colors":"4","texts":[["wt"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"5","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"16":{"id":16,"type":"text","material":{"lit":false,"margin":1,"floating":true,"edge":[1,1,1]},"vertices":"6","colors":"7","texts":[["hp"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"8","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"17":{"id":17,"type":"text","material":{"lit":false,"margin":2,"floating":true,"edge":[1,1,1]},"vertices":"9","colors":"10","texts":[["mpg"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"11","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"10":{"id":10,"type":"light","vertices":[[0,0,1]],"colors":[[1,1,1,1],[1,1,1,1],[1,1,1,1]],"viewpoint":true,"finite":false},"12":{"id":12,"type":"background","material":{"lit":false,"back":"lines"},"colors":"12","centers":"13","sphere":false,"fogtype":"none","fogscale":1,"flags":32768},"14":{"id":14,"type":"bboxdeco","material":{"front":"lines","back":"lines"},"vertices":"14","colors":"15","axes":{"mode":["pretty","pretty","pretty"],"step":[1,50,5],"nticks":[5,5,5],"marklen":[15,15,15],"expand":[1.029999971389771,1.029999971389771,1.029999971389771]},"draw_front":true,"flags":32769},"6":{"id":6,"type":"subscene","par3d":{"antialias":4,"FOV":30,"ignoreExtent":false,"listeners":6,"mouseMode":{"none":"none","left":"trackball","right":"zoom","middle":"fov","wheel":"pull"},"observer":[0,0,729.6987915039062],"modelMatrix":[[41.92441177368164,0,0,-145.4148254394531],[0,0.1981638669967651,6.556565284729004,-183.5726470947266],[0,-0.5444507598876953,2.386394739151001,-677.2062377929688],[0,0,0,1]],"projMatrix":[[3.732050657272339,0,0,0],[0,3.732050657272339,0,0],[0,0,-3.863703727722168,-2630.47998046875],[0,0,-1,0]],"skipRedraw":false,"userMatrix":[[1,0,0,0],[0,0.3420201433256682,0.9396926207859085,0],[0,-0.9396926207859085,0.3420201433256682,0],[0,0,0,1]],"userProjection":[[1,0,0,0],[0,1,0,0],[0,0,1,0],[0,0,0,1]],"scale":[41.92441177368164,0.5793923735618591,6.977351188659668],"viewport":{"x":0,"y":0,"width":1,"height":1},"zoom":1,"bbox":[1.415224075317383,5.521775722503662,44.92499923706055,342.0750122070312,9.812499046325684,34.48750305175781],"windowRect":[58,45,314,301],"family":"sans","font":1,"cex":1,"useFreeType":true,"fontname":"/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/rgl/fonts/FreeSans.ttf","maxClipPlanes":6,"glVersion":2.1,"activeSubscene":0},"embeddings":{"viewport":"replace","projection":"replace","model":"replace","mouse":"replace"},"objects":[12,14,13,15,16,17,10],"subscenes":[],"flags":34067}},"crosstalk":{"key":[],"group":[],"id":[],"options":[]},"width":576,"height":384,"buffer":{"accessors":[{"bufferView":0,"componentType":5126,"count":32,"type":"VEC3"},{"bufferView":1,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":2,"componentType":5126,"count":32,"type":"VEC3"},{"bufferView":3,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":4,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":5,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":6,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":7,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":8,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":9,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":10,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":11,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":12,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":13,"componentType":5121,"count":1,"type":"VEC3"},{"bufferView":14,"componentType":5126,"count":15,"type":"VEC3"},{"bufferView":15,"componentType":5121,"count":1,"type":"VEC4"}],"bufferViews":[{"buffer":0,"byteLength":384,"byteOffset":0},{"buffer":0,"byteLength":4,"byteOffset":384},{"buffer":0,"byteLength":384,"byteOffset":388},{"buffer":0,"byteLength":12,"byteOffset":772},{"buffer":0,"byteLength":4,"byteOffset":784},{"buffer":0,"byteLength":12,"byteOffset":788},{"buffer":0,"byteLength":12,"byteOffset":800},{"buffer":0,"byteLength":4,"byteOffset":812},{"buffer":0,"byteLength":12,"byteOffset":816},{"buffer":0,"byteLength":12,"byteOffset":828},{"buffer":0,"byteLength":4,"byteOffset":840},{"buffer":0,"byteLength":12,"byteOffset":844},{"buffer":0,"byteLength":4,"byteOffset":856},{"buffer":0,"byteLength":3,"byteOffset":860},{"buffer":0,"byteLength":180,"byteOffset":864},{"buffer":0,"byteLength":4,"byteOffset":1044}],"buffers":[{"byteLength":1048,"bytes":"FK4nQAAA3EIAAKhBAAA4QAAA3EIAAKhB4XoUQAAAukJmZrZBj8JNQAAA3EIzM6tB9ihcQAAA\nL0OamZVBpHBdQAAA0kLNzJBB4XpkQAAAdUPNzGRB9ihMQAAAeEIzM8NBmplJQAAAvkJmZrZB\n9ihcQAAA9kKamZlB9ihcQAAA9kJmZo5BcT2CQAAANEMzM4NBUrhuQAAANENmZopBhetxQAAA\nNEMzM3NBAACoQAAATUNmZiZBaJGtQAAAV0NmZiZBPQqrQAAAZkMzM2tBzcwMQAAAhEKamQFC\nUrjOPwAAUEIzM/NBSOHqPwAAgkKamQdCj8IdQAAAwkIAAKxBrkdhQAAAFkMAAHhBCtdbQAAA\nFkMzM3NBj8J1QAAAdUPNzFRBexR2QAAAL0OamZlBFK73PwAAhEJmZtpBw/UIQAAAtkIAANBB\n/KnBPwAA4kIzM/NBSOFKQAAAhEPNzHxBrkcxQAAAL0OamZ1B4XpkQACAp0MAAHBBhesxQAAA\n2kIzM6tBAQAAARSuJ0AAANxCAACoQQAAOEAAANxCAACoQeF6FEAAALpCZma2QY/CTUAAANxC\nMzOrQfYoXEAAAC9DmpmVQaRwXUAAANJCzcyQQeF6ZEAAAHVDzcxkQfYoTEAAAHhCMzPDQZqZ\nSUAAAL5CZma2QfYoXEAAAPZCmpmZQfYoXEAAAPZCZmaOQXE9gkAAADRDMzODQVK4bkAAADRD\nZmaKQYXrcUAAADRDMzNzQQAAqEAAAE1DZmYmQWiRrUAAAFdDZmYmQT0Kq0AAAGZDMzNrQc3M\nDEAAAIRCmpkBQlK4zj8AAFBCMzPzQUjh6j8AAIJCmpkHQo/CHUAAAMJCAACsQa5HYUAAABZD\nAAB4QQrXW0AAABZDMzNzQY/CdUAAAHVDzcxUQXsUdkAAAC9DmpmZQRSu9z8AAIRCZmbaQcP1\nCEAAALZCAADQQfypwT8AAOJCMzPzQUjhSkAAAIRDzcx8Qa5HMUAAAC9DmpmdQeF6ZEAAgKdD\nAABwQYXrMUAAANpCMzOrQQAAwH8AAIBAAACAPwAAAAEAAMB/AACAQAAAgD8AAMB/AACAQAAA\ngD8AAAABAADAfwAAgEAAAIA/AADAfwAAgEAAAIA/AAAAAQAAwH8AAIBAAACAPwEBAQEAAAAA\nAAAAQAAAwH8AAMB/AABAQAAAwH8AAMB/AACAQAAAwH8AAMB/AACgQAAAwH8AAMB/AADAfwAA\nSEIAAMB/AADAfwAAyEIAAMB/AADAfwAAFkMAAMB/AADAfwAASEMAAMB/AADAfwAAekMAAMB/\nAADAfwAAlkMAAMB/AADAfwAAwH8AACBBAADAfwAAwH8AAHBBAADAfwAAwH8AAKBBAADAfwAA\nwH8AAMhBAADAfwAAwH8AAPBBAAAAAQ=="}]},"context":{"shiny":false,"rmarkdown":"bookdown::gitbook"},"vertexShader":"#line 2 1\n// File 1 is the vertex shader\n#ifdef GL_ES\n#ifdef GL_FRAGMENT_PRECISION_HIGH\nprecision highp float;\n#else\nprecision mediump float;\n#endif\n#endif\n\nattribute vec3 aPos;\nattribute vec4 aCol;\nuniform mat4 mvMatrix;\nuniform mat4 prMatrix;\nvarying vec4 vCol;\nvarying vec4 vPosition;\n\n#ifdef NEEDS_VNORMAL\nattribute vec3 aNorm;\nuniform mat4 normMatrix;\nvarying vec4 vNormal;\n#endif\n\n#if defined(HAS_TEXTURE) || defined (IS_TEXT)\nattribute vec2 aTexcoord;\nvarying vec2 vTexcoord;\n#endif\n\n#ifdef FIXED_SIZE\nuniform vec3 textScale;\n#endif\n\n#ifdef FIXED_QUADS\nattribute vec3 aOfs;\n#endif\n\n#ifdef IS_TWOSIDED\n#ifdef HAS_NORMALS\nvarying float normz;\nuniform mat4 invPrMatrix;\n#else\nattribute vec3 aPos1;\nattribute vec3 aPos2;\nvarying float normz;\n#endif\n#endif // IS_TWOSIDED\n\n#ifdef FAT_LINES\nattribute vec3 aNext;\nattribute vec2 aPoint;\nvarying vec2 vPoint;\nvarying float vLength;\nuniform float uAspect;\nuniform float uLwd;\n#endif\n\n#ifdef USE_ENVMAP\nvarying vec3 vReflection;\n#endif\n\nvoid main(void) {\n  \n#ifndef IS_BRUSH\n#if defined(NCLIPPLANES) || !defined(FIXED_QUADS) || defined(HAS_FOG) || defined(USE_ENVMAP)\n  vPosition = mvMatrix * vec4(aPos, 1.);\n#endif\n  \n#ifndef FIXED_QUADS\n  gl_Position = prMatrix * vPosition;\n#endif\n#endif // !IS_BRUSH\n  \n#ifdef IS_POINTS\n  gl_PointSize = POINTSIZE;\n#endif\n  \n  vCol = aCol;\n  \n// USE_ENVMAP implies NEEDS_VNORMAL\n\n#ifdef NEEDS_VNORMAL\n  vNormal = normMatrix * vec4(-aNorm, dot(aNorm, aPos));\n#endif\n\n#ifdef USE_ENVMAP\n  vReflection = normalize(reflect(vPosition.xyz/vPosition.w, \n                        normalize(vNormal.xyz/vNormal.w)));\n#endif\n  \n#ifdef IS_TWOSIDED\n#ifdef HAS_NORMALS\n  /* normz should be calculated *after* projection */\n  normz = (invPrMatrix*vNormal).z;\n#else\n  vec4 pos1 = prMatrix*(mvMatrix*vec4(aPos1, 1.));\n  pos1 = pos1/pos1.w - gl_Position/gl_Position.w;\n  vec4 pos2 = prMatrix*(mvMatrix*vec4(aPos2, 1.));\n  pos2 = pos2/pos2.w - gl_Position/gl_Position.w;\n  normz = pos1.x*pos2.y - pos1.y*pos2.x;\n#endif\n#endif // IS_TWOSIDED\n  \n#ifdef NEEDS_VNORMAL\n  vNormal = vec4(normalize(vNormal.xyz), 1);\n#endif\n  \n#if defined(HAS_TEXTURE) || defined(IS_TEXT)\n  vTexcoord = aTexcoord;\n#endif\n  \n#if defined(FIXED_SIZE) && !defined(ROTATING)\n  vec4 pos = prMatrix * mvMatrix * vec4(aPos, 1.);\n  pos = pos/pos.w;\n  gl_Position = pos + vec4(aOfs*textScale, 0.);\n#endif\n  \n#if defined(IS_SPRITES) && !defined(FIXED_SIZE)\n  vec4 pos = mvMatrix * vec4(aPos, 1.);\n  pos = pos/pos.w + vec4(aOfs,  0.);\n  gl_Position = prMatrix*pos;\n#endif\n  \n#ifdef FAT_LINES\n  /* This code was inspired by Matt Deslauriers' code in \n   https://mattdesl.svbtle.com/drawing-lines-is-hard */\n  vec2 aspectVec = vec2(uAspect, 1.0);\n  mat4 projViewModel = prMatrix * mvMatrix;\n  vec4 currentProjected = projViewModel * vec4(aPos, 1.0);\n  currentProjected = currentProjected/currentProjected.w;\n  vec4 nextProjected = projViewModel * vec4(aNext, 1.0);\n  vec2 currentScreen = currentProjected.xy * aspectVec;\n  vec2 nextScreen = (nextProjected.xy / nextProjected.w) * aspectVec;\n  float len = uLwd;\n  vec2 dir = vec2(1.0, 0.0);\n  vPoint = aPoint;\n  vLength = length(nextScreen - currentScreen)/2.0;\n  vLength = vLength/(vLength + len);\n  if (vLength > 0.0) {\n    dir = normalize(nextScreen - currentScreen);\n  }\n  vec2 normal = vec2(-dir.y, dir.x);\n  dir.x /= uAspect;\n  normal.x /= uAspect;\n  vec4 offset = vec4(len*(normal*aPoint.x*aPoint.y - dir), 0.0, 0.0);\n  gl_Position = currentProjected + offset;\n#endif\n  \n#ifdef IS_BRUSH\n  gl_Position = vec4(aPos, 1.);\n#endif\n}","fragmentShader":"#line 2 2\n// File 2 is the fragment shader\n#ifdef GL_ES\n#ifdef GL_FRAGMENT_PRECISION_HIGH\nprecision highp float;\n#else\nprecision mediump float;\n#endif\n#endif\nvarying vec4 vCol; // carries alpha\nvarying vec4 vPosition;\n#if defined(HAS_TEXTURE) || defined (IS_TEXT)\nvarying vec2 vTexcoord;\nuniform sampler2D uSampler;\n#endif\n\n#ifdef HAS_FOG\nuniform int uFogMode;\nuniform vec3 uFogColor;\nuniform vec4 uFogParms;\n#endif\n\n#if defined(IS_LIT) && !defined(FIXED_QUADS)\nvarying vec4 vNormal;\n#endif\n\n#if NCLIPPLANES > 0\nuniform vec4 vClipplane[NCLIPPLANES];\n#endif\n\n#if NLIGHTS > 0\nuniform mat4 mvMatrix;\n#endif\n\n#ifdef IS_LIT\nuniform vec3 emission;\nuniform float shininess;\n#if NLIGHTS > 0\nuniform vec3 ambient[NLIGHTS];\nuniform vec3 specular[NLIGHTS]; // light*material\nuniform vec3 diffuse[NLIGHTS];\nuniform vec3 lightDir[NLIGHTS];\nuniform bool viewpoint[NLIGHTS];\nuniform bool finite[NLIGHTS];\n#endif\n#endif // IS_LIT\n\n#ifdef IS_TWOSIDED\nuniform bool front;\nvarying float normz;\n#endif\n\n#ifdef FAT_LINES\nvarying vec2 vPoint;\nvarying float vLength;\n#endif\n\n#ifdef USE_ENVMAP\nvarying vec3 vReflection;\n#endif\n\nvoid main(void) {\n  vec4 fragColor;\n#ifdef FAT_LINES\n  vec2 point = vPoint;\n  bool neg = point.y < 0.0;\n  point.y = neg ? (point.y + vLength)/(1.0 - vLength) :\n                 -(point.y - vLength)/(1.0 - vLength);\n#if defined(IS_TRANSPARENT) && defined(IS_LINESTRIP)\n  if (neg && length(point) <= 1.0) discard;\n#endif\n  point.y = min(point.y, 0.0);\n  if (length(point) > 1.0) discard;\n#endif // FAT_LINES\n  \n#ifdef ROUND_POINTS\n  vec2 coord = gl_PointCoord - vec2(0.5);\n  if (length(coord) > 0.5) discard;\n#endif\n  \n#if NCLIPPLANES > 0\n  for (int i = 0; i < NCLIPPLANES; i++)\n    if (dot(vPosition, vClipplane[i]) < 0.0) discard;\n#endif\n    \n#ifdef FIXED_QUADS\n    vec3 n = vec3(0., 0., 1.);\n#elif defined(IS_LIT)\n    vec3 n = normalize(vNormal.xyz);\n#endif\n    \n#ifdef IS_TWOSIDED\n    if ((normz <= 0.) != front) discard;\n#endif\n\n#ifdef IS_LIT\n    vec3 eye = normalize(-vPosition.xyz/vPosition.w);\n    vec3 lightdir;\n    vec4 colDiff;\n    vec3 halfVec;\n    vec4 lighteffect = vec4(emission, 0.);\n    vec3 col;\n    float nDotL;\n#ifdef FIXED_QUADS\n    n = -faceforward(n, n, eye);\n#endif\n    \n#if NLIGHTS > 0\n    // Simulate two-sided lighting\n    if (n.z < 0.0)\n      n = -n;\n    for (int i=0;i<NLIGHTS;i++) {\n      colDiff = vec4(vCol.rgb * diffuse[i], vCol.a);\n      lightdir = lightDir[i];\n      if (!viewpoint[i]) {\n        if (finite[i]) {\n          lightdir = (mvMatrix * vec4(lightdir, 1.)).xyz;\n        } else {\n          lightdir = (mvMatrix * vec4(lightdir, 0.)).xyz;\n        }\n      }\n      if (!finite[i]) {\n        halfVec = normalize(lightdir + eye);\n      } else {\n        lightdir = normalize(lightdir - vPosition.xyz/vPosition.w);\n        halfVec = normalize(lightdir + eye);\n      }\n      col = ambient[i];\n      nDotL = dot(n, lightdir);\n      col = col + max(nDotL, 0.) * colDiff.rgb;\n      col = col + pow(max(dot(halfVec, n), 0.), shininess) * specular[i];\n      lighteffect = lighteffect + vec4(col, colDiff.a);\n    }\n#endif\n    \n#else // not IS_LIT\n    vec4 colDiff = vCol;\n    vec4 lighteffect = colDiff;\n#endif\n    \n#ifdef IS_TEXT\n    vec4 textureColor = lighteffect*texture2D(uSampler, vTexcoord);\n#endif\n    \n#ifdef HAS_TEXTURE\n\n// These calculations use the definitions from \n// https://docs.gl/gl3/glTexEnv\n\n#ifdef USE_ENVMAP\n    float m = 2.0 * sqrt(dot(vReflection, vReflection) + 2.0*vReflection.z + 1.0);\n    vec4 textureColor = texture2D(uSampler, vReflection.xy / m + vec2(0.5, 0.5));\n#else\n    vec4 textureColor = texture2D(uSampler, vTexcoord);\n#endif\n\n#ifdef TEXTURE_rgb\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(textureColor.rgb, lighteffect.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = lighteffect*vec4(textureColor.rgb, 1.);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - textureColor.rgb) * lighteffect.rgb, lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(lighteffect.rgb + textureColor.rgb, lighteffect.a);\n#endif\n\n#endif //TEXTURE_rgb\n        \n#ifdef TEXTURE_rgba\n\n#ifdef TEXMODE_replace\n// already done\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = lighteffect*textureColor;\n#endif\n\n#ifdef TEXMODE_decal\n    textureColor = vec4((1. - textureColor.a)*lighteffect.rgb) +\n                     textureColor.a*textureColor.rgb, \n                     lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - textureColor.rgb) * lighteffect.rgb,\n                    lighteffect.a*textureColor.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(lighteffect.rgb + textureColor.rgb,\n                    lighteffect.a*textureColor.a);\n#endif\n    \n#endif //TEXTURE_rgba\n    \n#ifdef TEXTURE_alpha\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(lighteffect.rgb, luminance);\n#endif \n\n#if defined(TEXMODE_modulate) || defined(TEXMODE_blend) || defined(TEXMODE_add)\n    textureColor = vec4(lighteffect.rgb, lighteffect.a*luminance);\n#endif\n \n#endif // TEXTURE_alpha\n    \n// The TEXTURE_luminance values are not from that reference    \n#ifdef TEXTURE_luminance\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(luminance, luminance, luminance, lighteffect.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = vec4(luminance*lighteffect.rgb, lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - luminance)*lighteffect.rgb,\n                        lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(luminance + lighteffect.rgb, lighteffect.a);\n#endif\n\n#endif // TEXTURE_luminance\n \n    \n#ifdef TEXTURE_luminance_alpha\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(luminance, luminance, luminance, textureColor.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = vec4(luminance*lighteffect.rgb, \n                        textureColor.a*lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - luminance)*lighteffect.rgb,\n                        textureColor.a*lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(luminance + lighteffect.rgb, \n                        textureColor.a*lighteffect.a);\n\n#endif\n\n#endif // TEXTURE_luminance_alpha\n    \n    fragColor = textureColor;\n\n#elif defined(IS_TEXT)\n    if (textureColor.a < 0.1)\n      discard;\n    else\n      fragColor = textureColor;\n#else\n    fragColor = lighteffect;\n#endif // HAS_TEXTURE\n    \n#ifdef HAS_FOG\n    // uFogParms elements: x = near, y = far, z = fogscale, w = (1-sin(FOV/2))/(1+sin(FOV/2))\n    // In Exp and Exp2: use density = density/far\n    // fogF will be the proportion of fog\n    // Initialize it to the linear value\n    float fogF;\n    if (uFogMode > 0) {\n      fogF = (uFogParms.y - vPosition.z/vPosition.w)/(uFogParms.y - uFogParms.x);\n      if (uFogMode > 1)\n        fogF = mix(uFogParms.w, 1.0, fogF);\n      fogF = fogF*uFogParms.z;\n      if (uFogMode == 2)\n        fogF = 1.0 - exp(-fogF);\n      // Docs are wrong: use (density*c)^2, not density*c^2\n      // https://gitlab.freedesktop.org/mesa/mesa/-/blob/master/src/mesa/swrast/s_fog.c#L58\n      else if (uFogMode == 3)\n        fogF = 1.0 - exp(-fogF*fogF);\n      fogF = clamp(fogF, 0.0, 1.0);\n      gl_FragColor = vec4(mix(fragColor.rgb, uFogColor, fogF), fragColor.a);\n    } else gl_FragColor = fragColor;\n#else\n    gl_FragColor = fragColor;\n#endif // HAS_FOG\n    \n}","players":[],"webGLoptions":{"preserveDrawingBuffer":true},"fastTransparency":true},"evals":[],"jsHooks":[]}</script>
<p id="rgl87074-aria" hidden>
</p>
</div>
<p>We can add a 3d plane to our plot, representing the multiple regression model, with the <code>rgl.planes()</code> function as shown below.</p>
<div class="sourceCode" id="cb515"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb515-1"><a href="multiple-regression.html#cb515-1" tabindex="-1"></a>coefs <span class="ot">&lt;-</span> <span class="fu">coef</span>(fit.mpg)</span>
<span id="cb515-2"><a href="multiple-regression.html#cb515-2" tabindex="-1"></a>b1 <span class="ot">&lt;-</span> coefs[<span class="st">&quot;wt&quot;</span>]</span>
<span id="cb515-3"><a href="multiple-regression.html#cb515-3" tabindex="-1"></a>b2 <span class="ot">&lt;-</span> coefs[<span class="st">&quot;hp&quot;</span>]</span>
<span id="cb515-4"><a href="multiple-regression.html#cb515-4" tabindex="-1"></a>c <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb515-5"><a href="multiple-regression.html#cb515-5" tabindex="-1"></a>a <span class="ot">&lt;-</span> coefs[<span class="st">&quot;(Intercept)&quot;</span>]</span>
<span id="cb515-6"><a href="multiple-regression.html#cb515-6" tabindex="-1"></a><span class="fu">plot3d</span>(wt, hp, mpg, </span>
<span id="cb515-7"><a href="multiple-regression.html#cb515-7" tabindex="-1"></a>       <span class="at">type =</span> <span class="st">&quot;s&quot;</span>, </span>
<span id="cb515-8"><a href="multiple-regression.html#cb515-8" tabindex="-1"></a>       <span class="at">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb515-9"><a href="multiple-regression.html#cb515-9" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb515-10"><a href="multiple-regression.html#cb515-10" tabindex="-1"></a><span class="fu">rgl.planes</span>(b1, b2, c, a, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>)</span>
<span id="cb515-11"><a href="multiple-regression.html#cb515-11" tabindex="-1"></a><span class="fu">rglwidget</span>()</span></code></pre></div>
<div class="figure">
<div id="rgl35284" style="width:576px;height:384px;" class="rglWebGL html-widget" role="img" aria-labelledby="rgl35284-aria"></div>
<script type="application/json" data-for="rgl35284">{"x":{"material":{"color":"#BEBEBE","alpha":0.8980392217636108,"lit":true,"ambient":"#000000","specular":"#FFFFFF","emission":"#000000","shininess":50,"smooth":true,"front":"filled","back":"filled","size":3,"lwd":1,"fog":true,"point_antialias":false,"line_antialias":false,"texture":null,"textype":"rgb","texmode":"modulate","texmipmap":false,"texminfilter":"linear","texmagfilter":"linear","texenvmap":false,"depth_mask":true,"depth_test":"less","isTransparent":true,"polygon_offset":[0,0],"margin":"","floating":false,"tag":"","blend":["src_alpha","one_minus_src_alpha"]},"rootSubscene":6,"objects":{"25":{"id":25,"type":"spheres","material":{"alpha":1,"isTransparent":false},"vertices":"0","colors":"1","radii":[[4.09920072555542]],"centers":"2","ignoreExtent":false,"fastTransparency":true,"flags":32771},"27":{"id":27,"type":"text","material":{"alpha":1,"lit":false,"isTransparent":false,"margin":0,"floating":true,"edge":[0,1,1]},"vertices":"3","colors":"4","texts":[["wt"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"5","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"28":{"id":28,"type":"text","material":{"alpha":1,"lit":false,"isTransparent":false,"margin":1,"floating":true,"edge":[1,1,1]},"vertices":"6","colors":"7","texts":[["hp"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"8","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"29":{"id":29,"type":"text","material":{"alpha":1,"lit":false,"isTransparent":false,"margin":2,"floating":true,"edge":[1,1,1]},"vertices":"9","colors":"10","texts":[["mpg"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"11","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"30":{"id":30,"type":"planes","material":{},"vertices":"12","colors":"14","offsets":[[37.22726821899414]],"centers":"15","normals":"13","ignoreExtent":true,"flags":32803},"10":{"id":10,"type":"light","vertices":[[0,0,1]],"colors":[[1,1,1,1],[1,1,1,1],[1,1,1,1]],"viewpoint":true,"finite":false},"12":{"id":12,"type":"background","material":{"alpha":1,"lit":false,"back":"lines","isTransparent":false},"colors":"16","centers":"17","sphere":false,"fogtype":"none","fogscale":1,"flags":32768},"26":{"id":26,"type":"bboxdeco","material":{"alpha":1,"front":"lines","back":"lines","isTransparent":false},"vertices":"18","colors":"19","axes":{"mode":["pretty","pretty","pretty"],"step":[1,50,5],"nticks":[5,5,5],"marklen":[15,15,15],"expand":[1.029999971389771,1.029999971389771,1.029999971389771]},"draw_front":true,"flags":32769},"6":{"id":6,"type":"subscene","par3d":{"antialias":4,"FOV":30,"ignoreExtent":false,"listeners":6,"mouseMode":{"none":"none","left":"trackball","right":"zoom","middle":"fov","wheel":"pull"},"observer":[0,0,729.7014770507812],"modelMatrix":[[41.92483901977539,0,0,-145.4163055419922],[0,0.1981638520956039,6.556576251983643,-183.5728759765625],[0,-0.5444507002830505,2.386398553848267,-677.208984375],[0,0,0,1]],"projMatrix":[[3.732050895690918,0,0,0],[0,3.732050895690918,0,0],[0,0,-3.863703012466431,-2630.4892578125],[0,0,-1,0]],"skipRedraw":false,"userMatrix":[[1,0,0,0],[0,0.3420201433256682,0.9396926207859085,0],[0,-0.9396926207859085,0.3420201433256682,0],[0,0,0,1]],"userProjection":[[1,0,0,0],[0,1,0,0],[0,0,1,0],[0,0,0,1]],"scale":[41.92483901977539,0.5793923139572144,6.977362632751465],"viewport":{"x":0,"y":0,"width":1,"height":1},"zoom":1,"bbox":[1.415225028991699,5.521774768829346,44.92499923706055,342.0750122070312,9.8125,34.48750305175781],"windowRect":[58,45,314,301],"family":"sans","font":1,"cex":1,"useFreeType":true,"fontname":"/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/rgl/fonts/FreeSans.ttf","maxClipPlanes":6,"glVersion":2.1,"activeSubscene":0},"embeddings":{"viewport":"replace","projection":"replace","model":"replace","mouse":"replace"},"objects":[12,26,25,27,28,29,30,10],"subscenes":[],"flags":34099}},"crosstalk":{"key":[],"group":[],"id":[],"options":[]},"width":576,"height":384,"buffer":{"accessors":[{"bufferView":0,"componentType":5126,"count":32,"type":"VEC3"},{"bufferView":1,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":2,"componentType":5126,"count":32,"type":"VEC3"},{"bufferView":3,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":4,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":5,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":6,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":7,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":8,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":9,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":10,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":11,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":12,"componentType":5126,"count":3,"type":"VEC3"},{"bufferView":13,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":14,"componentType":5126,"count":1,"type":"VEC4"},{"bufferView":15,"componentType":5126,"count":4,"type":"VEC3"},{"bufferView":16,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":17,"componentType":5121,"count":1,"type":"VEC3"},{"bufferView":18,"componentType":5126,"count":15,"type":"VEC3"},{"bufferView":19,"componentType":5121,"count":1,"type":"VEC4"}],"bufferViews":[{"buffer":0,"byteLength":384,"byteOffset":0},{"buffer":0,"byteLength":4,"byteOffset":384},{"buffer":0,"byteLength":384,"byteOffset":388},{"buffer":0,"byteLength":12,"byteOffset":772},{"buffer":0,"byteLength":4,"byteOffset":784},{"buffer":0,"byteLength":12,"byteOffset":788},{"buffer":0,"byteLength":12,"byteOffset":800},{"buffer":0,"byteLength":4,"byteOffset":812},{"buffer":0,"byteLength":12,"byteOffset":816},{"buffer":0,"byteLength":12,"byteOffset":828},{"buffer":0,"byteLength":4,"byteOffset":840},{"buffer":0,"byteLength":12,"byteOffset":844},{"buffer":0,"byteLength":36,"byteOffset":856},{"buffer":0,"byteLength":12,"byteOffset":892},{"buffer":0,"byteLength":16,"byteOffset":904},{"buffer":0,"byteLength":48,"byteOffset":920},{"buffer":0,"byteLength":4,"byteOffset":968},{"buffer":0,"byteLength":3,"byteOffset":972},{"buffer":0,"byteLength":180,"byteOffset":976},{"buffer":0,"byteLength":4,"byteOffset":1156}],"buffers":[{"byteLength":1160,"bytes":"FK4nQAAA3EIAAKhBAAA4QAAA3EIAAKhB4XoUQAAAukJmZrZBj8JNQAAA3EIzM6tB9ihcQAAA\nL0OamZVBpHBdQAAA0kLNzJBB4XpkQAAAdUPNzGRB9ihMQAAAeEIzM8NBmplJQAAAvkJmZrZB\n9ihcQAAA9kKamZlB9ihcQAAA9kJmZo5BcT2CQAAANEMzM4NBUrhuQAAANENmZopBhetxQAAA\nNEMzM3NBAACoQAAATUNmZiZBaJGtQAAAV0NmZiZBPQqrQAAAZkMzM2tBzcwMQAAAhEKamQFC\nUrjOPwAAUEIzM/NBSOHqPwAAgkKamQdCj8IdQAAAwkIAAKxBrkdhQAAAFkMAAHhBCtdbQAAA\nFkMzM3NBj8J1QAAAdUPNzFRBexR2QAAAL0OamZlBFK73PwAAhEJmZtpBw/UIQAAAtkIAANBB\n/KnBPwAA4kIzM/NBSOFKQAAAhEPNzHxBrkcxQAAAL0OamZ1B4XpkQACAp0MAAHBBhesxQAAA\n2kIzM6tBAQAAARSuJ0AAANxCAACoQQAAOEAAANxCAACoQeF6FEAAALpCZma2QY/CTUAAANxC\nMzOrQfYoXEAAAC9DmpmVQaRwXUAAANJCzcyQQeF6ZEAAAHVDzcxkQfYoTEAAAHhCMzPDQZqZ\nSUAAAL5CZma2QfYoXEAAAPZCmpmZQfYoXEAAAPZCZmaOQXE9gkAAADRDMzODQVK4bkAAADRD\nZmaKQYXrcUAAADRDMzNzQQAAqEAAAE1DZmYmQWiRrUAAAFdDZmYmQT0Kq0AAAGZDMzNrQc3M\nDEAAAIRCmpkBQlK4zj8AAFBCMzPzQUjh6j8AAIJCmpkHQo/CHUAAAMJCAACsQa5HYUAAABZD\nAAB4QQrXW0AAABZDMzNzQY/CdUAAAHVDzcxUQXsUdkAAAC9DmpmZQRSu9z8AAIRCZmbaQcP1\nCEAAALZCAADQQfypwT8AAOJCMzPzQUjhSkAAAIRDzcx8Qa5HMUAAAC9DmpmdQeF6ZEAAgKdD\nAABwQYXrMUAAANpCMzOrQQAAwH8AAIBAAACAPwAAAAEAAMB/AACAQAAAgD8AAMB/AACAQAAA\ngD8AAAABAADAfwAAgEAAAIA/AADAfwAAgEAAAIA/AAAAAQAAwH8AAIBAAACAPxgmtT8zszNC\ns37yQRgmtT+aCatD2vamQduJiECaCatDAAAdQWEueMBaJAK9AACAv7++Pj+/vj4/v74+P+bl\nZT+aaBdAZwZzQy+nokHXA29AaPg/QzwqhUHD5IRAN9e5QlddkUEAAMB/AADAfwAAwH8BAQEB\nAAAAAAAAAEAAAMB/AADAfwAAQEAAAMB/AADAfwAAgEAAAMB/AADAfwAAoEAAAMB/AADAfwAA\nwH8AAEhCAADAfwAAwH8AAMhCAADAfwAAwH8AABZDAADAfwAAwH8AAEhDAADAfwAAwH8AAHpD\nAADAfwAAwH8AAJZDAADAfwAAwH8AAMB/AAAgQQAAwH8AAMB/AABwQQAAwH8AAMB/AACgQQAA\nwH8AAMB/AADIQQAAwH8AAMB/AADwQQAAAAE="}]},"context":{"shiny":false,"rmarkdown":"bookdown::gitbook"},"vertexShader":"#line 2 1\n// File 1 is the vertex shader\n#ifdef GL_ES\n#ifdef GL_FRAGMENT_PRECISION_HIGH\nprecision highp float;\n#else\nprecision mediump float;\n#endif\n#endif\n\nattribute vec3 aPos;\nattribute vec4 aCol;\nuniform mat4 mvMatrix;\nuniform mat4 prMatrix;\nvarying vec4 vCol;\nvarying vec4 vPosition;\n\n#ifdef NEEDS_VNORMAL\nattribute vec3 aNorm;\nuniform mat4 normMatrix;\nvarying vec4 vNormal;\n#endif\n\n#if defined(HAS_TEXTURE) || defined (IS_TEXT)\nattribute vec2 aTexcoord;\nvarying vec2 vTexcoord;\n#endif\n\n#ifdef FIXED_SIZE\nuniform vec3 textScale;\n#endif\n\n#ifdef FIXED_QUADS\nattribute vec3 aOfs;\n#endif\n\n#ifdef IS_TWOSIDED\n#ifdef HAS_NORMALS\nvarying float normz;\nuniform mat4 invPrMatrix;\n#else\nattribute vec3 aPos1;\nattribute vec3 aPos2;\nvarying float normz;\n#endif\n#endif // IS_TWOSIDED\n\n#ifdef FAT_LINES\nattribute vec3 aNext;\nattribute vec2 aPoint;\nvarying vec2 vPoint;\nvarying float vLength;\nuniform float uAspect;\nuniform float uLwd;\n#endif\n\n#ifdef USE_ENVMAP\nvarying vec3 vReflection;\n#endif\n\nvoid main(void) {\n  \n#ifndef IS_BRUSH\n#if defined(NCLIPPLANES) || !defined(FIXED_QUADS) || defined(HAS_FOG) || defined(USE_ENVMAP)\n  vPosition = mvMatrix * vec4(aPos, 1.);\n#endif\n  \n#ifndef FIXED_QUADS\n  gl_Position = prMatrix * vPosition;\n#endif\n#endif // !IS_BRUSH\n  \n#ifdef IS_POINTS\n  gl_PointSize = POINTSIZE;\n#endif\n  \n  vCol = aCol;\n  \n// USE_ENVMAP implies NEEDS_VNORMAL\n\n#ifdef NEEDS_VNORMAL\n  vNormal = normMatrix * vec4(-aNorm, dot(aNorm, aPos));\n#endif\n\n#ifdef USE_ENVMAP\n  vReflection = normalize(reflect(vPosition.xyz/vPosition.w, \n                        normalize(vNormal.xyz/vNormal.w)));\n#endif\n  \n#ifdef IS_TWOSIDED\n#ifdef HAS_NORMALS\n  /* normz should be calculated *after* projection */\n  normz = (invPrMatrix*vNormal).z;\n#else\n  vec4 pos1 = prMatrix*(mvMatrix*vec4(aPos1, 1.));\n  pos1 = pos1/pos1.w - gl_Position/gl_Position.w;\n  vec4 pos2 = prMatrix*(mvMatrix*vec4(aPos2, 1.));\n  pos2 = pos2/pos2.w - gl_Position/gl_Position.w;\n  normz = pos1.x*pos2.y - pos1.y*pos2.x;\n#endif\n#endif // IS_TWOSIDED\n  \n#ifdef NEEDS_VNORMAL\n  vNormal = vec4(normalize(vNormal.xyz), 1);\n#endif\n  \n#if defined(HAS_TEXTURE) || defined(IS_TEXT)\n  vTexcoord = aTexcoord;\n#endif\n  \n#if defined(FIXED_SIZE) && !defined(ROTATING)\n  vec4 pos = prMatrix * mvMatrix * vec4(aPos, 1.);\n  pos = pos/pos.w;\n  gl_Position = pos + vec4(aOfs*textScale, 0.);\n#endif\n  \n#if defined(IS_SPRITES) && !defined(FIXED_SIZE)\n  vec4 pos = mvMatrix * vec4(aPos, 1.);\n  pos = pos/pos.w + vec4(aOfs,  0.);\n  gl_Position = prMatrix*pos;\n#endif\n  \n#ifdef FAT_LINES\n  /* This code was inspired by Matt Deslauriers' code in \n   https://mattdesl.svbtle.com/drawing-lines-is-hard */\n  vec2 aspectVec = vec2(uAspect, 1.0);\n  mat4 projViewModel = prMatrix * mvMatrix;\n  vec4 currentProjected = projViewModel * vec4(aPos, 1.0);\n  currentProjected = currentProjected/currentProjected.w;\n  vec4 nextProjected = projViewModel * vec4(aNext, 1.0);\n  vec2 currentScreen = currentProjected.xy * aspectVec;\n  vec2 nextScreen = (nextProjected.xy / nextProjected.w) * aspectVec;\n  float len = uLwd;\n  vec2 dir = vec2(1.0, 0.0);\n  vPoint = aPoint;\n  vLength = length(nextScreen - currentScreen)/2.0;\n  vLength = vLength/(vLength + len);\n  if (vLength > 0.0) {\n    dir = normalize(nextScreen - currentScreen);\n  }\n  vec2 normal = vec2(-dir.y, dir.x);\n  dir.x /= uAspect;\n  normal.x /= uAspect;\n  vec4 offset = vec4(len*(normal*aPoint.x*aPoint.y - dir), 0.0, 0.0);\n  gl_Position = currentProjected + offset;\n#endif\n  \n#ifdef IS_BRUSH\n  gl_Position = vec4(aPos, 1.);\n#endif\n}","fragmentShader":"#line 2 2\n// File 2 is the fragment shader\n#ifdef GL_ES\n#ifdef GL_FRAGMENT_PRECISION_HIGH\nprecision highp float;\n#else\nprecision mediump float;\n#endif\n#endif\nvarying vec4 vCol; // carries alpha\nvarying vec4 vPosition;\n#if defined(HAS_TEXTURE) || defined (IS_TEXT)\nvarying vec2 vTexcoord;\nuniform sampler2D uSampler;\n#endif\n\n#ifdef HAS_FOG\nuniform int uFogMode;\nuniform vec3 uFogColor;\nuniform vec4 uFogParms;\n#endif\n\n#if defined(IS_LIT) && !defined(FIXED_QUADS)\nvarying vec4 vNormal;\n#endif\n\n#if NCLIPPLANES > 0\nuniform vec4 vClipplane[NCLIPPLANES];\n#endif\n\n#if NLIGHTS > 0\nuniform mat4 mvMatrix;\n#endif\n\n#ifdef IS_LIT\nuniform vec3 emission;\nuniform float shininess;\n#if NLIGHTS > 0\nuniform vec3 ambient[NLIGHTS];\nuniform vec3 specular[NLIGHTS]; // light*material\nuniform vec3 diffuse[NLIGHTS];\nuniform vec3 lightDir[NLIGHTS];\nuniform bool viewpoint[NLIGHTS];\nuniform bool finite[NLIGHTS];\n#endif\n#endif // IS_LIT\n\n#ifdef IS_TWOSIDED\nuniform bool front;\nvarying float normz;\n#endif\n\n#ifdef FAT_LINES\nvarying vec2 vPoint;\nvarying float vLength;\n#endif\n\n#ifdef USE_ENVMAP\nvarying vec3 vReflection;\n#endif\n\nvoid main(void) {\n  vec4 fragColor;\n#ifdef FAT_LINES\n  vec2 point = vPoint;\n  bool neg = point.y < 0.0;\n  point.y = neg ? (point.y + vLength)/(1.0 - vLength) :\n                 -(point.y - vLength)/(1.0 - vLength);\n#if defined(IS_TRANSPARENT) && defined(IS_LINESTRIP)\n  if (neg && length(point) <= 1.0) discard;\n#endif\n  point.y = min(point.y, 0.0);\n  if (length(point) > 1.0) discard;\n#endif // FAT_LINES\n  \n#ifdef ROUND_POINTS\n  vec2 coord = gl_PointCoord - vec2(0.5);\n  if (length(coord) > 0.5) discard;\n#endif\n  \n#if NCLIPPLANES > 0\n  for (int i = 0; i < NCLIPPLANES; i++)\n    if (dot(vPosition, vClipplane[i]) < 0.0) discard;\n#endif\n    \n#ifdef FIXED_QUADS\n    vec3 n = vec3(0., 0., 1.);\n#elif defined(IS_LIT)\n    vec3 n = normalize(vNormal.xyz);\n#endif\n    \n#ifdef IS_TWOSIDED\n    if ((normz <= 0.) != front) discard;\n#endif\n\n#ifdef IS_LIT\n    vec3 eye = normalize(-vPosition.xyz/vPosition.w);\n    vec3 lightdir;\n    vec4 colDiff;\n    vec3 halfVec;\n    vec4 lighteffect = vec4(emission, 0.);\n    vec3 col;\n    float nDotL;\n#ifdef FIXED_QUADS\n    n = -faceforward(n, n, eye);\n#endif\n    \n#if NLIGHTS > 0\n    // Simulate two-sided lighting\n    if (n.z < 0.0)\n      n = -n;\n    for (int i=0;i<NLIGHTS;i++) {\n      colDiff = vec4(vCol.rgb * diffuse[i], vCol.a);\n      lightdir = lightDir[i];\n      if (!viewpoint[i]) {\n        if (finite[i]) {\n          lightdir = (mvMatrix * vec4(lightdir, 1.)).xyz;\n        } else {\n          lightdir = (mvMatrix * vec4(lightdir, 0.)).xyz;\n        }\n      }\n      if (!finite[i]) {\n        halfVec = normalize(lightdir + eye);\n      } else {\n        lightdir = normalize(lightdir - vPosition.xyz/vPosition.w);\n        halfVec = normalize(lightdir + eye);\n      }\n      col = ambient[i];\n      nDotL = dot(n, lightdir);\n      col = col + max(nDotL, 0.) * colDiff.rgb;\n      col = col + pow(max(dot(halfVec, n), 0.), shininess) * specular[i];\n      lighteffect = lighteffect + vec4(col, colDiff.a);\n    }\n#endif\n    \n#else // not IS_LIT\n    vec4 colDiff = vCol;\n    vec4 lighteffect = colDiff;\n#endif\n    \n#ifdef IS_TEXT\n    vec4 textureColor = lighteffect*texture2D(uSampler, vTexcoord);\n#endif\n    \n#ifdef HAS_TEXTURE\n\n// These calculations use the definitions from \n// https://docs.gl/gl3/glTexEnv\n\n#ifdef USE_ENVMAP\n    float m = 2.0 * sqrt(dot(vReflection, vReflection) + 2.0*vReflection.z + 1.0);\n    vec4 textureColor = texture2D(uSampler, vReflection.xy / m + vec2(0.5, 0.5));\n#else\n    vec4 textureColor = texture2D(uSampler, vTexcoord);\n#endif\n\n#ifdef TEXTURE_rgb\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(textureColor.rgb, lighteffect.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = lighteffect*vec4(textureColor.rgb, 1.);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - textureColor.rgb) * lighteffect.rgb, lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(lighteffect.rgb + textureColor.rgb, lighteffect.a);\n#endif\n\n#endif //TEXTURE_rgb\n        \n#ifdef TEXTURE_rgba\n\n#ifdef TEXMODE_replace\n// already done\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = lighteffect*textureColor;\n#endif\n\n#ifdef TEXMODE_decal\n    textureColor = vec4((1. - textureColor.a)*lighteffect.rgb) +\n                     textureColor.a*textureColor.rgb, \n                     lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - textureColor.rgb) * lighteffect.rgb,\n                    lighteffect.a*textureColor.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(lighteffect.rgb + textureColor.rgb,\n                    lighteffect.a*textureColor.a);\n#endif\n    \n#endif //TEXTURE_rgba\n    \n#ifdef TEXTURE_alpha\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(lighteffect.rgb, luminance);\n#endif \n\n#if defined(TEXMODE_modulate) || defined(TEXMODE_blend) || defined(TEXMODE_add)\n    textureColor = vec4(lighteffect.rgb, lighteffect.a*luminance);\n#endif\n \n#endif // TEXTURE_alpha\n    \n// The TEXTURE_luminance values are not from that reference    \n#ifdef TEXTURE_luminance\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(luminance, luminance, luminance, lighteffect.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = vec4(luminance*lighteffect.rgb, lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - luminance)*lighteffect.rgb,\n                        lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(luminance + lighteffect.rgb, lighteffect.a);\n#endif\n\n#endif // TEXTURE_luminance\n \n    \n#ifdef TEXTURE_luminance_alpha\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(luminance, luminance, luminance, textureColor.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = vec4(luminance*lighteffect.rgb, \n                        textureColor.a*lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - luminance)*lighteffect.rgb,\n                        textureColor.a*lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(luminance + lighteffect.rgb, \n                        textureColor.a*lighteffect.a);\n\n#endif\n\n#endif // TEXTURE_luminance_alpha\n    \n    fragColor = textureColor;\n\n#elif defined(IS_TEXT)\n    if (textureColor.a < 0.1)\n      discard;\n    else\n      fragColor = textureColor;\n#else\n    fragColor = lighteffect;\n#endif // HAS_TEXTURE\n    \n#ifdef HAS_FOG\n    // uFogParms elements: x = near, y = far, z = fogscale, w = (1-sin(FOV/2))/(1+sin(FOV/2))\n    // In Exp and Exp2: use density = density/far\n    // fogF will be the proportion of fog\n    // Initialize it to the linear value\n    float fogF;\n    if (uFogMode > 0) {\n      fogF = (uFogParms.y - vPosition.z/vPosition.w)/(uFogParms.y - uFogParms.x);\n      if (uFogMode > 1)\n        fogF = mix(uFogParms.w, 1.0, fogF);\n      fogF = fogF*uFogParms.z;\n      if (uFogMode == 2)\n        fogF = 1.0 - exp(-fogF);\n      // Docs are wrong: use (density*c)^2, not density*c^2\n      // https://gitlab.freedesktop.org/mesa/mesa/-/blob/master/src/mesa/swrast/s_fog.c#L58\n      else if (uFogMode == 3)\n        fogF = 1.0 - exp(-fogF*fogF);\n      fogF = clamp(fogF, 0.0, 1.0);\n      gl_FragColor = vec4(mix(fragColor.rgb, uFogColor, fogF), fragColor.a);\n    } else gl_FragColor = fragColor;\n#else\n    gl_FragColor = fragColor;\n#endif // HAS_FOG\n    \n}","players":[],"webGLoptions":{"preserveDrawingBuffer":true},"fastTransparency":true},"evals":[],"jsHooks":[]}</script>
<p id="rgl35284-aria" hidden>
</p>
</div>
</div>
<div id="examining-the-residuals" class="section level2 hasAnchor" number="23.12">
<h2><span class="header-section-number">23.12</span> Examining the residuals<a href="multiple-regression.html#examining-the-residuals" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Residual plots are useful for multiple regression, just as they were for bivariate regression.</p>
<p>First we plot the residuals versus each of the predictor variable individually.</p>
<div class="sourceCode" id="cb516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb516-1"><a href="multiple-regression.html#cb516-1" tabindex="-1"></a>wt.resids <span class="ot">&lt;-</span></span>
<span id="cb516-2"><a href="multiple-regression.html#cb516-2" tabindex="-1"></a>  mtcars.subset.augmented <span class="sc">%&gt;%</span></span>
<span id="cb516-3"><a href="multiple-regression.html#cb516-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> wt, <span class="at">y =</span> .resid)) <span class="sc">+</span></span>
<span id="cb516-4"><a href="multiple-regression.html#cb516-4" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb516-5"><a href="multiple-regression.html#cb516-5" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>, <span class="at">color=</span><span class="st">&#39;red&#39;</span>) <span class="sc">+</span> </span>
<span id="cb516-6"><a href="multiple-regression.html#cb516-6" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Weight&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Residuals&quot;</span>) </span>
<span id="cb516-7"><a href="multiple-regression.html#cb516-7" tabindex="-1"></a></span>
<span id="cb516-8"><a href="multiple-regression.html#cb516-8" tabindex="-1"></a>hp.resids <span class="ot">&lt;-</span></span>
<span id="cb516-9"><a href="multiple-regression.html#cb516-9" tabindex="-1"></a>  mtcars.subset.augmented <span class="sc">%&gt;%</span></span>
<span id="cb516-10"><a href="multiple-regression.html#cb516-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> hp, <span class="at">y =</span> .resid)) <span class="sc">+</span></span>
<span id="cb516-11"><a href="multiple-regression.html#cb516-11" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb516-12"><a href="multiple-regression.html#cb516-12" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>, <span class="at">color=</span><span class="st">&#39;red&#39;</span>) <span class="sc">+</span> </span>
<span id="cb516-13"><a href="multiple-regression.html#cb516-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Horsepower&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Residuals&quot;</span>) </span>
<span id="cb516-14"><a href="multiple-regression.html#cb516-14" tabindex="-1"></a></span>
<span id="cb516-15"><a href="multiple-regression.html#cb516-15" tabindex="-1"></a><span class="fu">plot_grid</span>(wt.resids, hp.resids, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>))</span></code></pre></div>
<p><img src="bio304-book_files/figure-html/unnamed-chunk-534-1.png" width="1152" /></p>
<p>And now we plot the residuals in 3D space, with a plane parallel to the xy-plane (wt, hp-plane) representing the plane about which the residuals should be homogeneously scattered if the assumptions of the linear regression model hold.</p>
<div class="sourceCode" id="cb517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb517-1"><a href="multiple-regression.html#cb517-1" tabindex="-1"></a>.resid <span class="ot">&lt;-</span> mtcars.subset.augmented<span class="sc">$</span>.resid</span>
<span id="cb517-2"><a href="multiple-regression.html#cb517-2" tabindex="-1"></a></span>
<span id="cb517-3"><a href="multiple-regression.html#cb517-3" tabindex="-1"></a><span class="co"># coefficients for plane perpindicular to </span></span>
<span id="cb517-4"><a href="multiple-regression.html#cb517-4" tabindex="-1"></a><span class="co"># xy-axis intercepting the z-axis at 0</span></span>
<span id="cb517-5"><a href="multiple-regression.html#cb517-5" tabindex="-1"></a>b1 <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb517-6"><a href="multiple-regression.html#cb517-6" tabindex="-1"></a>b2 <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb517-7"><a href="multiple-regression.html#cb517-7" tabindex="-1"></a>c <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb517-8"><a href="multiple-regression.html#cb517-8" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb517-9"><a href="multiple-regression.html#cb517-9" tabindex="-1"></a></span>
<span id="cb517-10"><a href="multiple-regression.html#cb517-10" tabindex="-1"></a><span class="fu">plot3d</span>(wt, hp, .resid, </span>
<span id="cb517-11"><a href="multiple-regression.html#cb517-11" tabindex="-1"></a>       <span class="at">type =</span> <span class="st">&quot;s&quot;</span>, </span>
<span id="cb517-12"><a href="multiple-regression.html#cb517-12" tabindex="-1"></a>       <span class="at">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb517-13"><a href="multiple-regression.html#cb517-13" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb517-14"><a href="multiple-regression.html#cb517-14" tabindex="-1"></a>       <span class="at">aspect =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>))</span>
<span id="cb517-15"><a href="multiple-regression.html#cb517-15" tabindex="-1"></a><span class="fu">rgl.planes</span>(b1, b2, c, a, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>)</span>
<span id="cb517-16"><a href="multiple-regression.html#cb517-16" tabindex="-1"></a><span class="fu">rglwidget</span>()</span></code></pre></div>
<div class="figure">
<div id="rgl34247" style="width:576px;height:384px;" class="rglWebGL html-widget" role="img" aria-labelledby="rgl34247-aria"></div>
<script type="application/json" data-for="rgl34247">{"x":{"material":{"color":"#BEBEBE","alpha":0.8980392217636108,"lit":true,"ambient":"#000000","specular":"#FFFFFF","emission":"#000000","shininess":50,"smooth":true,"front":"filled","back":"filled","size":3,"lwd":1,"fog":true,"point_antialias":false,"line_antialias":false,"texture":null,"textype":"rgb","texmode":"modulate","texmipmap":false,"texminfilter":"linear","texmagfilter":"linear","texenvmap":false,"depth_mask":true,"depth_test":"less","isTransparent":true,"polygon_offset":[0,0],"margin":"","floating":false,"tag":"","blend":["src_alpha","one_minus_src_alpha"]},"rootSubscene":6,"objects":{"38":{"id":38,"type":"spheres","material":{"alpha":1,"isTransparent":false},"vertices":"0","colors":"1","radii":[[4.087588787078857]],"centers":"2","ignoreExtent":false,"fastTransparency":true,"flags":32771},"40":{"id":40,"type":"text","material":{"alpha":1,"lit":false,"isTransparent":false,"margin":0,"floating":true,"edge":[0,1,1]},"vertices":"3","colors":"4","texts":[["wt"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"5","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"41":{"id":41,"type":"text","material":{"alpha":1,"lit":false,"isTransparent":false,"margin":1,"floating":true,"edge":[1,1,1]},"vertices":"6","colors":"7","texts":[["hp"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"8","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"42":{"id":42,"type":"text","material":{"alpha":1,"lit":false,"isTransparent":false,"margin":2,"floating":true,"edge":[1,1,1]},"vertices":"9","colors":"10","texts":[[".resid"]],"cex":[[1]],"adj":[[0.5,0.5,0.5]],"centers":"11","family":[["sans"]],"font":[[1]],"ignoreExtent":true,"flags":33808},"43":{"id":43,"type":"planes","material":{},"vertices":"12","colors":"14","offsets":[[0]],"centers":"15","normals":"13","ignoreExtent":true,"flags":32803},"10":{"id":10,"type":"light","vertices":[[0,0,1]],"colors":[[1,1,1,1],[1,1,1,1],[1,1,1,1]],"viewpoint":true,"finite":false},"12":{"id":12,"type":"background","material":{"alpha":1,"lit":false,"back":"lines","isTransparent":false},"colors":"16","centers":"17","sphere":false,"fogtype":"none","fogscale":1,"flags":32768},"39":{"id":39,"type":"bboxdeco","material":{"alpha":1,"front":"lines","back":"lines","isTransparent":false},"vertices":"18","colors":"19","axes":{"mode":["pretty","pretty","pretty"],"step":[1,50,2],"nticks":[5,5,5],"marklen":[15,15,15],"expand":[1.029999971389771,1.029999971389771,1.029999971389771]},"draw_front":true,"flags":32769},"6":{"id":6,"type":"subscene","par3d":{"antialias":4,"FOV":30,"ignoreExtent":false,"listeners":6,"mouseMode":{"none":"none","left":"trackball","right":"zoom","middle":"fov","wheel":"pull"},"observer":[0,0,1230.329833984375],"modelMatrix":[[83.61475372314453,0,0,-290.0177612304688],[0,0.3952172696590424,15.29456329345703,-91.10234832763672],[0,-1.085850477218628,5.566766262054443,-1025.541870117188],[0,0,0,1]],"projMatrix":[[3.732050895690918,0,0,0],[0,3.732050895690918,0,0],[0,0,-3.863702774047852,-4435.1962890625],[0,0,-1,0]],"skipRedraw":false,"userMatrix":[[1,0,0,0],[0,0.3420201433256682,0.9396926207859085,0],[0,-0.9396926207859085,0.3420201433256682,0],[0,0,0,1]],"userProjection":[[1,0,0,0],[0,1,0,0],[0,0,1,0],[0,0,0,1]],"scale":[83.61475372314453,1.155537962913513,16.2761344909668],"viewport":{"x":0,"y":0,"width":1,"height":1},"zoom":1,"bbox":[1.46411406993866,5.472885608673096,48.46260833740234,338.5373840332031,-4.192119598388672,6.104930877685547],"windowRect":[58,45,314,301],"family":"sans","font":1,"cex":1,"useFreeType":true,"fontname":"/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/rgl/fonts/FreeSans.ttf","maxClipPlanes":6,"glVersion":2.1,"activeSubscene":0},"embeddings":{"viewport":"replace","projection":"replace","model":"replace","mouse":"replace"},"objects":[12,39,38,40,41,42,43,10],"subscenes":[],"flags":34099}},"crosstalk":{"key":[],"group":[],"id":[],"options":[]},"width":576,"height":384,"buffer":{"accessors":[{"bufferView":0,"componentType":5126,"count":32,"type":"VEC3"},{"bufferView":1,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":2,"componentType":5126,"count":32,"type":"VEC3"},{"bufferView":3,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":4,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":5,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":6,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":7,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":8,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":9,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":10,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":11,"componentType":5126,"count":1,"type":"VEC3"},{"bufferView":12,"componentType":5126,"count":3,"type":"VEC3"},{"bufferView":13,"componentType":5120,"count":1,"type":"VEC3"},{"bufferView":14,"componentType":5126,"count":1,"type":"VEC4"},{"bufferView":15,"componentType":5126,"count":4,"type":"VEC3"},{"bufferView":16,"componentType":5121,"count":1,"type":"VEC4"},{"bufferView":17,"componentType":5121,"count":1,"type":"VEC3"},{"bufferView":18,"componentType":5126,"count":16,"type":"VEC3"},{"bufferView":19,"componentType":5121,"count":1,"type":"VEC4"}],"bufferViews":[{"buffer":0,"byteLength":384,"byteOffset":0},{"buffer":0,"byteLength":4,"byteOffset":384},{"buffer":0,"byteLength":384,"byteOffset":388},{"buffer":0,"byteLength":12,"byteOffset":772},{"buffer":0,"byteLength":4,"byteOffset":784},{"buffer":0,"byteLength":12,"byteOffset":788},{"buffer":0,"byteLength":12,"byteOffset":800},{"buffer":0,"byteLength":4,"byteOffset":812},{"buffer":0,"byteLength":12,"byteOffset":816},{"buffer":0,"byteLength":12,"byteOffset":828},{"buffer":0,"byteLength":4,"byteOffset":840},{"buffer":0,"byteLength":12,"byteOffset":844},{"buffer":0,"byteLength":36,"byteOffset":856},{"buffer":0,"byteLength":3,"byteOffset":892},{"buffer":0,"byteLength":16,"byteOffset":896},{"buffer":0,"byteLength":48,"byteOffset":912},{"buffer":0,"byteLength":4,"byteOffset":960},{"buffer":0,"byteLength":3,"byteOffset":964},{"buffer":0,"byteLength":192,"byteOffset":968},{"buffer":0,"byteLength":4,"byteOffset":1160}],"buffers":[{"byteLength":1164,"bytes":"FK4nQAAA3EIMoSTAAAA4QAAA3EKPr8q/4XoUQAAAukLQcx7Aj8JNQAAA3EIrOAo+9ihcQAAA\nL0Po1r4+pHBdQAAA0kKb7BfA4XpkQAAAdUMFR6a/9ihMQAAAeELHp8E/mplJQAAAvkJta04/\n9ihcQAAA9kKvike/9ihcQAAA9kJFfAvAcT2CQAAANEOmtCw/UrhuQAAANEOWKIM+hetxQAAA\nNEM3MdO/AACoQAAATUNlezc9aJGtQAAAV0MuxYQ/PQqrQAAAZkOMPbBAzcwMQAAAhEKQoblA\nUrjOPwAAUEIgN4s/SOHqPwAAgkJBUrtAj8IdQAAAwkJCiEXArkdhQAAAFkNj7VPACtdbQAAA\nFkMCOXzAj8J1QAAAdUN1QqC/exR2QAAAL0NJXhxAFK73PwAAhEIUP6e+w/UIQAAAtkKkFRm9\n/KnBPwAA4kK0VShASOFKQAAAhEPhGT+/rkcxQAAAL0NX2py/4XpkQACAp0NorBBAhesxQAAA\n2kIGtcq/AQAAARSuJ0AAANxCDKEkwAAAOEAAANxCj6/Kv+F6FEAAALpC0HMewI/CTUAAANxC\nKzgKPvYoXEAAAC9D6Na+PqRwXUAAANJCm+wXwOF6ZEAAAHVDBUemv/YoTEAAAHhCx6fBP5qZ\nSUAAAL5CbWtOP/YoXEAAAPZCr4pHv/YoXEAAAPZCRXwLwHE9gkAAADRDprQsP1K4bkAAADRD\nliiDPoXrcUAAADRDNzHTvwAAqEAAAE1DZXs3PWiRrUAAAFdDLsWEPz0Kq0AAAGZDjD2wQM3M\nDEAAAIRCkKG5QFK4zj8AAFBCIDeLP0jh6j8AAIJCQVK7QI/CHUAAAMJCQohFwK5HYUAAABZD\nY+1TwArXW0AAABZDAjl8wI/CdUAAAHVDdUKgv3sUdkAAAC9DSV4cQBSu9z8AAIRCFD+nvsP1\nCEAAALZCpBUZvfypwT8AAOJCtFUoQEjhSkAAAIRD4Rk/v65HMUAAAC9DV9qcv+F6ZEAAgKdD\naKwQQIXrMUAAANpCBrXKvwAAwH8AAIBAAACAPwAAAAEAAMB/AACAQAAAgD8AAMB/AACAQAAA\ngD8AAAABAADAfwAAgEAAAIA/AADAfwAAgEAAAIA/AAAAAQAAwH8AAIBAAACAPxdouz+22UFC\nAAAAABdouz/JRKlDAAAAAOEhr0DJRKlDAAAAAAAA/wC/vj4/v74+P7++Pj/m5WU/SDkzQIbY\ncUMAAAAAQ1+EQHonEUMAAAAAAADAfwAAwH8AAMB/AADAfwAAwH8AAMB/AQEBAQAAAAAAAABA\nAADAfwAAwH8AAEBAAADAfwAAwH8AAIBAAADAfwAAwH8AAKBAAADAfwAAwH8AAMB/AABIQgAA\nwH8AAMB/AADIQgAAwH8AAMB/AAAWQwAAwH8AAMB/AABIQwAAwH8AAMB/AAB6QwAAwH8AAMB/\nAACWQwAAwH8AAMB/AADAfwAAgMAAAMB/AADAfwAAAMAAAMB/AADAfwAAAAAAAMB/AADAfwAA\nAEAAAMB/AADAfwAAgEAAAMB/AADAfwAAwEAAAAAB"}]},"context":{"shiny":false,"rmarkdown":"bookdown::gitbook"},"vertexShader":"#line 2 1\n// File 1 is the vertex shader\n#ifdef GL_ES\n#ifdef GL_FRAGMENT_PRECISION_HIGH\nprecision highp float;\n#else\nprecision mediump float;\n#endif\n#endif\n\nattribute vec3 aPos;\nattribute vec4 aCol;\nuniform mat4 mvMatrix;\nuniform mat4 prMatrix;\nvarying vec4 vCol;\nvarying vec4 vPosition;\n\n#ifdef NEEDS_VNORMAL\nattribute vec3 aNorm;\nuniform mat4 normMatrix;\nvarying vec4 vNormal;\n#endif\n\n#if defined(HAS_TEXTURE) || defined (IS_TEXT)\nattribute vec2 aTexcoord;\nvarying vec2 vTexcoord;\n#endif\n\n#ifdef FIXED_SIZE\nuniform vec3 textScale;\n#endif\n\n#ifdef FIXED_QUADS\nattribute vec3 aOfs;\n#endif\n\n#ifdef IS_TWOSIDED\n#ifdef HAS_NORMALS\nvarying float normz;\nuniform mat4 invPrMatrix;\n#else\nattribute vec3 aPos1;\nattribute vec3 aPos2;\nvarying float normz;\n#endif\n#endif // IS_TWOSIDED\n\n#ifdef FAT_LINES\nattribute vec3 aNext;\nattribute vec2 aPoint;\nvarying vec2 vPoint;\nvarying float vLength;\nuniform float uAspect;\nuniform float uLwd;\n#endif\n\n#ifdef USE_ENVMAP\nvarying vec3 vReflection;\n#endif\n\nvoid main(void) {\n  \n#ifndef IS_BRUSH\n#if defined(NCLIPPLANES) || !defined(FIXED_QUADS) || defined(HAS_FOG) || defined(USE_ENVMAP)\n  vPosition = mvMatrix * vec4(aPos, 1.);\n#endif\n  \n#ifndef FIXED_QUADS\n  gl_Position = prMatrix * vPosition;\n#endif\n#endif // !IS_BRUSH\n  \n#ifdef IS_POINTS\n  gl_PointSize = POINTSIZE;\n#endif\n  \n  vCol = aCol;\n  \n// USE_ENVMAP implies NEEDS_VNORMAL\n\n#ifdef NEEDS_VNORMAL\n  vNormal = normMatrix * vec4(-aNorm, dot(aNorm, aPos));\n#endif\n\n#ifdef USE_ENVMAP\n  vReflection = normalize(reflect(vPosition.xyz/vPosition.w, \n                        normalize(vNormal.xyz/vNormal.w)));\n#endif\n  \n#ifdef IS_TWOSIDED\n#ifdef HAS_NORMALS\n  /* normz should be calculated *after* projection */\n  normz = (invPrMatrix*vNormal).z;\n#else\n  vec4 pos1 = prMatrix*(mvMatrix*vec4(aPos1, 1.));\n  pos1 = pos1/pos1.w - gl_Position/gl_Position.w;\n  vec4 pos2 = prMatrix*(mvMatrix*vec4(aPos2, 1.));\n  pos2 = pos2/pos2.w - gl_Position/gl_Position.w;\n  normz = pos1.x*pos2.y - pos1.y*pos2.x;\n#endif\n#endif // IS_TWOSIDED\n  \n#ifdef NEEDS_VNORMAL\n  vNormal = vec4(normalize(vNormal.xyz), 1);\n#endif\n  \n#if defined(HAS_TEXTURE) || defined(IS_TEXT)\n  vTexcoord = aTexcoord;\n#endif\n  \n#if defined(FIXED_SIZE) && !defined(ROTATING)\n  vec4 pos = prMatrix * mvMatrix * vec4(aPos, 1.);\n  pos = pos/pos.w;\n  gl_Position = pos + vec4(aOfs*textScale, 0.);\n#endif\n  \n#if defined(IS_SPRITES) && !defined(FIXED_SIZE)\n  vec4 pos = mvMatrix * vec4(aPos, 1.);\n  pos = pos/pos.w + vec4(aOfs,  0.);\n  gl_Position = prMatrix*pos;\n#endif\n  \n#ifdef FAT_LINES\n  /* This code was inspired by Matt Deslauriers' code in \n   https://mattdesl.svbtle.com/drawing-lines-is-hard */\n  vec2 aspectVec = vec2(uAspect, 1.0);\n  mat4 projViewModel = prMatrix * mvMatrix;\n  vec4 currentProjected = projViewModel * vec4(aPos, 1.0);\n  currentProjected = currentProjected/currentProjected.w;\n  vec4 nextProjected = projViewModel * vec4(aNext, 1.0);\n  vec2 currentScreen = currentProjected.xy * aspectVec;\n  vec2 nextScreen = (nextProjected.xy / nextProjected.w) * aspectVec;\n  float len = uLwd;\n  vec2 dir = vec2(1.0, 0.0);\n  vPoint = aPoint;\n  vLength = length(nextScreen - currentScreen)/2.0;\n  vLength = vLength/(vLength + len);\n  if (vLength > 0.0) {\n    dir = normalize(nextScreen - currentScreen);\n  }\n  vec2 normal = vec2(-dir.y, dir.x);\n  dir.x /= uAspect;\n  normal.x /= uAspect;\n  vec4 offset = vec4(len*(normal*aPoint.x*aPoint.y - dir), 0.0, 0.0);\n  gl_Position = currentProjected + offset;\n#endif\n  \n#ifdef IS_BRUSH\n  gl_Position = vec4(aPos, 1.);\n#endif\n}","fragmentShader":"#line 2 2\n// File 2 is the fragment shader\n#ifdef GL_ES\n#ifdef GL_FRAGMENT_PRECISION_HIGH\nprecision highp float;\n#else\nprecision mediump float;\n#endif\n#endif\nvarying vec4 vCol; // carries alpha\nvarying vec4 vPosition;\n#if defined(HAS_TEXTURE) || defined (IS_TEXT)\nvarying vec2 vTexcoord;\nuniform sampler2D uSampler;\n#endif\n\n#ifdef HAS_FOG\nuniform int uFogMode;\nuniform vec3 uFogColor;\nuniform vec4 uFogParms;\n#endif\n\n#if defined(IS_LIT) && !defined(FIXED_QUADS)\nvarying vec4 vNormal;\n#endif\n\n#if NCLIPPLANES > 0\nuniform vec4 vClipplane[NCLIPPLANES];\n#endif\n\n#if NLIGHTS > 0\nuniform mat4 mvMatrix;\n#endif\n\n#ifdef IS_LIT\nuniform vec3 emission;\nuniform float shininess;\n#if NLIGHTS > 0\nuniform vec3 ambient[NLIGHTS];\nuniform vec3 specular[NLIGHTS]; // light*material\nuniform vec3 diffuse[NLIGHTS];\nuniform vec3 lightDir[NLIGHTS];\nuniform bool viewpoint[NLIGHTS];\nuniform bool finite[NLIGHTS];\n#endif\n#endif // IS_LIT\n\n#ifdef IS_TWOSIDED\nuniform bool front;\nvarying float normz;\n#endif\n\n#ifdef FAT_LINES\nvarying vec2 vPoint;\nvarying float vLength;\n#endif\n\n#ifdef USE_ENVMAP\nvarying vec3 vReflection;\n#endif\n\nvoid main(void) {\n  vec4 fragColor;\n#ifdef FAT_LINES\n  vec2 point = vPoint;\n  bool neg = point.y < 0.0;\n  point.y = neg ? (point.y + vLength)/(1.0 - vLength) :\n                 -(point.y - vLength)/(1.0 - vLength);\n#if defined(IS_TRANSPARENT) && defined(IS_LINESTRIP)\n  if (neg && length(point) <= 1.0) discard;\n#endif\n  point.y = min(point.y, 0.0);\n  if (length(point) > 1.0) discard;\n#endif // FAT_LINES\n  \n#ifdef ROUND_POINTS\n  vec2 coord = gl_PointCoord - vec2(0.5);\n  if (length(coord) > 0.5) discard;\n#endif\n  \n#if NCLIPPLANES > 0\n  for (int i = 0; i < NCLIPPLANES; i++)\n    if (dot(vPosition, vClipplane[i]) < 0.0) discard;\n#endif\n    \n#ifdef FIXED_QUADS\n    vec3 n = vec3(0., 0., 1.);\n#elif defined(IS_LIT)\n    vec3 n = normalize(vNormal.xyz);\n#endif\n    \n#ifdef IS_TWOSIDED\n    if ((normz <= 0.) != front) discard;\n#endif\n\n#ifdef IS_LIT\n    vec3 eye = normalize(-vPosition.xyz/vPosition.w);\n    vec3 lightdir;\n    vec4 colDiff;\n    vec3 halfVec;\n    vec4 lighteffect = vec4(emission, 0.);\n    vec3 col;\n    float nDotL;\n#ifdef FIXED_QUADS\n    n = -faceforward(n, n, eye);\n#endif\n    \n#if NLIGHTS > 0\n    // Simulate two-sided lighting\n    if (n.z < 0.0)\n      n = -n;\n    for (int i=0;i<NLIGHTS;i++) {\n      colDiff = vec4(vCol.rgb * diffuse[i], vCol.a);\n      lightdir = lightDir[i];\n      if (!viewpoint[i]) {\n        if (finite[i]) {\n          lightdir = (mvMatrix * vec4(lightdir, 1.)).xyz;\n        } else {\n          lightdir = (mvMatrix * vec4(lightdir, 0.)).xyz;\n        }\n      }\n      if (!finite[i]) {\n        halfVec = normalize(lightdir + eye);\n      } else {\n        lightdir = normalize(lightdir - vPosition.xyz/vPosition.w);\n        halfVec = normalize(lightdir + eye);\n      }\n      col = ambient[i];\n      nDotL = dot(n, lightdir);\n      col = col + max(nDotL, 0.) * colDiff.rgb;\n      col = col + pow(max(dot(halfVec, n), 0.), shininess) * specular[i];\n      lighteffect = lighteffect + vec4(col, colDiff.a);\n    }\n#endif\n    \n#else // not IS_LIT\n    vec4 colDiff = vCol;\n    vec4 lighteffect = colDiff;\n#endif\n    \n#ifdef IS_TEXT\n    vec4 textureColor = lighteffect*texture2D(uSampler, vTexcoord);\n#endif\n    \n#ifdef HAS_TEXTURE\n\n// These calculations use the definitions from \n// https://docs.gl/gl3/glTexEnv\n\n#ifdef USE_ENVMAP\n    float m = 2.0 * sqrt(dot(vReflection, vReflection) + 2.0*vReflection.z + 1.0);\n    vec4 textureColor = texture2D(uSampler, vReflection.xy / m + vec2(0.5, 0.5));\n#else\n    vec4 textureColor = texture2D(uSampler, vTexcoord);\n#endif\n\n#ifdef TEXTURE_rgb\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(textureColor.rgb, lighteffect.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = lighteffect*vec4(textureColor.rgb, 1.);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - textureColor.rgb) * lighteffect.rgb, lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(lighteffect.rgb + textureColor.rgb, lighteffect.a);\n#endif\n\n#endif //TEXTURE_rgb\n        \n#ifdef TEXTURE_rgba\n\n#ifdef TEXMODE_replace\n// already done\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = lighteffect*textureColor;\n#endif\n\n#ifdef TEXMODE_decal\n    textureColor = vec4((1. - textureColor.a)*lighteffect.rgb) +\n                     textureColor.a*textureColor.rgb, \n                     lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - textureColor.rgb) * lighteffect.rgb,\n                    lighteffect.a*textureColor.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(lighteffect.rgb + textureColor.rgb,\n                    lighteffect.a*textureColor.a);\n#endif\n    \n#endif //TEXTURE_rgba\n    \n#ifdef TEXTURE_alpha\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(lighteffect.rgb, luminance);\n#endif \n\n#if defined(TEXMODE_modulate) || defined(TEXMODE_blend) || defined(TEXMODE_add)\n    textureColor = vec4(lighteffect.rgb, lighteffect.a*luminance);\n#endif\n \n#endif // TEXTURE_alpha\n    \n// The TEXTURE_luminance values are not from that reference    \n#ifdef TEXTURE_luminance\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(luminance, luminance, luminance, lighteffect.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = vec4(luminance*lighteffect.rgb, lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - luminance)*lighteffect.rgb,\n                        lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(luminance + lighteffect.rgb, lighteffect.a);\n#endif\n\n#endif // TEXTURE_luminance\n \n    \n#ifdef TEXTURE_luminance_alpha\n    float luminance = dot(vec3(1.,1.,1.),textureColor.rgb)/3.;\n\n#if defined(TEXMODE_replace) || defined(TEXMODE_decal)\n    textureColor = vec4(luminance, luminance, luminance, textureColor.a);\n#endif \n\n#ifdef TEXMODE_modulate\n    textureColor = vec4(luminance*lighteffect.rgb, \n                        textureColor.a*lighteffect.a);\n#endif\n\n#ifdef TEXMODE_blend\n    textureColor = vec4((1. - luminance)*lighteffect.rgb,\n                        textureColor.a*lighteffect.a);\n#endif\n\n#ifdef TEXMODE_add\n    textureColor = vec4(luminance + lighteffect.rgb, \n                        textureColor.a*lighteffect.a);\n\n#endif\n\n#endif // TEXTURE_luminance_alpha\n    \n    fragColor = textureColor;\n\n#elif defined(IS_TEXT)\n    if (textureColor.a < 0.1)\n      discard;\n    else\n      fragColor = textureColor;\n#else\n    fragColor = lighteffect;\n#endif // HAS_TEXTURE\n    \n#ifdef HAS_FOG\n    // uFogParms elements: x = near, y = far, z = fogscale, w = (1-sin(FOV/2))/(1+sin(FOV/2))\n    // In Exp and Exp2: use density = density/far\n    // fogF will be the proportion of fog\n    // Initialize it to the linear value\n    float fogF;\n    if (uFogMode > 0) {\n      fogF = (uFogParms.y - vPosition.z/vPosition.w)/(uFogParms.y - uFogParms.x);\n      if (uFogMode > 1)\n        fogF = mix(uFogParms.w, 1.0, fogF);\n      fogF = fogF*uFogParms.z;\n      if (uFogMode == 2)\n        fogF = 1.0 - exp(-fogF);\n      // Docs are wrong: use (density*c)^2, not density*c^2\n      // https://gitlab.freedesktop.org/mesa/mesa/-/blob/master/src/mesa/swrast/s_fog.c#L58\n      else if (uFogMode == 3)\n        fogF = 1.0 - exp(-fogF*fogF);\n      fogF = clamp(fogF, 0.0, 1.0);\n      gl_FragColor = vec4(mix(fragColor.rgb, uFogColor, fogF), fragColor.a);\n    } else gl_FragColor = fragColor;\n#else\n    gl_FragColor = fragColor;\n#endif // HAS_FOG\n    \n}","players":[],"webGLoptions":{"preserveDrawingBuffer":true},"fastTransparency":true},"evals":[],"jsHooks":[]}</script>
<p id="rgl34247-aria" hidden>
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bivariate-linear-regression.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="logistic-regression.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["bio304-book.pdf"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  },
  "toolbar": {
    "position": "fixed"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
